# Oxford Planarian Bioinformatics Community

<img src="Asset 6OxfordPlanarian.png" alt="hi" class="inline"/>

Jakke Neiro<sup>1</sup> & Nathan Kenny<sup>2</sup>

1. Aboobaker lab, University of Oxford, <jakke.neiro@zoo.ox.ac.uk>

2. Solana lab, Oxford Brookes University, <nkenny@brookes.ac.uk>

This site includes code, files and other resources generated by the Oxford Planarian Bioinformatics Community.

## Annotation of the *Schmidtea mediterranea* genome
A new expression-driven annotation of the sexual planarian genome (*S. mediterranea*) was generated based on total RNA-seq data from this species. The annotation is based on 183 planarian RNA-seq samples (see the [sample list](https://github.com/jakke-neiro/Oxplatys/blob/gh-pages/Smed_annotation.list)), the [PlanMine sexual genome annotation (SMESG-high)](http://planmine.mpi-cbg.de/planmine/aspect.do?name=Gene%20Predictions), and the [PlanMine sexual genome (SMESG.1)](http://planmine.mpi-cbg.de/planmine/aspect.do?name=Genomes). 

[Notebook for annotation pipline](https://github.com/jakke-neiro/Oxplatys/blob/gh-pages/Annotation_Smed.ipynb)

[Download *S. mediterranea* annotation](https://github.com/jakke-neiro/Oxplatys/raw/gh-pages/Schmidtea_mediterranea_Oxford_v1.gtf.zip)

## Annotation of the *Dugesia japonica* genome
Similarly to the *S. mediterranea* annotation, the new expression-driven *D. japonica* annotation was generated based on 43 RNA-seq samples (see the [sample list](https://github.com/jakke-neiro/oxplatys/blob/gh-pages/Djap_annotation.list), the [Augustus-derived annotation](http://planarian.jp/index.html) and the [genome assembly](http://planarian.jp/index.html) at planarian.jp.  

[Download *D. japonica* annotation](https://github.com/jakke-neiro/Oxplatys/raw/gh-pages/Dugesia_japonica_Oxford_v1.gtf.zip)

## rds files of analyses presented in ACME paper
These files allow easy uploading to R of the analyses presented in the García-Castro et al. 2020 ACME paper.
[Download link](https://doi.org/10.6084/m9.figshare.13614605)

## Publications
### ACME dissociation: a versatile cell fixation-dissociation method for single-cell transcriptomics
Helena García-Castro, Nathan J. Kenny, Marta Iglesias, Patricia Álvarez-Campos, Vincent Mason, Anamaria Elek, Anna Schönauer, Victoria A. Sleight, Jakke Neiro, Aziz Aboobaker, Jon Permanyer, Manuel Irimia, Arnau Sebé-Pedrós & Jordi Solana

Single-cell sequencing technologies are revolutionizing biology, but they are limited by the need to dissociate live samples. Here, we present ACME (ACetic-MEthanol), a dissociation approach for single-cell transcriptomics that simultaneously fixes cells. ACME-dissociated cells have high RNA integrity, can be cryopreserved multiple times, and are sortable and permeable. As a proof of principle, we provide single-cell transcriptomic data of different species, using both droplet-based and combinatorial barcoding single-cell methods. ACME uses affordable reagents, can be done in most laboratories and even in the field, and thus will accelerate our knowledge of cell types across the tree of life.

