{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "e55083c0",
   "metadata": {},
   "source": [
    "# 2-4-Transcription-factors-literature-2015-2021\n",
    "Jakke Neiro$^1$\n",
    "1. Aboobaker laboratory, Department of Zoology, University of Oxford"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "270694cb",
   "metadata": {},
   "source": [
    "## Contents of notebook\n",
    "* 1. Introduction\n",
    "* 2. 2015\n",
    "    * 2.1 A generic and cell-type-specific wound response precedes regeneration in planarians\n",
    "    * 2.2 A mex3 homolog is required for differentiation during planarian stem cell lineage development\n",
    "    * 2.3 Egr-5 is a post-mitotic regulator of planarian epidermal differentiation\n",
    "* 3. 2016\n",
    "    * 3.1 HOX gene complement and expression in the planarian Schmidtea mediterranea\n",
    "    * 3.2 Forkhead containing transcription factor Albino controls tetrapyrrole-based body pigmentation in planarian\n",
    "    * 3.3 A functional genomics screen in planarians reveals regulators of whole-brain regeneration\n",
    "    * 3.4 Essential role for the planarian intestinal GATA transcription factor in stem cells and regeneration\n",
    "    * 3.5 NF-YB Regulates Spermatogonial Stem Cell Self-Renewal and Proliferation in the Planarian Schmidtea mediterranea\n",
    "    * 3.6 In silico lineage tracing through single cell transcriptomics identifies a neural stem cell population in planarians\n",
    "    * 3.7 omputational discovery and in vivo validation of hnf4 as a regulatory gene in planarian regeneration\n",
    "* 4. 2017\n",
    "    * 4.1 Orthogonal muscle fibres have different instructive roles in planarian regeneration\n",
    "    * 4.2 FOX and ETS family transcription factors regulate the pigment cell lineage in planarians\n",
    "    * 4.3 Epithelial-mesenchymal transition transcription factors control pluripotent adult stem cell migration in vivo in planarians\n",
    "    * 4.4 The pioneer factor Smed-gata456-1 is required for gut cell differentiation and maintenance in planarians\n",
    "    * 4.5 Eye absence does not regulate planarian stem cells during eye regeneration\n",
    "    * 4.6 Generic wound signals initiate regeneration in missing-tissue contexts\n",
    "    * 4.7 Planarian epidermal stem cells respond to positional cues to promote cell type diversity\n",
    "    * 4.8 Antagonistic Self-Organizing Patterning Systems Control Maintenance and Regeneration of the Anteroposterior Axis in Planarians\n",
    "* 5. 2018\n",
    "    * 5.1 foxF-1 controls specification of non-body wall muscle and phagocytic cells in planarians\n",
    "    * 5.2 Prospectively isolated tetraspanin+ neoblasts are adult pluripotent stem cells underlying planaria regeneration\n",
    "    * 5.3 The planarian TCF/LEF factor Smed-tcf1 is required for the regeneration of dorsal-lateral neuronal subtypes\n",
    "    * 5.4 Smed-myb-1 Specifies Early Temporal Identity during Planarian Epidermal Differentiation\n",
    "    * 5.5 SoxB1 Activity Regulates Sensory Neuron Regeneration, Maintenance, and Function in Planarians\n",
    "    * 5.6 Comparative transcriptomic analyses and single-cell RNA sequencing of the freshwater planarian Schmidtea mediterranea identify major cell types and pathway conservation\n",
    "    * 5.7 Conservation of epigenetic regulation by the MLL3/4 tumour suppressor in planarian pluripotent stem cells\n",
    "    * 5.8 Cellular, ultrastructural and molecular analyses of epidermal cell development in the planarian S. mediterranea\n",
    "    * 5.9 Cell type transcriptome atlas for the planarian Schmidtea mediterranea\n",
    "    * 5.10 \n",
    "* 6. 2019\n",
    "    * 6.1 Neoblast-enriched zinc finger protein FIR1 triggers local proliferation during planarian regeneration\n",
    "    * 6.2 A small set of conserved genes, including sp5 and Hox, are activated by Wnt signaling in the posterior of planarians and acoels\n",
    "    * 6.3 Nuclear receptor NR4A is required for patterning at the ends of the planarian anterior-posterior axis\n",
    "    * 6.4 \n",
    "* 7. 2020\n",
    "    * 7.1 Muscle and neuronal guidepost-like cells facilitate planarian visual system regeneration\n",
    "    * 7.2 Cell-type diversity and regionalized gene expression in the planarian intestine\n",
    "    * 7.3 \n",
    "* 8. 2021\n",
    "    * 8.1 Analysis of Fox genes in Schmidtea mediterranea reveals new families and a conserved role of Smed-foxO in controlling cell death\n",
    "    * 8.2 Planarian stem cells specify fate yet retain potency during the cell cycle"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a7ca926b",
   "metadata": {},
   "source": [
    "# Files\n",
    "* Input: fasta files\n",
    "* Output: ID correspondence csv files"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c54f346f",
   "metadata": {},
   "source": [
    "# 1. Introduction"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6e6db4ca",
   "metadata": {},
   "source": [
    "The planarian literature on transcription factors between 2015 and 2021 was reviewed and the correspondence between TFs described in the literature and TFs decribed in this study was established."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "05703c60",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "gffcmp = pd.read_csv(\"/hydra/sexual_genome_annotation_files/ncrna_Neiro/gffcmp.stringtie_merged.gtf.tmap\", sep=\"\\t\")\n",
    "gene2transcript = gffcmp.iloc[:,[0,3,4]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "faf1eaa9",
   "metadata": {},
   "outputs": [],
   "source": [
    "def tf_table(blast_table, reference_name, dataframe):\n",
    "    transcript_id = []\n",
    "    gene_id_rink = []\n",
    "    gene_id_neiro = []\n",
    "    original_id = []\n",
    "    original_name = []\n",
    "    for i in range(len(blast_table)):\n",
    "        t_id = blast_table.iloc[i,1]\n",
    "        o_id = blast_table.iloc[i,0]\n",
    "        transcript_id.append(t_id)\n",
    "        gene_id_rink.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,0])\n",
    "        gene_id_neiro.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,1])\n",
    "        original_id.append(o_id.rpartition(\".\")[0])\n",
    "        original_name.append(dataframe[dataframe[\"ID\"] == o_id].iloc[0,1])\n",
    "    source = [reference_name] * len(original_name)\n",
    "    TF_table = pd.DataFrame({\"Transcript.ID\": transcript_id, \"Rink.ID\": gene_id_rink, \"Neiro.ID\": gene_id_neiro, \"Original.ID\": original_id, \"Original.name\": original_name, \"Source\": source})\n",
    "    return TF_table"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "61382984",
   "metadata": {},
   "source": [
    "# 2. 2015"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3d65172b",
   "metadata": {},
   "source": [
    "## 2.1 A generic and cell-type-specific wound response precedes regeneration in planarians\n",
    "Wurtzel, O., Cote, L.E., Poirier, A., Satija, R., Regev, A. and Reddien, P.W., 2015. A generic and cell-type-specific wound response precedes regeneration in planarians. Developmental cell, 35(5), pp.632-645."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c3f8256",
   "metadata": {},
   "source": [
    "* znf91: dd_Smed_v4_7664_0_1\n",
    "* zfp-1: dd_Smed_v4_8720_0_1\n",
    "* hnf4: dd_Smed_v4_1694_0_1\n",
    "* egr-2: dd_Smed_v4_9273_0_1\n",
    "* klf: dd_Smed_v4_3638_0_1\n",
    "* mex-3: dd_Smed_v4_6053_0_1\n",
    "* egr-l1: dd_Smed_v4_7731_0_1\n",
    "* runt-1: dd_Smed_v4_16222_0_1\n",
    "* fos-1: dd_Smed_v4_2789_0_1\n",
    "* egr-4: dd_Smed_v4_9410_0_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "adb774aa",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "echo \"dd_Smed_v6_7664_0_1\" > Wurtzel2015.list\n",
    "echo \"dd_Smed_v6_8720_0_1\" >> Wurtzel2015.list\n",
    "echo \"dd_Smed_v6_1694_0_1\" >> Wurtzel2015.list\n",
    "echo \"dd_Smed_v6_9273_0_1\" >> Wurtzel2015.list\n",
    "echo \"dd_Smed_v6_3638_0_1\" >> Wurtzel2015.list\n",
    "echo \"dd_Smed_v6_6053_0_1\" >> Wurtzel2015.list\n",
    "echo \"dd_Smed_v6_7731_0_1\" >> Wurtzel2015.list\n",
    "echo \"dd_Smed_v6_16222_0_1\" >> Wurtzel2015.list\n",
    "echo \"dd_Smed_v6_2789_0_1\" >> Wurtzel2015.list\n",
    "echo \"dd_Smed_v6_9410_0_1\" >> Wurtzel2015.list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "8881e1f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "seqtk subseq /hydra/sexual_genome_annotation_files/sexual_genome_annotation/dd_Smed_v6.pcf.contigs.fasta Wurtzel2015.list > Wurtzel2015.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "26b9d8df",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Wurtzel2015.fa -out blast_Wurtzel2015 -outfmt 6 &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "70cd079e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Wurtzel2015 = pd.read_csv(\"/hydra/TF_data/blast_Wurtzel2015\", sep=\"\\t\", header=None)\n",
    "blast_Wurtzel2015.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Wurtzel2015 = blast_Wurtzel2015.loc[blast_Wurtzel2015.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "e3b588f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "Wurtzel2015 = pd.DataFrame({\"ID\": [\"dd_Smed_v6_7664_0_1\", \"dd_Smed_v6_8720_0_1\", \"dd_Smed_v6_1694_0_1\", \"dd_Smed_v6_9273_0_1\", \n",
    "                                   \"dd_Smed_v6_3638_0_1\", \"dd_Smed_v6_6053_0_1\", \"dd_Smed_v6_7731_0_1\", \"dd_Smed_v6_16222_0_1\", \"dd_Smed_v6_2789_0_1\", \"dd_Smed_v6_9410_0_1\"], \n",
    "                           \"Name\": [\"znf91\", \"zfp-1\", \"hnf4\", \"egr-2\", \"klf\", \"mex3\", \"egr-l1\", \"runt-1\", \"fos-1\", \"egr-4\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "3a1be278",
   "metadata": {},
   "outputs": [],
   "source": [
    "transcript_id = []\n",
    "gene_id_rink = []\n",
    "gene_id_neiro = []\n",
    "original_id = []\n",
    "original_name = []\n",
    "for i in range(len(blast_Wurtzel2015)):\n",
    "    t_id = blast_Wurtzel2015.iloc[i,1]\n",
    "    o_id = blast_Wurtzel2015.iloc[i,0]\n",
    "    transcript_id.append(t_id)\n",
    "    gene_id_rink.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,0])\n",
    "    gene_id_neiro.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,1])\n",
    "    original_id.append(o_id)\n",
    "    original_name.append(Wurtzel2015[Wurtzel2015[\"ID\"] == o_id].iloc[0,1])\n",
    "source = [\"Wurtzel et al. (2015)\"] * len(original_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "590e2c37",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Wurtzel2015 = pd.DataFrame({\"Transcript.ID\": transcript_id, \"Rink.ID\": gene_id_rink, \"Neiro.ID\": gene_id_neiro, \"Original.ID\": original_id, \"Original.name\": original_name, \"Source\": source})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "c8667db6",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Wurtzel2015.to_csv(\"/hydra/TF_data/TF_Wurtzel2015.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "75e9af9f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SMEST014779001.1</td>\n",
       "      <td>SMESG000014779.1</td>\n",
       "      <td>MSTRG.4469</td>\n",
       "      <td>dd_Smed_v6_16222_0_1</td>\n",
       "      <td>runt-1</td>\n",
       "      <td>Wurtzel et al. (2015)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMEST014588001.1</td>\n",
       "      <td>SMESG000014588.1</td>\n",
       "      <td>MSTRG.4403</td>\n",
       "      <td>dd_Smed_v6_1694_0_1</td>\n",
       "      <td>hnf4</td>\n",
       "      <td>Wurtzel et al. (2015)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SMEST003328001.1</td>\n",
       "      <td>SMESG000003328.1</td>\n",
       "      <td>MSTRG.707</td>\n",
       "      <td>dd_Smed_v6_2789_0_1</td>\n",
       "      <td>fos-1</td>\n",
       "      <td>Wurtzel et al. (2015)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MSTRG.396.5</td>\n",
       "      <td>SMESG000025482.1</td>\n",
       "      <td>MSTRG.396</td>\n",
       "      <td>dd_Smed_v6_3638_0_1</td>\n",
       "      <td>klf</td>\n",
       "      <td>Wurtzel et al. (2015)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SMEST071674001.1</td>\n",
       "      <td>SMESG000071674.1</td>\n",
       "      <td>MSTRG.21084</td>\n",
       "      <td>dd_Smed_v6_6053_0_1</td>\n",
       "      <td>mex3</td>\n",
       "      <td>Wurtzel et al. (2015)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SMEST026243001.1</td>\n",
       "      <td>SMESG000026243.1</td>\n",
       "      <td>MSTRG.624</td>\n",
       "      <td>dd_Smed_v6_7664_0_1</td>\n",
       "      <td>znf91</td>\n",
       "      <td>Wurtzel et al. (2015)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SMEST000960001.1</td>\n",
       "      <td>SMESG000000960.1</td>\n",
       "      <td>MSTRG.1218</td>\n",
       "      <td>dd_Smed_v6_7731_0_1</td>\n",
       "      <td>egr-l1</td>\n",
       "      <td>Wurtzel et al. (2015)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MSTRG.19740.1</td>\n",
       "      <td>SMESG000066384.1</td>\n",
       "      <td>MSTRG.19740</td>\n",
       "      <td>dd_Smed_v6_8720_0_1</td>\n",
       "      <td>zfp-1</td>\n",
       "      <td>Wurtzel et al. (2015)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SMEST004840001.1</td>\n",
       "      <td>SMESG000004840.1</td>\n",
       "      <td>MSTRG.2271</td>\n",
       "      <td>dd_Smed_v6_9273_0_1</td>\n",
       "      <td>egr-2</td>\n",
       "      <td>Wurtzel et al. (2015)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MSTRG.9193.1</td>\n",
       "      <td>SMESG000028801.1</td>\n",
       "      <td>MSTRG.9193</td>\n",
       "      <td>dd_Smed_v6_9410_0_1</td>\n",
       "      <td>egr-4</td>\n",
       "      <td>Wurtzel et al. (2015)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Transcript.ID           Rink.ID     Neiro.ID           Original.ID  \\\n",
       "0  SMEST014779001.1  SMESG000014779.1   MSTRG.4469  dd_Smed_v6_16222_0_1   \n",
       "1  SMEST014588001.1  SMESG000014588.1   MSTRG.4403   dd_Smed_v6_1694_0_1   \n",
       "2  SMEST003328001.1  SMESG000003328.1    MSTRG.707   dd_Smed_v6_2789_0_1   \n",
       "3       MSTRG.396.5  SMESG000025482.1    MSTRG.396   dd_Smed_v6_3638_0_1   \n",
       "4  SMEST071674001.1  SMESG000071674.1  MSTRG.21084   dd_Smed_v6_6053_0_1   \n",
       "5  SMEST026243001.1  SMESG000026243.1    MSTRG.624   dd_Smed_v6_7664_0_1   \n",
       "6  SMEST000960001.1  SMESG000000960.1   MSTRG.1218   dd_Smed_v6_7731_0_1   \n",
       "7     MSTRG.19740.1  SMESG000066384.1  MSTRG.19740   dd_Smed_v6_8720_0_1   \n",
       "8  SMEST004840001.1  SMESG000004840.1   MSTRG.2271   dd_Smed_v6_9273_0_1   \n",
       "9      MSTRG.9193.1  SMESG000028801.1   MSTRG.9193   dd_Smed_v6_9410_0_1   \n",
       "\n",
       "  Original.name                 Source  \n",
       "0        runt-1  Wurtzel et al. (2015)  \n",
       "1          hnf4  Wurtzel et al. (2015)  \n",
       "2         fos-1  Wurtzel et al. (2015)  \n",
       "3           klf  Wurtzel et al. (2015)  \n",
       "4          mex3  Wurtzel et al. (2015)  \n",
       "5         znf91  Wurtzel et al. (2015)  \n",
       "6        egr-l1  Wurtzel et al. (2015)  \n",
       "7         zfp-1  Wurtzel et al. (2015)  \n",
       "8         egr-2  Wurtzel et al. (2015)  \n",
       "9         egr-4  Wurtzel et al. (2015)  "
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Wurtzel2015.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9acd8994",
   "metadata": {},
   "source": [
    "## 2.2 A mex3 homolog is required for differentiation during planarian stem cell lineage development\n",
    "Zhu, S.J., Hallows, S.E., Currie, K.W., Xu, C. and Pearson, B.J., 2015. A mex3 homolog is required for differentiation during planarian stem cell lineage development. Elife, 4, p.e07025."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d44f1c46",
   "metadata": {},
   "source": [
    "* egr-1 SmedASXL_008042\n",
    "* sox9-1 SmedASXL_004704\n",
    "* egr-4 SmedASXL_019323\n",
    "* mex3-1 SmedASXL_011384\n",
    "* myb-1\tSmedASXL_005044\n",
    "* NF-YA\tSmedASXL_004386\n",
    "* OTX2-1 SmedASXL_013104\n",
    "* runx2 SmedASXL_011806\n",
    "* ZFHX3 SmedASXL_016786"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "e334a0f3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">GCZZ01001800.1 TSA: Schmidtea mediterranea SmedASXL_008042 transcribed RNA sequence\n",
      ">GCZZ01053598.1 TSA: Schmidtea mediterranea SmedASXL_004704 transcribed RNA sequence\n",
      ">GCZZ01068894.1 TSA: Schmidtea mediterranea SmedASXL_019323 transcribed RNA sequence\n",
      ">GCZZ01055034.1 TSA: Schmidtea mediterranea SmedASXL_011384 transcribed RNA sequence\n",
      ">GCZZ01025710.1 TSA: Schmidtea mediterranea SmedASXL_005044 transcribed RNA sequence\n",
      ">GCZZ01000613.1 TSA: Schmidtea mediterranea SmedASXL_004386 transcribed RNA sequence\n",
      ">GCZZ01025082.1 TSA: Schmidtea mediterranea SmedASXL_013104 transcribed RNA sequence\n",
      ">GCZZ01069305.1 TSA: Schmidtea mediterranea SmedASXL_011806 transcribed RNA sequence\n",
      ">GCZZ01054037.1 TSA: Schmidtea mediterranea SmedASXL_016786 transcribed RNA sequence\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "#wget https://sra-download.ncbi.nlm.nih.gov/traces/wgs03/wgs_aux/GC/ZZ/GCZZ01/GCZZ01.1.fsa_nt.gz\n",
    "grep \"SmedASXL_008042\" GCZZ01.1.fsa_nt\n",
    "grep \"SmedASXL_004704\" GCZZ01.1.fsa_nt\n",
    "grep \"SmedASXL_019323\" GCZZ01.1.fsa_nt\n",
    "grep \"SmedASXL_011384\" GCZZ01.1.fsa_nt\n",
    "grep \"SmedASXL_005044\" GCZZ01.1.fsa_nt\n",
    "grep \"SmedASXL_004386\" GCZZ01.1.fsa_nt\n",
    "grep \"SmedASXL_013104\" GCZZ01.1.fsa_nt\n",
    "grep \"SmedASXL_011806\" GCZZ01.1.fsa_nt\n",
    "grep \"SmedASXL_016786\" GCZZ01.1.fsa_nt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ad580b12",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "echo \"GCZZ01001800.1 TSA: Schmidtea mediterranea SmedASXL_008042 transcribed RNA sequence\" > Zhu2015.list\n",
    "echo \"GCZZ01053598.1 TSA: Schmidtea mediterranea SmedASXL_004704 transcribed RNA sequence\" >> Zhu2015.list\n",
    "echo \"GCZZ01068894.1 TSA: Schmidtea mediterranea SmedASXL_019323 transcribed RNA sequence\" >> Zhu2015.list\n",
    "echo \"GCZZ01055034.1 TSA: Schmidtea mediterranea SmedASXL_011384 transcribed RNA sequence\" >> Zhu2015.list\n",
    "echo \"GCZZ01025710.1 TSA: Schmidtea mediterranea SmedASXL_005044 transcribed RNA sequence\" >> Zhu2015.list\n",
    "echo \"GCZZ01000613.1 TSA: Schmidtea mediterranea SmedASXL_004386 transcribed RNA sequence\" >> Zhu2015.list\n",
    "echo \"GCZZ01025082.1 TSA: Schmidtea mediterranea SmedASXL_013104 transcribed RNA sequence\" >> Zhu2015.list\n",
    "echo \"GCZZ01069305.1 TSA: Schmidtea mediterranea SmedASXL_011806 transcribed RNA sequence\" >> Zhu2015.list\n",
    "echo \"GCZZ01054037.1 TSA: Schmidtea mediterranea SmedASXL_016786 transcribed RNA sequence\" >> Zhu2015.list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "4e70dfa5",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "seqtk subseq GCZZ01.1.fsa_nt Zhu2015.list > Zhu2015.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "ec289c3f",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Zhu2015.fa -out blast_Zhu2015 -outfmt 6 &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "f06ccc52",
   "metadata": {},
   "outputs": [],
   "source": [
    "Zhu2015 = pd.DataFrame({\"ID\": [\"GCZZ01001800.1\", \"GCZZ01053598.1\", \"GCZZ01068894.1\", \"GCZZ01055034.1\", \"GCZZ01025710.1\", \n",
    "                                 \"GCZZ01000613.1\", \"GCZZ01025082.1\", \"GCZZ01069305.1\", \"GCZZ01054037.1\"], \n",
    "                           \"Name\": [\"egr-1\", \"sox9-1\", \"egr-4\", \"mex3-1\", \"myb-1\", \"NF-YA\", \"OTX2-1\", \"runx2\", \"ZFHX3\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "0a7d98a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Zhu2015 = pd.read_csv(\"/hydra/TF_data/blast_Zhu2015\", sep=\"\\t\", header=None)\n",
    "blast_Zhu2015.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Zhu2015 = blast_Zhu2015.loc[blast_Zhu2015.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "d8a0f1e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Zhu2015= tf_table(blast_Zhu2015, \"Zhu et al. (2015)\", Zhu2015)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "b478a123",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Zhu2015.to_csv(\"/hydra/TF_data/TF_Zhu2015.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "c1211e2e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SMEST045451001.1</td>\n",
       "      <td>SMESG000045451.1</td>\n",
       "      <td>MSTRG.14130</td>\n",
       "      <td>GCZZ01000613</td>\n",
       "      <td>NF-YA</td>\n",
       "      <td>Zhu et al. (2015)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMEST034285001.1</td>\n",
       "      <td>SMESG000034285.1</td>\n",
       "      <td>MSTRG.11197</td>\n",
       "      <td>GCZZ01001800</td>\n",
       "      <td>egr-1</td>\n",
       "      <td>Zhu et al. (2015)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MSTRG.10065.2</td>\n",
       "      <td>SMESG000031965.1</td>\n",
       "      <td>MSTRG.10065</td>\n",
       "      <td>GCZZ01025082</td>\n",
       "      <td>OTX2-1</td>\n",
       "      <td>Zhu et al. (2015)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SMEST016880001.1</td>\n",
       "      <td>SMESG000016880.1</td>\n",
       "      <td>MSTRG.5224</td>\n",
       "      <td>GCZZ01025710</td>\n",
       "      <td>myb-1</td>\n",
       "      <td>Zhu et al. (2015)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MSTRG.13411.1</td>\n",
       "      <td>SMESG000051170.1</td>\n",
       "      <td>MSTRG.13411</td>\n",
       "      <td>GCZZ01053598</td>\n",
       "      <td>sox9-1</td>\n",
       "      <td>Zhu et al. (2015)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SMEST064905001.1</td>\n",
       "      <td>SMESG000064905.1</td>\n",
       "      <td>MSTRG.17691</td>\n",
       "      <td>GCZZ01054037</td>\n",
       "      <td>ZFHX3</td>\n",
       "      <td>Zhu et al. (2015)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SMEST025422001.1</td>\n",
       "      <td>SMESG000025422.1</td>\n",
       "      <td>MSTRG.385</td>\n",
       "      <td>GCZZ01055034</td>\n",
       "      <td>mex3-1</td>\n",
       "      <td>Zhu et al. (2015)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SMEST000960001.1</td>\n",
       "      <td>SMESG000000960.1</td>\n",
       "      <td>MSTRG.1218</td>\n",
       "      <td>GCZZ01068894</td>\n",
       "      <td>egr-4</td>\n",
       "      <td>Zhu et al. (2015)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SMEST049171002.1</td>\n",
       "      <td>SMESG000049171.1</td>\n",
       "      <td>MSTRG.15139</td>\n",
       "      <td>GCZZ01069305</td>\n",
       "      <td>runx2</td>\n",
       "      <td>Zhu et al. (2015)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Transcript.ID           Rink.ID     Neiro.ID   Original.ID  \\\n",
       "0  SMEST045451001.1  SMESG000045451.1  MSTRG.14130  GCZZ01000613   \n",
       "1  SMEST034285001.1  SMESG000034285.1  MSTRG.11197  GCZZ01001800   \n",
       "2     MSTRG.10065.2  SMESG000031965.1  MSTRG.10065  GCZZ01025082   \n",
       "3  SMEST016880001.1  SMESG000016880.1   MSTRG.5224  GCZZ01025710   \n",
       "4     MSTRG.13411.1  SMESG000051170.1  MSTRG.13411  GCZZ01053598   \n",
       "5  SMEST064905001.1  SMESG000064905.1  MSTRG.17691  GCZZ01054037   \n",
       "6  SMEST025422001.1  SMESG000025422.1    MSTRG.385  GCZZ01055034   \n",
       "7  SMEST000960001.1  SMESG000000960.1   MSTRG.1218  GCZZ01068894   \n",
       "8  SMEST049171002.1  SMESG000049171.1  MSTRG.15139  GCZZ01069305   \n",
       "\n",
       "  Original.name             Source  \n",
       "0         NF-YA  Zhu et al. (2015)  \n",
       "1         egr-1  Zhu et al. (2015)  \n",
       "2        OTX2-1  Zhu et al. (2015)  \n",
       "3         myb-1  Zhu et al. (2015)  \n",
       "4        sox9-1  Zhu et al. (2015)  \n",
       "5         ZFHX3  Zhu et al. (2015)  \n",
       "6        mex3-1  Zhu et al. (2015)  \n",
       "7         egr-4  Zhu et al. (2015)  \n",
       "8         runx2  Zhu et al. (2015)  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Zhu2015.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5d39b09d",
   "metadata": {},
   "source": [
    "## 2.3 Egr-5 is a post-mitotic regulator of planarian epidermal differentiation\n",
    "Tu, K.C., Cheng, L.C., Vu, H.T., Lange, J.J., McKinney, S.A., Seidel, C.W. and Alvarado, A.S., 2015. Egr-5 is a post-mitotic regulator of planarian epidermal differentiation. Elife, 4, p.e10501."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "708460db",
   "metadata": {},
   "source": [
    "# 3. 2016"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f03b8e25",
   "metadata": {},
   "source": [
    "## 3.1 HOX gene complement and expression in the planarian Schmidtea mediterranea\n",
    "Currie, K.W., Brown, D.D., Zhu, S., Xu, C., Voisin, V., Bader, G.D. and Pearson, B.J., 2016. HOX gene complement and expression in the planarian Schmidtea mediterranea. EvoDevo, 7(1), pp.1-11."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1c084977",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Currie2016.xlsx\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "ls Currie2016.xlsx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "14f0aff4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "Currie2016xlsx = pd.read_excel(\"/hydra/TF_data/Currie2016.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "51d453c4",
   "metadata": {},
   "outputs": [],
   "source": [
    "Currie = Currie2016xlsx.iloc[:16,:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bbc825b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "f = open(\"../TF_data/Currie.fa\", \"a\")\n",
    "for i in range(len(Currie)):\n",
    "    f.write(\">\"+Currie.iloc[i,0]+\"\\n\")\n",
    "    f.write(Currie.iloc[i,7]+\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "64e62ff4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Smed-Post-2a\n",
      "MGLWKDNDSTFQDEFPYKSSSINETMMNIFSPSPSSPTKPMIYPYIESTNNISGFQYPYITSQIANLNQMTNTSFSYRGFTFPSNVPPSLYDDSSVNVDQNSWEHSLKPPVHLSYPIDINTIGSLPSYNLGAGVFECNPGREMIQDAMMLARFKMNNPMMEQEYVPEMDHRCRKKRKPYTRYQNLVLESEFNSNSYITRQKRWEISCKLNLSERQIKVWFQNRRMKKKKLQIKKPSTG\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "head -2 Currie.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b7814418",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "tblastn -db ~/FACS/stringtie_transcripts.fa -query Currie.fa -out blast_Currie -outfmt 6 &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3691f422",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Currie = pd.read_csv(\"/hydra/TF_data/blast_Currie\", sep=\"\\t\", header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aa95232d",
   "metadata": {},
   "outputs": [],
   "source": [
    "blast_Currie.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Currie = blast_Currie.loc[blast_Currie.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "487eee6d",
   "metadata": {},
   "outputs": [],
   "source": [
    "transcript_id = []\n",
    "gene_id_rink = []\n",
    "gene_id_neiro = []\n",
    "original_id = []\n",
    "original_name = []\n",
    "for i in range(len(blast_Currie)):\n",
    "    t_id = blast_Currie.iloc[i,1]\n",
    "    o_id = Currie[Currie.iloc[:,0] == blast_Currie.iloc[i,0]].iloc[0,5]\n",
    "    transcript_id.append(t_id)\n",
    "    gene_id_rink.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,0])\n",
    "    gene_id_neiro.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,1])\n",
    "    original_id.append(o_id)\n",
    "    original_name.append(blast_Currie.iloc[i,0])\n",
    "source = [\"Currie (2016)\"] * len(original_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "639b6ca0",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Currie2016 = pd.DataFrame({\"Transcript.ID\": transcript_id, \"Rink.ID\": gene_id_rink, \"Neiro.ID\": gene_id_neiro, \"Original.ID\": original_id, \"Original.name\": original_name, \"Source\": source})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "62b42d32",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Currie2016.to_csv(\"/hydra/TF_data/TF_Currie2016.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7dd2e8a5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "TF_table_Currie2016 = pd.read_csv(\"/hydra/TF_data/TF_Currie2016.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1e5e7b74",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>SMEST045025001.1</td>\n",
       "      <td>SMESG000045025.1</td>\n",
       "      <td>MSTRG.14008</td>\n",
       "      <td>JN983829; SmedASXL_012281</td>\n",
       "      <td>Smed-Dll-1</td>\n",
       "      <td>Currie (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>SMEST025335001.1</td>\n",
       "      <td>SMESG000025335.1</td>\n",
       "      <td>MSTRG.345</td>\n",
       "      <td>SmedASXL_019490</td>\n",
       "      <td>Smed-En</td>\n",
       "      <td>Currie (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>MSTRG.5503.2</td>\n",
       "      <td>SMESG000015136.1</td>\n",
       "      <td>MSTRG.5503</td>\n",
       "      <td>SmedASXL_008181</td>\n",
       "      <td>Smed-Eve</td>\n",
       "      <td>Currie (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>SMEST080661001.1</td>\n",
       "      <td>SMESG000080661.1</td>\n",
       "      <td>MSTRG.23578</td>\n",
       "      <td>SmedASXL_003878</td>\n",
       "      <td>Smed-Hox1</td>\n",
       "      <td>Currie (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>SMEST029666001.1</td>\n",
       "      <td>SMESG000029666.1</td>\n",
       "      <td>MSTRG.9913</td>\n",
       "      <td>SmedASXL_008435</td>\n",
       "      <td>Smed-Hox2/3a</td>\n",
       "      <td>Currie (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>MSTRG.442.2</td>\n",
       "      <td>SMESG000025656.1</td>\n",
       "      <td>MSTRG.442</td>\n",
       "      <td>SmedASXL_007522</td>\n",
       "      <td>Smed-Hox3b</td>\n",
       "      <td>Currie (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>SMEST071019001.1</td>\n",
       "      <td>SMESG000071019.1</td>\n",
       "      <td>MSTRG.20904</td>\n",
       "      <td>SmedASXL_007160</td>\n",
       "      <td>Smed-Hox4a</td>\n",
       "      <td>Currie (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>7</td>\n",
       "      <td>SMEST044469002.1</td>\n",
       "      <td>SMESG000044469.1</td>\n",
       "      <td>MSTRG.13843</td>\n",
       "      <td>SmedASXL_008878</td>\n",
       "      <td>Smed-Hox4b</td>\n",
       "      <td>Currie (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>8</td>\n",
       "      <td>SMEST022667001.1</td>\n",
       "      <td>SMESG000022667.1</td>\n",
       "      <td>MSTRG.7998</td>\n",
       "      <td>SmedASXL_008764/SmedASXL_066865</td>\n",
       "      <td>Smed-Lox5a</td>\n",
       "      <td>Currie (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>9</td>\n",
       "      <td>SMEST003389001.1</td>\n",
       "      <td>SMESG000003389.1</td>\n",
       "      <td>MSTRG.934</td>\n",
       "      <td>smed013934 (transcriptome from Labbe, et al., ...</td>\n",
       "      <td>Smed-Post-1</td>\n",
       "      <td>Currie (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>10</td>\n",
       "      <td>SMEST054107001.1</td>\n",
       "      <td>SMESG000054107.1</td>\n",
       "      <td>MSTRG.16300</td>\n",
       "      <td>SmedASXL_008903</td>\n",
       "      <td>Smed-Post-2a</td>\n",
       "      <td>Currie (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>11</td>\n",
       "      <td>SMEST033048001.1</td>\n",
       "      <td>SMESG000033048.1</td>\n",
       "      <td>MSTRG.10796</td>\n",
       "      <td>SmedASXL_008593</td>\n",
       "      <td>Smed-Post-2b</td>\n",
       "      <td>Currie (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>12</td>\n",
       "      <td>MSTRG.7999.4</td>\n",
       "      <td>SMESG000022668.1</td>\n",
       "      <td>MSTRG.7999</td>\n",
       "      <td>SmedASXL_007760</td>\n",
       "      <td>Smed-Post-2c</td>\n",
       "      <td>Currie (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>13</td>\n",
       "      <td>MSTRG.9883.1</td>\n",
       "      <td>-</td>\n",
       "      <td>MSTRG.9883</td>\n",
       "      <td>SmedASXL_012639</td>\n",
       "      <td>Smed-Post-2d</td>\n",
       "      <td>Currie (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>14</td>\n",
       "      <td>SMEST043155001.1</td>\n",
       "      <td>SMESG000043155.1</td>\n",
       "      <td>MSTRG.13724</td>\n",
       "      <td>SmedASXL_005553/SmedASXL_043981</td>\n",
       "      <td>Smed-lox4</td>\n",
       "      <td>Currie (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>15</td>\n",
       "      <td>SMEST079807002.1</td>\n",
       "      <td>SMESG000079807.1</td>\n",
       "      <td>MSTRG.23420</td>\n",
       "      <td>SmedASXL_002885</td>\n",
       "      <td>Smed-lox5b</td>\n",
       "      <td>Currie (2016)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0     Transcript.ID           Rink.ID     Neiro.ID  \\\n",
       "0            0  SMEST045025001.1  SMESG000045025.1  MSTRG.14008   \n",
       "1            1  SMEST025335001.1  SMESG000025335.1    MSTRG.345   \n",
       "2            2      MSTRG.5503.2  SMESG000015136.1   MSTRG.5503   \n",
       "3            3  SMEST080661001.1  SMESG000080661.1  MSTRG.23578   \n",
       "4            4  SMEST029666001.1  SMESG000029666.1   MSTRG.9913   \n",
       "5            5       MSTRG.442.2  SMESG000025656.1    MSTRG.442   \n",
       "6            6  SMEST071019001.1  SMESG000071019.1  MSTRG.20904   \n",
       "7            7  SMEST044469002.1  SMESG000044469.1  MSTRG.13843   \n",
       "8            8  SMEST022667001.1  SMESG000022667.1   MSTRG.7998   \n",
       "9            9  SMEST003389001.1  SMESG000003389.1    MSTRG.934   \n",
       "10          10  SMEST054107001.1  SMESG000054107.1  MSTRG.16300   \n",
       "11          11  SMEST033048001.1  SMESG000033048.1  MSTRG.10796   \n",
       "12          12      MSTRG.7999.4  SMESG000022668.1   MSTRG.7999   \n",
       "13          13      MSTRG.9883.1                 -   MSTRG.9883   \n",
       "14          14  SMEST043155001.1  SMESG000043155.1  MSTRG.13724   \n",
       "15          15  SMEST079807002.1  SMESG000079807.1  MSTRG.23420   \n",
       "\n",
       "                                          Original.ID Original.name  \\\n",
       "0                           JN983829; SmedASXL_012281    Smed-Dll-1   \n",
       "1                                     SmedASXL_019490       Smed-En   \n",
       "2                                     SmedASXL_008181      Smed-Eve   \n",
       "3                                     SmedASXL_003878     Smed-Hox1   \n",
       "4                                     SmedASXL_008435  Smed-Hox2/3a   \n",
       "5                                     SmedASXL_007522    Smed-Hox3b   \n",
       "6                                     SmedASXL_007160    Smed-Hox4a   \n",
       "7                                     SmedASXL_008878    Smed-Hox4b   \n",
       "8                     SmedASXL_008764/SmedASXL_066865    Smed-Lox5a   \n",
       "9   smed013934 (transcriptome from Labbe, et al., ...   Smed-Post-1   \n",
       "10                                    SmedASXL_008903  Smed-Post-2a   \n",
       "11                                    SmedASXL_008593  Smed-Post-2b   \n",
       "12                                    SmedASXL_007760  Smed-Post-2c   \n",
       "13                                    SmedASXL_012639  Smed-Post-2d   \n",
       "14                    SmedASXL_005553/SmedASXL_043981     Smed-lox4   \n",
       "15                                    SmedASXL_002885    Smed-lox5b   \n",
       "\n",
       "           Source  \n",
       "0   Currie (2016)  \n",
       "1   Currie (2016)  \n",
       "2   Currie (2016)  \n",
       "3   Currie (2016)  \n",
       "4   Currie (2016)  \n",
       "5   Currie (2016)  \n",
       "6   Currie (2016)  \n",
       "7   Currie (2016)  \n",
       "8   Currie (2016)  \n",
       "9   Currie (2016)  \n",
       "10  Currie (2016)  \n",
       "11  Currie (2016)  \n",
       "12  Currie (2016)  \n",
       "13  Currie (2016)  \n",
       "14  Currie (2016)  \n",
       "15  Currie (2016)  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TF_table_Currie2016"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5f676200",
   "metadata": {},
   "source": [
    "## 3.2 Forkhead containing transcription factor Albino controls tetrapyrrole-based body pigmentation in planarian\n",
    "Wang, C., Han, X.S., Li, F.F., Huang, S., Qin, Y.W., Zhao, X.X. and Jing, Q., 2016. Forkhead containing transcription factor Albino controls tetrapyrrole-based body pigmentation in planarian. Cell discovery, 2(1), pp.1-17."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ca4f2a42",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "ls Wang2016.xls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "32572700",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "Wang2016 = pd.read_excel(\"/hydra/TF_data/Wang2016.xls\", sheet_name=2)\n",
    "Wang2016 = Wang2016.iloc[0:3,:] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "234d6318",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>Has homologue</th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Albino</td>\n",
       "      <td>FoxP</td>\n",
       "      <td>AAAATGCATTTAATGTTTGCAGTTTGATTCGCGGTTGTGGAATATT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>tk_fk_062</td>\n",
       "      <td>FoxF2</td>\n",
       "      <td>TTTAAATGAATGTTTTATCAAACTTCCCAAAGGTATGGGCAGACCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>tk_fk_061</td>\n",
       "      <td>FoxF1</td>\n",
       "      <td>AATTGCCAACCACAATTCAACTAAAGACGATCATCGATAGTTTAAA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene  Has homologue                                           sequence\n",
       "0     Albino          FoxP  AAAATGCATTTAATGTTTGCAGTTTGATTCGCGGTTGTGGAATATT...\n",
       "1  tk_fk_062         FoxF2  TTTAAATGAATGTTTTATCAAACTTCCCAAAGGTATGGGCAGACCT...\n",
       "2  tk_fk_061         FoxF1  AATTGCCAACCACAATTCAACTAAAGACGATCATCGATAGTTTAAA..."
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Wang2016"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6bb1b496",
   "metadata": {},
   "outputs": [],
   "source": [
    "f = open(\"/hydra/TF_data/Wang2016.fa\", \"a\")\n",
    "for i in range(len(Wang2016)):\n",
    "    f.write(\">\"+Wang2016.iloc[i,0]+\"\\n\")\n",
    "    f.write(Wang2016.iloc[i,2]+\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1698c894",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Wang2016.fa -out blast_Wang2016 -outfmt 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7658c4d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Wang2016 = pd.read_csv(\"/hydra/TF_data/blast_Wang2016\", sep=\"\\t\", header=None)\n",
    "blast_Wang2016.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Wang2016 = blast_Wang2016.loc[blast_Wang2016.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1589e819",
   "metadata": {},
   "outputs": [],
   "source": [
    "transcript_id = []\n",
    "gene_id_rink = []\n",
    "gene_id_neiro = []\n",
    "original_id = []\n",
    "original_name = []\n",
    "for i in range(len(blast_Wang2016)):\n",
    "    t_id = blast_Wang2016.iloc[i,1]\n",
    "    o_id = \"NA\"\n",
    "    transcript_id.append(t_id)\n",
    "    gene_id_rink.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,0])\n",
    "    gene_id_neiro.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,1])\n",
    "    original_id.append(o_id)\n",
    "    original_name.append(blast_Wang2016.iloc[i,0])\n",
    "source = [\"Wang et al. (2016)\"] * len(original_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21c4bc5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Wang2016 = pd.DataFrame({\"Transcript.ID\": transcript_id, \"Rink.ID\": gene_id_rink, \"Neiro.ID\": gene_id_neiro, \"Original.ID\": original_id, \"Original.name\": original_name, \"Source\": source})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b8675fee",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Wang2016.to_csv(\"/hydra/TF_data/TF_Wang2016.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "bfaf3ba7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MSTRG.20250.4</td>\n",
       "      <td>SMESG000068148.1</td>\n",
       "      <td>MSTRG.20250</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Albino</td>\n",
       "      <td>Wang et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MSTRG.19765.2</td>\n",
       "      <td>SMESG000066497.1</td>\n",
       "      <td>MSTRG.19765</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tk_fk_061</td>\n",
       "      <td>Wang et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MSTRG.19765.4</td>\n",
       "      <td>SMESG000066497.1</td>\n",
       "      <td>MSTRG.19765</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tk_fk_062</td>\n",
       "      <td>Wang et al. (2016)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Transcript.ID           Rink.ID     Neiro.ID  Original.ID Original.name  \\\n",
       "0  MSTRG.20250.4  SMESG000068148.1  MSTRG.20250          NaN        Albino   \n",
       "1  MSTRG.19765.2  SMESG000066497.1  MSTRG.19765          NaN     tk_fk_061   \n",
       "2  MSTRG.19765.4  SMESG000066497.1  MSTRG.19765          NaN     tk_fk_062   \n",
       "\n",
       "               Source  \n",
       "0  Wang et al. (2016)  \n",
       "1  Wang et al. (2016)  \n",
       "2  Wang et al. (2016)  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Wang2016.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "46ba3912",
   "metadata": {},
   "source": [
    "FoxF2 and FoxF1 are the same"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "02540e0c",
   "metadata": {},
   "source": [
    "## 3.3 A functional genomics screen in planarians reveals regulators of whole-brain regeneration\n",
    "Roberts-Galbraith, R.H., Brubacher, J.L. and Newmark, P.A., 2016. A functional genomics screen in planarians reveals regulators of whole-brain regeneration. Elife, 5, p.e17002.\n",
    "\n",
    "We capitalized on this distinctive trait and completed a gene expression-guided functional screen to identify factors that regulate diverse aspects of neural regeneration in Schmidtea mediterranea"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d8bbb97",
   "metadata": {},
   "source": [
    "* abd-Ba dd_Smed_v6_12467_0_1 \n",
    "* arrowhead dd_Smed_v6_47123_0_1\n",
    "* FLI-1 dd_Smed_v6_14611_0_1\n",
    "* FLI-2 dd_Smed_v6_37276_0_1\n",
    "* foxA dd_Smed_v6_10718_0_4\n",
    "* hesl-3 dd_Smed_v6_22479_0_1\n",
    "* hlh-1 dd_Smed_v6_31217_0_1\n",
    "* hoxD\tdd_Smed_v6_2442_0_1\n",
    "* lhx1\tdd_Smed_v6_8252_0_1\n",
    "* mox\tdd_Smed_v6_12317_0_1\n",
    "* otxA\tdd_Smed_v6_14633_0_1\n",
    "* POU2/3-2\tdd_Smed_v6_38342_0_1\n",
    "* runt dd_Smed_v6_16222_0_1\n",
    "* soxB2-2 dd_Smed_v6_25321_0_1\n",
    "* soxP-1 dd_Smed_v6_7227_0_1 \n",
    "* TF7-like\tdd_Smed_v6_34144_0_1\n",
    "* TFAP-2b dd_Smed_v6_15104_0_1\n",
    "* zfp-4\tdd_Smed_v6_14413_0_1\n",
    "* TRIP13 dd_Smed_v6_46627_0_1\n",
    "* zmym-2 dd_Smed_v6_13305_1_1\n",
    "* zmym-3 dd_Smed_v6_13305_1_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "08901798",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "echo \"dd_Smed_v6_12467_0_1\" > RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_47123_0_1\" >> RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_14611_0_1\" >> RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_37276_0_1\" >> RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_10718_0_4\" >> RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_22479_0_1\" >> RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_31217_0_1\" >> RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_2442_0_1\" >> RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_8252_0_1\" >> RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_12317_0_1\" >> RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_14633_0_1\" >> RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_38342_0_1\" >> RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_16222_0_1\" >> RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_25321_0_1\" >> RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_7227_0_1\" >> RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_34144_0_1\" >> RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_15104_0_1\" >> RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_14413_0_1\" >> RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_46627_0_1\" >> RobertsGalbraith2016.list\n",
    "echo \"dd_Smed_v6_13305_1_1\" >> RobertsGalbraith2016.list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6f9aa789",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "seqtk subseq /hydra/sexual_genome_annotation_files/sexual_genome_annotation/dd_Smed_v6.pcf.contigs.fasta RobertsGalbraith2016.list > RobertsGalbraith2016.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "62b062f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query RobertsGalbraith2016.fa -out blast_RobertsGalbraith2016 -outfmt 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "112b3a8a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "RobertsGalbraith2016 = pd.DataFrame({\"ID\": [\"dd_Smed_v6_12467_0_1\", \"dd_Smed_v6_47123_0_1\", \"dd_Smed_v6_14611_0_1\", \"dd_Smed_v6_37276_0_1\", \n",
    "                                   \"dd_Smed_v6_10718_0_4\", \"dd_Smed_v6_22479_0_1\", \"dd_Smed_v6_31217_0_1\", \"dd_Smed_v6_2442_0_1\", \n",
    "                                   \"dd_Smed_v6_8252_0_1\", \"dd_Smed_v6_12317_0_1\", \"dd_Smed_v6_14633_0_1\", \"dd_Smed_v6_38342_0_1\",\n",
    "                                  \"dd_Smed_v6_16222_0_1\", \"dd_Smed_v6_25321_0_1\", \"dd_Smed_v6_7227_0_1\", \"dd_Smed_v6_34144_0_1\",\n",
    "                                           \"dd_Smed_v6_15104_0_1\", \"dd_Smed_v6_14413_0_1\", \"dd_Smed_v6_46627_0_1\", \"dd_Smed_v6_13305_1_1\"], \n",
    "                           \"Name\": [\"abdBa\", \"arhw\", \"fli-1\", \"fli-2\", \"foxA\", \"hesl-3\", \"hlh-1\", \"hoxD\", \"lhx1\", \"mox\", \"otxA\", \"POU2/3-2\",\n",
    "                                   \"runt\", \"soxB2-2\", \"soxP-1\", \"TF7-like\", \"TFAP-2b\", \"zfp-4\", \"TRIP13\", \"zmym-2\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "f5c4c709",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_RobertsGalbraith2016 = pd.read_csv(\"/hydra/TF_data/blast_RobertsGalbraith2016\", sep=\"\\t\", header=None)\n",
    "blast_RobertsGalbraith2016.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_RobertsGalbraith2016 = blast_RobertsGalbraith2016.loc[blast_RobertsGalbraith2016.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "4041753d",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_RobertsGalbraith2016 = tf_table(blast_RobertsGalbraith2016, \"Roberts-Galbraith et al. (2016)\", RobertsGalbraith2016)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "06c870b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_RobertsGalbraith2016.to_csv(\"/hydra/TF_data/TF_RobertsGalbraith2016.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "ae94ad3a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MSTRG.19539.1</td>\n",
       "      <td>SMESG000065669.1</td>\n",
       "      <td>MSTRG.19539</td>\n",
       "      <td>NaN</td>\n",
       "      <td>foxA</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMEST062714001.1</td>\n",
       "      <td>SMESG000062714.1</td>\n",
       "      <td>MSTRG.18714</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mox</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SMEST022668001.1</td>\n",
       "      <td>SMESG000022668.1</td>\n",
       "      <td>MSTRG.7999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>abdBa</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SMEST043416002.1</td>\n",
       "      <td>SMESG000043416.1</td>\n",
       "      <td>SMESG000043416.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>zmym-2</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SMEST020158002.1</td>\n",
       "      <td>SMESG000020158.1</td>\n",
       "      <td>MSTRG.7049</td>\n",
       "      <td>NaN</td>\n",
       "      <td>zfp-4</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SMEST080027001.1</td>\n",
       "      <td>SMESG000080027.1</td>\n",
       "      <td>MSTRG.23472</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fli-1</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MSTRG.10065.1</td>\n",
       "      <td>SMESG000031965.1</td>\n",
       "      <td>MSTRG.10065</td>\n",
       "      <td>NaN</td>\n",
       "      <td>otxA</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SMEST022356001.1</td>\n",
       "      <td>SMESG000022356.1</td>\n",
       "      <td>MSTRG.7890</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TFAP-2b</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SMEST014779001.1</td>\n",
       "      <td>SMESG000014779.1</td>\n",
       "      <td>MSTRG.4469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>runt</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SMEST009969001.1</td>\n",
       "      <td>SMESG000009969.1</td>\n",
       "      <td>MSTRG.3741</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hesl-3</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MSTRG.13843.2</td>\n",
       "      <td>SMESG000044469.1</td>\n",
       "      <td>MSTRG.13843</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hoxD</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SMEST077970001.1</td>\n",
       "      <td>SMESG000077970.1</td>\n",
       "      <td>MSTRG.21711</td>\n",
       "      <td>NaN</td>\n",
       "      <td>soxB2-2</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MSTRG.20970.1</td>\n",
       "      <td>-</td>\n",
       "      <td>MSTRG.20970</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hlh-1</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MSTRG.15807.2</td>\n",
       "      <td>SMESG000059443.1</td>\n",
       "      <td>MSTRG.15807</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TF7-like</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SMEST014306002.1</td>\n",
       "      <td>SMESG000014306.1</td>\n",
       "      <td>MSTRG.4327</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fli-2</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SMEST008858001.1</td>\n",
       "      <td>SMESG000008858.1</td>\n",
       "      <td>MSTRG.2646</td>\n",
       "      <td>NaN</td>\n",
       "      <td>POU2/3-2</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SMEST040170001.1</td>\n",
       "      <td>SMESG000040170.1</td>\n",
       "      <td>MSTRG.12611</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TRIP13</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SMEST081952001.1</td>\n",
       "      <td>SMESG000081952.1</td>\n",
       "      <td>MSTRG.22894</td>\n",
       "      <td>NaN</td>\n",
       "      <td>arhw</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MSTRG.3856.1</td>\n",
       "      <td>SMESG000010063.1</td>\n",
       "      <td>MSTRG.3856</td>\n",
       "      <td>NaN</td>\n",
       "      <td>soxP-1</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SMEST022856001.1</td>\n",
       "      <td>SMESG000022856.1</td>\n",
       "      <td>MSTRG.7427</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lhx1</td>\n",
       "      <td>Roberts-Galbraith et al. (2016)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Transcript.ID           Rink.ID          Neiro.ID  Original.ID  \\\n",
       "0      MSTRG.19539.1  SMESG000065669.1       MSTRG.19539          NaN   \n",
       "1   SMEST062714001.1  SMESG000062714.1       MSTRG.18714          NaN   \n",
       "2   SMEST022668001.1  SMESG000022668.1        MSTRG.7999          NaN   \n",
       "3   SMEST043416002.1  SMESG000043416.1  SMESG000043416.1          NaN   \n",
       "4   SMEST020158002.1  SMESG000020158.1        MSTRG.7049          NaN   \n",
       "5   SMEST080027001.1  SMESG000080027.1       MSTRG.23472          NaN   \n",
       "6      MSTRG.10065.1  SMESG000031965.1       MSTRG.10065          NaN   \n",
       "7   SMEST022356001.1  SMESG000022356.1        MSTRG.7890          NaN   \n",
       "8   SMEST014779001.1  SMESG000014779.1        MSTRG.4469          NaN   \n",
       "9   SMEST009969001.1  SMESG000009969.1        MSTRG.3741          NaN   \n",
       "10     MSTRG.13843.2  SMESG000044469.1       MSTRG.13843          NaN   \n",
       "11  SMEST077970001.1  SMESG000077970.1       MSTRG.21711          NaN   \n",
       "12     MSTRG.20970.1                 -       MSTRG.20970          NaN   \n",
       "13     MSTRG.15807.2  SMESG000059443.1       MSTRG.15807          NaN   \n",
       "14  SMEST014306002.1  SMESG000014306.1        MSTRG.4327          NaN   \n",
       "15  SMEST008858001.1  SMESG000008858.1        MSTRG.2646          NaN   \n",
       "16  SMEST040170001.1  SMESG000040170.1       MSTRG.12611          NaN   \n",
       "17  SMEST081952001.1  SMESG000081952.1       MSTRG.22894          NaN   \n",
       "18      MSTRG.3856.1  SMESG000010063.1        MSTRG.3856          NaN   \n",
       "19  SMEST022856001.1  SMESG000022856.1        MSTRG.7427          NaN   \n",
       "\n",
       "   Original.name                           Source  \n",
       "0           foxA  Roberts-Galbraith et al. (2016)  \n",
       "1            mox  Roberts-Galbraith et al. (2016)  \n",
       "2          abdBa  Roberts-Galbraith et al. (2016)  \n",
       "3         zmym-2  Roberts-Galbraith et al. (2016)  \n",
       "4          zfp-4  Roberts-Galbraith et al. (2016)  \n",
       "5          fli-1  Roberts-Galbraith et al. (2016)  \n",
       "6           otxA  Roberts-Galbraith et al. (2016)  \n",
       "7        TFAP-2b  Roberts-Galbraith et al. (2016)  \n",
       "8           runt  Roberts-Galbraith et al. (2016)  \n",
       "9         hesl-3  Roberts-Galbraith et al. (2016)  \n",
       "10          hoxD  Roberts-Galbraith et al. (2016)  \n",
       "11       soxB2-2  Roberts-Galbraith et al. (2016)  \n",
       "12         hlh-1  Roberts-Galbraith et al. (2016)  \n",
       "13      TF7-like  Roberts-Galbraith et al. (2016)  \n",
       "14         fli-2  Roberts-Galbraith et al. (2016)  \n",
       "15      POU2/3-2  Roberts-Galbraith et al. (2016)  \n",
       "16        TRIP13  Roberts-Galbraith et al. (2016)  \n",
       "17          arhw  Roberts-Galbraith et al. (2016)  \n",
       "18        soxP-1  Roberts-Galbraith et al. (2016)  \n",
       "19          lhx1  Roberts-Galbraith et al. (2016)  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_RobertsGalbraith2016.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "72b1d21a",
   "metadata": {},
   "source": [
    "## 3.4 Essential role for the planarian intestinal GATA transcription factor in stem cells and regeneration\n",
    "\n",
    "### Materials and methods\n",
    "Smed-gata4/5/6: JF802198"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ec1ef09f",
   "metadata": {},
   "outputs": [],
   "source": [
    "Flores2016 = pd.DataFrame({\"ID\": [\"JF802198.1\"], \"Name\": [\"gata456\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "7524bfe2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">JF802198.1 Schmidtea mediterranea strain CIW4 GATA456 (gata456) mRNA, complete cds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "grep \">\" Flores2016.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d633037e",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db ~/FACS/stringtie_transcripts.fa -query Flores2016.fa -out blast_Flores2016 -outfmt 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "926d3351",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Flores2016 = pd.read_csv(\"/hydra/TF_data/blast_Flores2016\", sep=\"\\t\", header=None)\n",
    "blast_Flores2016.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Flores2016 = blast_Flores2016.loc[blast_Flores2016.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "626a27c6",
   "metadata": {},
   "outputs": [],
   "source": [
    "transcript_id = []\n",
    "gene_id_rink = []\n",
    "gene_id_neiro = []\n",
    "original_id = []\n",
    "original_name = []\n",
    "for i in range(len(blast_Flores2016)):\n",
    "    t_id = blast_Flores2016.iloc[i,1]\n",
    "    o_id = blast_Flores2016.iloc[i,0]\n",
    "    transcript_id.append(t_id)\n",
    "    gene_id_rink.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,0])\n",
    "    gene_id_neiro.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,1])\n",
    "    original_id.append(o_id.rpartition(\".\")[0])\n",
    "    original_name.append(Flores2016[Flores2016[\"ID\"] == o_id].iloc[0,1])\n",
    "source = [\"Flores et al. (2016)\"] * len(original_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eb3e40c6",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Flores2016 = pd.DataFrame({\"Transcript.ID\": transcript_id, \"Rink.ID\": gene_id_rink, \"Neiro.ID\": gene_id_neiro, \"Original.ID\": original_id, \"Original.name\": original_name, \"Source\": source})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3c2da6b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Flores2016.to_csv(\"../TF_data/TF_Flores2016.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "a036b483",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MSTRG.12700.1</td>\n",
       "      <td>SMESG000039559.1</td>\n",
       "      <td>MSTRG.12700</td>\n",
       "      <td>JF802198</td>\n",
       "      <td>gata456</td>\n",
       "      <td>Flores et al. (2016)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Transcript.ID           Rink.ID     Neiro.ID Original.ID Original.name  \\\n",
       "0  MSTRG.12700.1  SMESG000039559.1  MSTRG.12700    JF802198       gata456   \n",
       "\n",
       "                 Source  \n",
       "0  Flores et al. (2016)  "
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Flores2016.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ce4511a3",
   "metadata": {},
   "source": [
    "## 3.5 NF-YB Regulates Spermatogonial Stem Cell Self-Renewal and Proliferation in the Planarian Schmidtea mediterranea\n",
    "Iyer, H., Collins III, J.J. and Newmark, P.A., 2016. NF-YB regulates spermatogonial stem cell self-renewal and proliferation in the planarian Schmidtea mediterranea. PLoS genetics, 12(6), p.e1006109."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "0ac6bb22",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">NF-YB\n",
      ">NF-YB2\n",
      ">NF-YA1\n",
      ">NF-YA2\n",
      ">NF-YC\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "grep \">\" Iyer2016.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef3d1b76",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Iyer2016.fa -out blast_Iyer2016 -outfmt 6 -evalue 1.0 -word_size 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "765536fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Iyer2016 = pd.read_csv(\"/hydra/TF_data/blast_Iyer2016\", sep=\"\\t\", header=None)\n",
    "blast_Iyer2016.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Iyer2016 = blast_Iyer2016.loc[blast_Iyer2016.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "bbaa6fa4",
   "metadata": {},
   "outputs": [],
   "source": [
    "transcript_id = []\n",
    "gene_id_rink = []\n",
    "gene_id_neiro = []\n",
    "original_id = []\n",
    "original_name = []\n",
    "for i in range(len(blast_Iyer2016)):\n",
    "    t_id = blast_Iyer2016.iloc[i,1]\n",
    "    o_id = \"NA\"\n",
    "    transcript_id.append(t_id)\n",
    "    gene_id_rink.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,0])\n",
    "    gene_id_neiro.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,1])\n",
    "    original_id.append(o_id)\n",
    "    original_name.append(blast_Iyer2016.iloc[i,0])\n",
    "source = [\"Iyer et al. (2016)\"] * len(original_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "7848bfeb",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Iyer2016 = pd.DataFrame({\"Transcript.ID\": transcript_id, \"Rink.ID\": gene_id_rink, \"Neiro.ID\": gene_id_neiro, \"Original.ID\": original_id, \"Original.name\": original_name, \"Source\": source})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "08830f04",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Iyer2016.to_csv(\"/hydra/TF_data/TF_Iyer2016.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "eb3c044d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SMEST045451001.1</td>\n",
       "      <td>SMESG000045451.1</td>\n",
       "      <td>MSTRG.14130</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NF-YA1</td>\n",
       "      <td>Iyer et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MSTRG.3824.5</td>\n",
       "      <td>SMESG000010136.1</td>\n",
       "      <td>MSTRG.3824</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NF-YA2</td>\n",
       "      <td>Iyer et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MSTRG.5449.6</td>\n",
       "      <td>SMESG000017262.1</td>\n",
       "      <td>MSTRG.5449</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NF-YB</td>\n",
       "      <td>Iyer et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SMEST040141001.1</td>\n",
       "      <td>SMESG000040141.1</td>\n",
       "      <td>MSTRG.12597</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NF-YB2</td>\n",
       "      <td>Iyer et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SMEST077821002.1</td>\n",
       "      <td>SMESG000077821.1</td>\n",
       "      <td>MSTRG.21651</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NF-YC</td>\n",
       "      <td>Iyer et al. (2016)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Transcript.ID           Rink.ID     Neiro.ID  Original.ID Original.name  \\\n",
       "0  SMEST045451001.1  SMESG000045451.1  MSTRG.14130          NaN        NF-YA1   \n",
       "1      MSTRG.3824.5  SMESG000010136.1   MSTRG.3824          NaN        NF-YA2   \n",
       "2      MSTRG.5449.6  SMESG000017262.1   MSTRG.5449          NaN         NF-YB   \n",
       "3  SMEST040141001.1  SMESG000040141.1  MSTRG.12597          NaN        NF-YB2   \n",
       "4  SMEST077821002.1  SMESG000077821.1  MSTRG.21651          NaN         NF-YC   \n",
       "\n",
       "               Source  \n",
       "0  Iyer et al. (2016)  \n",
       "1  Iyer et al. (2016)  \n",
       "2  Iyer et al. (2016)  \n",
       "3  Iyer et al. (2016)  \n",
       "4  Iyer et al. (2016)  "
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Iyer2016.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a9b17e37",
   "metadata": {},
   "source": [
    "## 3.6 In silico lineage tracing through single cell transcriptomics identifies a neural stem cell population in planarians\n",
    "Molinaro, A.M. and Pearson, B.J., 2016. In silico lineage tracing through single cell transcriptomics identifies a neural stem cell population in planarians. Genome biology, 17(1), pp.1-17."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "29ee92ae",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">cux-1\n",
      ">ski-1\n",
      ">ski-3\n",
      ">smad-4\n",
      ">alx-3\n",
      ">znf-91\n",
      ">Smed-Ybp2-1\n",
      ">Smed-Ybp2-2\n",
      ">Smed-yps-1\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "grep \">\" Molinaro2016.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6cdc9773",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Molinaro2016.fa -out blast_Molinaro2016 -outfmt 6 &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "9e86f7e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "Molinaro2016 = pd.DataFrame({\"ID\": [\"cux-1\", \"ski-1\", \"ski-3\", \"smad-4\", \n",
    "                                   \"alx-3\", \"znf-91\", \"Smed-Ybp2-1\", \"Smed-Ybp2-2\", \n",
    "                                   \"Smed-yps-1\"], \n",
    "                           \"Name\": [\"cux-1\", \"ski-1\", \"ski-3\", \"smad-4\", \n",
    "                                   \"alx-3\", \"znf-91\", \"Smed-Ybp2-1\", \"Smed-Ybp2-2\", \n",
    "                                   \"Smed-yps-1\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "0671b63f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Molinaro2016 = pd.read_csv(\"/hydra/TF_data/blast_Molinaro2016\", sep=\"\\t\", header=None)\n",
    "blast_Molinaro2016.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Molinaro2016 = blast_Molinaro2016.loc[blast_Molinaro2016.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "62aeef31",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Molinaro2016= tf_table(blast_Molinaro2016, \"Molinaro et al. (2016)\", Molinaro2016)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "99a6f4b6",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Molinaro2016.to_csv(\"/hydra/TF_data/TF_Molinaro2016.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ac20ac32",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SMEST058066001.1</td>\n",
       "      <td>SMESG000058066.1</td>\n",
       "      <td>MSTRG.15848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smed-Ybp2-1</td>\n",
       "      <td>Molinaro et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMEST058043001.1</td>\n",
       "      <td>SMESG000058043.1</td>\n",
       "      <td>MSTRG.15846</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smed-Ybp2-2</td>\n",
       "      <td>Molinaro et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MSTRG.10123.5</td>\n",
       "      <td>SMESG000032180.1</td>\n",
       "      <td>MSTRG.10123</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smed-yps-1</td>\n",
       "      <td>Molinaro et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MSTRG.5100.2</td>\n",
       "      <td>SMESG000016670.1</td>\n",
       "      <td>MSTRG.5100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alx-3</td>\n",
       "      <td>Molinaro et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SMEST065026002.1</td>\n",
       "      <td>SMESG000065026.1</td>\n",
       "      <td>MSTRG.17728</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cux-1</td>\n",
       "      <td>Molinaro et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SMEST067290001.1</td>\n",
       "      <td>SMESG000067290.1</td>\n",
       "      <td>MSTRG.19902</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ski-1</td>\n",
       "      <td>Molinaro et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SMEST044463001.1</td>\n",
       "      <td>SMESG000044463.1</td>\n",
       "      <td>MSTRG.13841</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ski-3</td>\n",
       "      <td>Molinaro et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MSTRG.9898.2</td>\n",
       "      <td>SMESG000029614.1</td>\n",
       "      <td>MSTRG.9898</td>\n",
       "      <td>NaN</td>\n",
       "      <td>smad-4</td>\n",
       "      <td>Molinaro et al. (2016)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SMEST026243001.1</td>\n",
       "      <td>SMESG000026243.1</td>\n",
       "      <td>MSTRG.624</td>\n",
       "      <td>NaN</td>\n",
       "      <td>znf-91</td>\n",
       "      <td>Molinaro et al. (2016)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Transcript.ID           Rink.ID     Neiro.ID  Original.ID Original.name  \\\n",
       "0  SMEST058066001.1  SMESG000058066.1  MSTRG.15848          NaN   Smed-Ybp2-1   \n",
       "1  SMEST058043001.1  SMESG000058043.1  MSTRG.15846          NaN   Smed-Ybp2-2   \n",
       "2     MSTRG.10123.5  SMESG000032180.1  MSTRG.10123          NaN    Smed-yps-1   \n",
       "3      MSTRG.5100.2  SMESG000016670.1   MSTRG.5100          NaN         alx-3   \n",
       "4  SMEST065026002.1  SMESG000065026.1  MSTRG.17728          NaN         cux-1   \n",
       "5  SMEST067290001.1  SMESG000067290.1  MSTRG.19902          NaN         ski-1   \n",
       "6  SMEST044463001.1  SMESG000044463.1  MSTRG.13841          NaN         ski-3   \n",
       "7      MSTRG.9898.2  SMESG000029614.1   MSTRG.9898          NaN        smad-4   \n",
       "8  SMEST026243001.1  SMESG000026243.1    MSTRG.624          NaN        znf-91   \n",
       "\n",
       "                   Source  \n",
       "0  Molinaro et al. (2016)  \n",
       "1  Molinaro et al. (2016)  \n",
       "2  Molinaro et al. (2016)  \n",
       "3  Molinaro et al. (2016)  \n",
       "4  Molinaro et al. (2016)  \n",
       "5  Molinaro et al. (2016)  \n",
       "6  Molinaro et al. (2016)  \n",
       "7  Molinaro et al. (2016)  \n",
       "8  Molinaro et al. (2016)  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Molinaro2016.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5719dd01",
   "metadata": {},
   "source": [
    "## 3.7 Computational discovery and in vivo validation of hnf4 as a regulatory gene in planarian regeneration\n",
    "Lobo, D., Morokuma, J. and Levin, M., 2016. Computational discovery and in vivo validation of hnf4 as a regulatory gene in planarian regeneration. Bioinformatics, 32(17), pp.2681-2685."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2bd2833c",
   "metadata": {},
   "source": [
    "# 4. 2017"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6ea16884",
   "metadata": {},
   "source": [
    "## 4.1 Orthogonal muscle fibres have different instructive roles in planarian regeneration\n",
    "Scimone, M.L., Cote, L.E. and Reddien, P.W., 2017. Orthogonal muscle fibres have different instructive roles in planarian regeneration. Nature, 551(7682), pp.623-628."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c4e1a417",
   "metadata": {},
   "source": [
    "The TF-encoding gene nkx1-1 was required for the formation of circular fibres. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "17641641",
   "metadata": {},
   "source": [
    "New transcription factors:\n",
    "* NK homeobox1-1 (nkx1-1)\n",
    "* NK homeobox6-2 (nkx6-2)\n",
    "* ladybird (lbx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "2d9c6dc2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">MF070478.1 Schmidtea mediterranea NK homeobox1-1 (nkx1-1) mRNA, partial cds\n",
      ">MF070480.1 Schmidtea mediterranea NK homeobox6-2 (nkx6-2) mRNA, complete cds\n",
      ">MF070479.1 Schmidtea mediterranea ladybird (lbx) mRNA, partial cds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "grep \">\" Scimone2017.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "bd4a3349",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Scimone2017.fa -out blast_Scimone2017 -outfmt 6 &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2a652d06",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Scimone2017 = pd.read_csv(\"/hydra/TF_data/blast_Scimone2017\", sep=\"\\t\", header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "f82fd591",
   "metadata": {},
   "outputs": [],
   "source": [
    "blast_Scimone2017.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Scimone2017 = blast_Scimone2017.loc[blast_Scimone2017.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "db105c32",
   "metadata": {},
   "outputs": [],
   "source": [
    "Scimone2017 = pd.DataFrame({\"ID\": [\"MF070478.1\", \"MF070480.1\", \"MF070479.1\"], \"Name\": [\"nkx1-1\", \"nkx6-2\", \"lbx\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "eece8dfa",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Scimone2017= tf_table(blast_Scimone2017, \"Scimone et al. (2017)\", Scimone2017)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "53cbec3a",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Scimone2017.to_csv(\"/hydra/TF_data/TF_Scimone2017.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "34a036df",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "TF_table_Scimone2017 = pd.read_csv(\"/hydra/TF_data/TF_Scimone2017.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "2bf44e66",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>SMEST051002001.1</td>\n",
       "      <td>SMESG000051002.1</td>\n",
       "      <td>MSTRG.13371</td>\n",
       "      <td>MF070478</td>\n",
       "      <td>nkx1-1</td>\n",
       "      <td>Scimone et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>SMEST063219001.1</td>\n",
       "      <td>SMESG000063219.1</td>\n",
       "      <td>MSTRG.18888</td>\n",
       "      <td>MF070479</td>\n",
       "      <td>lbx</td>\n",
       "      <td>Scimone et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>SMEST023118001.1</td>\n",
       "      <td>SMESG000023118.1</td>\n",
       "      <td>MSTRG.7287</td>\n",
       "      <td>MF070480</td>\n",
       "      <td>nkx6-2</td>\n",
       "      <td>Scimone et al. (2017)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0     Transcript.ID           Rink.ID     Neiro.ID Original.ID  \\\n",
       "0           0  SMEST051002001.1  SMESG000051002.1  MSTRG.13371    MF070478   \n",
       "1           1  SMEST063219001.1  SMESG000063219.1  MSTRG.18888    MF070479   \n",
       "2           2  SMEST023118001.1  SMESG000023118.1   MSTRG.7287    MF070480   \n",
       "\n",
       "  Original.name                 Source  \n",
       "0        nkx1-1  Scimone et al. (2017)  \n",
       "1           lbx  Scimone et al. (2017)  \n",
       "2        nkx6-2  Scimone et al. (2017)  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TF_table_Scimone2017"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "644899bd",
   "metadata": {},
   "source": [
    "## 4.2 FOX and ETS family transcription factors regulate the pigment cell lineage in planarians\n",
    "He, X., Lindsay-Mosher, N., Li, Y., Molinaro, A.M., Pellettieri, J. and Pearson, B.J., 2017. FOX and ETS family transcription factors regulate the pigment cell lineage in planarians. Development, 144(24), pp.4540-4551."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3140046d",
   "metadata": {},
   "source": [
    "Pigment cells in adult organisms during tissue homeostasis and repair. We also report roles for FOXF-1 and ETS-1 transcription factors in the specification and maintenance of this cell type. \n",
    "\n",
    "All three pigment cells highly express the putative endodermal-specific ʻgamma-neoblast' marker hnf4, which indicated that planarian pigment cells might originate from stem cells with endodermal characteristics. We tested this idea by RNAi knockdown of hnf4, and although the experiment was complicated by the health defects that occur when RNAi animals are also exposed to light, we did observe some changes in the intensity of dendritic marker staining.\n",
    "\n",
    "We next analyzed the co-expression of transcription factors that were highly correlated with PBGD-1 in the scRNAseq data (Pearson correlation ≥0.3). In total, we found 17 transcription factors that met this criterion (Table S2). We performed an RNAi screen of the transcription factors co-expressed in the pigment cells, and found two that were required for normal pigmentation pattern or intensity of pigmentation, similar to Smed-fgfrL-1 and albino: a FoxF homolog (Smed-foxF-1) and an ETS-1 family homolog (Smed-ets-1)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "65a45237",
   "metadata": {},
   "outputs": [],
   "source": [
    "#%%bash\n",
    "#cd /hydra/TF_data\n",
    "#wget http://www.biologists.com/DEV_Movies/DEV156349/TableS2.xlsx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "2f615a0d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "He2017TFs = pd.read_excel(\"/hydra/TF_data/TableS2.xlsx\").iloc[10:40,:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "0df2aa52",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">GCZZ01033334.1 TSA: Schmidtea mediterranea SmedASXL_079276 transcribed RNA sequence\n",
      ">GCZZ01068088.1 TSA: Schmidtea mediterranea SmedASXL_011146 transcribed RNA sequence\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "grep \"SmedASXL_079276\" GCZZ01.1.fsa_nt\n",
    "grep \"SmedASXL_011146\" GCZZ01.1.fsa_nt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "df7ceb97",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "echo \"GCZZ01033334.1 TSA: Schmidtea mediterranea SmedASXL_079276 transcribed RNA sequence\" > He2017.list\n",
    "echo \"GCZZ01068088.1 TSA: Schmidtea mediterranea SmedASXL_011146 transcribed RNA sequence\" >> He2017.list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "3f4b7a55",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "seqtk subseq GCZZ01.1.fsa_nt He2017.list > He2017.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "0f4de74b",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query He2017.fa -out blast_He2017 -outfmt 6 &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "135ae674",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_He2017 = pd.read_csv(\"/hydra/TF_data/blast_He2017\", sep=\"\\t\", header=None)\n",
    "blast_He2017.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_He2017 = blast_He2017.loc[blast_He2017.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "c5e4a2dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "He2017 = pd.DataFrame({\"ID\": [\"GCZZ01033334.1\", \"GCZZ01068088.1\"], \"Name\": [\"ets-1\", \"foxF-1\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "3a37ba46",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_He2017 = tf_table(blast_He2017, \"He et al. (2017)\", He2017)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "05a90635",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_He2017.to_csv(\"/hydra/TF_data/He2017.csv\" ,index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "41930467",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MSTRG.19528.3</td>\n",
       "      <td>SMESG000065612.1</td>\n",
       "      <td>MSTRG.19528</td>\n",
       "      <td>GCZZ01033334</td>\n",
       "      <td>ets-1</td>\n",
       "      <td>He et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMEST066497001.1</td>\n",
       "      <td>SMESG000066497.1</td>\n",
       "      <td>MSTRG.19765</td>\n",
       "      <td>GCZZ01068088</td>\n",
       "      <td>foxF-1</td>\n",
       "      <td>He et al. (2017)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Transcript.ID           Rink.ID     Neiro.ID   Original.ID  \\\n",
       "0     MSTRG.19528.3  SMESG000065612.1  MSTRG.19528  GCZZ01033334   \n",
       "1  SMEST066497001.1  SMESG000066497.1  MSTRG.19765  GCZZ01068088   \n",
       "\n",
       "  Original.name            Source  \n",
       "0         ets-1  He et al. (2017)  \n",
       "1        foxF-1  He et al. (2017)  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/He2017.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7b0bf4d3",
   "metadata": {},
   "source": [
    "## 4.3 Epithelial-mesenchymal transition transcription factors control pluripotent adult stem cell migration in vivo in planarians\n",
    "\n",
    "Abnave, P., Aboukhatwa, E., Kosaka, N., Thompson, J., Hill, M.A. and Aboobaker, A.A., 2017. Epithelial-mesenchymal transition transcription factors control pluripotent adult stem cell migration in vivo in planarians. Development, 144(19), pp.3440-3453."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6f086760",
   "metadata": {},
   "source": [
    "### Materials and methods\n",
    "* snail-1: GenBank KY964486\n",
    "* snail-2: GenBank KJ934814.1\n",
    "* zeb-1: GenBank, KY992929"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "68c1802c",
   "metadata": {},
   "outputs": [],
   "source": [
    "Abnave2017 = pd.DataFrame({\"ID\": [\"KY964486.1\", \"KJ934814.1\", \"KY992929.1\"], \"Name\": [\"snail-1\", \"snail-2\", \"zeb-1\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "a7ba6c10",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">KY964486.1 Schmidtea mediterranea snail-1 mRNA, complete cds\n",
      ">KJ934814.1 UNVERIFIED: Schmidtea mediterranea snail-like mRNA, complete sequence\n",
      ">KY992929.1 Schmidtea mediterranea Zeb-1 mRNA, complete cds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "grep \">\" Abnave2017.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "25c82398",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db ~/FACS/stringtie_transcripts.fa -query Abnave2017.fa -out blast_Abnave2017 -outfmt 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "6c0bd508",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Abnave2017 = pd.read_csv(\"/hydra/TF_data/blast_Abnave2017\", sep=\"\\t\", header=None)\n",
    "blast_Abnave2017.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Abnave2017 = blast_Abnave2017.loc[blast_Abnave2017.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "e763236c",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Abnave2017 = tf_table(blast_Abnave2017, \"Abnave et al. (2017)\", Abnave2017)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2ec2fd09",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Abnave2017.to_csv(\"/hydra/TF_data/TF_Abnave2017.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "d57427ce",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MSTRG.17678.2</td>\n",
       "      <td>SMESG000064846.1</td>\n",
       "      <td>MSTRG.17678</td>\n",
       "      <td>KJ934814</td>\n",
       "      <td>snail-2</td>\n",
       "      <td>Abnave et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMEST076673001.1</td>\n",
       "      <td>SMESG000076673.1</td>\n",
       "      <td>MSTRG.22489</td>\n",
       "      <td>KY964486</td>\n",
       "      <td>snail-1</td>\n",
       "      <td>Abnave et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MSTRG.4666.2</td>\n",
       "      <td>SMESG000013012.1</td>\n",
       "      <td>MSTRG.4666</td>\n",
       "      <td>KY992929</td>\n",
       "      <td>zeb-1</td>\n",
       "      <td>Abnave et al. (2017)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Transcript.ID           Rink.ID     Neiro.ID Original.ID Original.name  \\\n",
       "0     MSTRG.17678.2  SMESG000064846.1  MSTRG.17678    KJ934814       snail-2   \n",
       "1  SMEST076673001.1  SMESG000076673.1  MSTRG.22489    KY964486       snail-1   \n",
       "2      MSTRG.4666.2  SMESG000013012.1   MSTRG.4666    KY992929         zeb-1   \n",
       "\n",
       "                 Source  \n",
       "0  Abnave et al. (2017)  \n",
       "1  Abnave et al. (2017)  \n",
       "2  Abnave et al. (2017)  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Abnave2017.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea07cd38",
   "metadata": {},
   "source": [
    "## 4.4 The pioneer factor Smed-gata456-1 is required for gut cell differentiation and maintenance in planarians\n",
    "González Sastre, A., Sousa, N.D., Adell i Creixell, T. and Saló i Boix, E., 2017. The pioneer factor Smed-gata456-1 is required for gut cell differentiation and maintenance in planarians. International Journal of Developmental Biology, 2017, vol. 61, p. 53-63."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4fc14f5c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">KX827244.1 Schmidtea mediterranea GATA456-2 mRNA, complete cds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "grep \">\" Gonzalez2017.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b247658d",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Gonzalez2017.fa -out blast_Gonzalez2017 -outfmt 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ed265467",
   "metadata": {},
   "outputs": [],
   "source": [
    "Gonzalez2017 = pd.DataFrame({\"ID\": [\"KX827244.1\"], \"Name\": [\"gata456-2\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "362ccfa4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Gonzalez2017 = pd.read_csv(\"/hydra/TF_data/blast_Gonzalez2017\", sep=\"\\t\", header=None)\n",
    "blast_Gonzalez2017.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Gonzalez2017 = blast_Gonzalez2017.loc[blast_Gonzalez2017.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "068f9174",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Gonzalez2017 = tf_table(blast_Gonzalez2017, \"González Sastre et al. (2017)\", Gonzalez2017)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "cd8bdb2b",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Gonzalez2017.to_csv(\"/hydra/TF_data/TF_Gonzalez2017.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "2f97408a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MSTRG.8018.10</td>\n",
       "      <td>SMESG000022702.1</td>\n",
       "      <td>MSTRG.8018</td>\n",
       "      <td>KX827244</td>\n",
       "      <td>gata456-2</td>\n",
       "      <td>González Sastre et al. (2017)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Transcript.ID           Rink.ID    Neiro.ID Original.ID Original.name  \\\n",
       "0  MSTRG.8018.10  SMESG000022702.1  MSTRG.8018    KX827244     gata456-2   \n",
       "\n",
       "                          Source  \n",
       "0  González Sastre et al. (2017)  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Gonzalez2017.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb1a71b3",
   "metadata": {},
   "source": [
    "## 4.5 Eye absence does not regulate planarian stem cells during eye regeneration\n",
    "LoCascio, S.A., Lapan, S.W. and Reddien, P.W., 2017. Eye absence does not regulate planarian stem cells during eye regeneration. Developmental cell, 40(4), pp.381-391."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3022696",
   "metadata": {},
   "source": [
    "## 4.6 Generic wound signals initiate regeneration in missing-tissue contexts\n",
    "Owlarn, S., Klenner, F., Schmidt, D., Rabert, F., Tomasso, A., Reuter, H., Mulaw, M.A., Moritz, S., Gentile, L., Weidinger, G. and Bartscherer, K., 2017. Generic wound signals initiate regeneration in missing-tissue contexts. Nature communications, 8(1), pp.1-13."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "ea858b76",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">egr2-F\n",
      ">egr2-R\n",
      ">egrl1-F\n",
      ">egrl1-R\n",
      ">mex3-F\n",
      ">mex3-R\n",
      ">runt1-F\n",
      ">runt1-R\n",
      ">runt2-F\n",
      ">runt2-R\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "grep \">\" Owlarn2017.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "0eec241e",
   "metadata": {},
   "outputs": [],
   "source": [
    "Owlarn2017 = pd.DataFrame({\"ID\": [\"egr2-F\", \"egr2-R\", \"egrl1-F\", \"egrl1-R\", \"mex3-F\", \"mex3-R\",\n",
    "                                 \"runt1-F\", \"runt1-R\", \"runt2-F\", \"runt2-R\"], \"Name\": [\"egr2\", \n",
    "                                    \"egr2\", \"egrl1\", \"egrl1\", \"mex3\", \"mex3\", \"runt1\", \"runt1\", \"runt2\"\n",
    "                                , \"runt2\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "efb838fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Owlarn2017.fa -out blast_Owlarn2017 -outfmt 6 -evalue 1.0 -word_size 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "156e257c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Owlarn2017 = pd.read_csv(\"/hydra/TF_data/blast_Owlarn2017\", sep=\"\\t\", header=None)\n",
    "blast_Owlarn2017.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Owlarn2017 = blast_Owlarn2017.loc[blast_Owlarn2017.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "e10aaddd",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Owlarn2017 = tf_table(blast_Owlarn2017, \"Owlarn et al. (2017)\", Owlarn2017)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "b8ecd85b",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Owlarn2017.to_csv(\"/hydra/TF_data/TF_Owlarn2017.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "bcb5918e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SMEST004840001.1</td>\n",
       "      <td>SMESG000004840.1</td>\n",
       "      <td>MSTRG.2271</td>\n",
       "      <td>NaN</td>\n",
       "      <td>egr2</td>\n",
       "      <td>Owlarn et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMEST004840001.1</td>\n",
       "      <td>SMESG000004840.1</td>\n",
       "      <td>MSTRG.2271</td>\n",
       "      <td>NaN</td>\n",
       "      <td>egr2</td>\n",
       "      <td>Owlarn et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SMEST000960001.1</td>\n",
       "      <td>SMESG000000960.1</td>\n",
       "      <td>MSTRG.1218</td>\n",
       "      <td>NaN</td>\n",
       "      <td>egrl1</td>\n",
       "      <td>Owlarn et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SMEST000960001.1</td>\n",
       "      <td>SMESG000000960.1</td>\n",
       "      <td>MSTRG.1218</td>\n",
       "      <td>NaN</td>\n",
       "      <td>egrl1</td>\n",
       "      <td>Owlarn et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SMEST071674001.1</td>\n",
       "      <td>SMESG000071674.1</td>\n",
       "      <td>MSTRG.21084</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mex3</td>\n",
       "      <td>Owlarn et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SMEST071674001.1</td>\n",
       "      <td>SMESG000071674.1</td>\n",
       "      <td>MSTRG.21084</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mex3</td>\n",
       "      <td>Owlarn et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MSTRG.4469.4</td>\n",
       "      <td>SMESG000014779.1</td>\n",
       "      <td>MSTRG.4469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>runt1</td>\n",
       "      <td>Owlarn et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MSTRG.4469.4</td>\n",
       "      <td>SMESG000014779.1</td>\n",
       "      <td>MSTRG.4469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>runt1</td>\n",
       "      <td>Owlarn et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MSTRG.15139.5</td>\n",
       "      <td>SMESG000049171.1</td>\n",
       "      <td>MSTRG.15139</td>\n",
       "      <td>NaN</td>\n",
       "      <td>runt2</td>\n",
       "      <td>Owlarn et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MSTRG.15139.5</td>\n",
       "      <td>SMESG000049171.1</td>\n",
       "      <td>MSTRG.15139</td>\n",
       "      <td>NaN</td>\n",
       "      <td>runt2</td>\n",
       "      <td>Owlarn et al. (2017)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Transcript.ID           Rink.ID     Neiro.ID  Original.ID Original.name  \\\n",
       "0  SMEST004840001.1  SMESG000004840.1   MSTRG.2271          NaN          egr2   \n",
       "1  SMEST004840001.1  SMESG000004840.1   MSTRG.2271          NaN          egr2   \n",
       "2  SMEST000960001.1  SMESG000000960.1   MSTRG.1218          NaN         egrl1   \n",
       "3  SMEST000960001.1  SMESG000000960.1   MSTRG.1218          NaN         egrl1   \n",
       "4  SMEST071674001.1  SMESG000071674.1  MSTRG.21084          NaN          mex3   \n",
       "5  SMEST071674001.1  SMESG000071674.1  MSTRG.21084          NaN          mex3   \n",
       "6      MSTRG.4469.4  SMESG000014779.1   MSTRG.4469          NaN         runt1   \n",
       "7      MSTRG.4469.4  SMESG000014779.1   MSTRG.4469          NaN         runt1   \n",
       "8     MSTRG.15139.5  SMESG000049171.1  MSTRG.15139          NaN         runt2   \n",
       "9     MSTRG.15139.5  SMESG000049171.1  MSTRG.15139          NaN         runt2   \n",
       "\n",
       "                 Source  \n",
       "0  Owlarn et al. (2017)  \n",
       "1  Owlarn et al. (2017)  \n",
       "2  Owlarn et al. (2017)  \n",
       "3  Owlarn et al. (2017)  \n",
       "4  Owlarn et al. (2017)  \n",
       "5  Owlarn et al. (2017)  \n",
       "6  Owlarn et al. (2017)  \n",
       "7  Owlarn et al. (2017)  \n",
       "8  Owlarn et al. (2017)  \n",
       "9  Owlarn et al. (2017)  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Owlarn2017.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f69ee264",
   "metadata": {},
   "source": [
    "## 4.7 Planarian epidermal stem cells respond to positional cues to promote cell type diversity\n",
    "Wurtzel, O., Oderberg, I.M. and Reddien, P.W., 2017. Planarian epidermal stem cells respond to positional cues to promote cell-type diversity. Developmental cell, 40(5), pp.491-504."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "766f20fa",
   "metadata": {},
   "source": [
    "* zfp-1\tdd_Smed_v4_8720_0_1\n",
    "* ovo-2\tdd_Smed_v4_10673_0_1\n",
    "* soxP-3\tdd_Smed_v4_5942_0_1\n",
    "* foxJ1-4\tdd_Smed_v4_10152_0_1\n",
    "* Post-2b\tdd_Smed_v4_9061_0_1\n",
    "* ovo-3\tdd_Smed_v4_12010_0_1\n",
    "* BARHL1\tdd_Smed_v4_18719_0_1\n",
    "* FOXJ2\tdd_Smed_v4_12170_0_1\n",
    "* Post-2a\tdd_Smed_v4_16021_0_1\n",
    "* tlx-1\tdd_Smed_v4_21347_0_1\n",
    "* TCF/LEF-1-like\tdd_Smed_v4_13056_0_1\n",
    "* snail-like\tdd_Smed_v4_10216_0_1\n",
    "* otxb-like\tdd_Smed_v4_15516_0_1\n",
    "* TBX2\tdd_Smed_v4_11693_0_1\n",
    "* PRDM1-1\tdd_Smed_v4_11840_0_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "bb779674",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Wurtzel2017.list\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "echo \"dd_Smed_v6_8720_0_1\" > Wurtzel2017.list\n",
    "echo \"dd_Smed_v6_10673_0_1\" >> Wurtzel2017.list\n",
    "echo \"dd_Smed_v6_5942_0_1\" >> Wurtzel2017.list\n",
    "echo \"dd_Smed_v6_10152_0_1\" >> Wurtzel2017.list\n",
    "echo \"dd_Smed_v6_9061_0_1\" >> Wurtzel2017.list\n",
    "echo \"dd_Smed_v6_12010_0_1\" >> Wurtzel2017.list\n",
    "echo \"dd_Smed_v6_18719_0_1\" >> Wurtzel2017.list\n",
    "echo \"dd_Smed_v6_12170_0_1\" >> Wurtzel2017.list\n",
    "echo \"dd_Smed_v6_16021_0_1\" >> Wurtzel2017.list\n",
    "echo \"dd_Smed_v6_21347_0_1\" >> Wurtzel2017.list\n",
    "echo \"dd_Smed_v6_13056_0_1\" >> Wurtzel2017.list\n",
    "echo \"dd_Smed_v6_10216_0_1\" >> Wurtzel2017.list\n",
    "echo \"dd_Smed_v6_15516_0_1\" >> Wurtzel2017.list\n",
    "echo \"dd_Smed_v6_11693_0_1\" >> Wurtzel2017.list\n",
    "echo \"dd_Smed_v6_11840_0_1\" >> Wurtzel2017.list\n",
    "wc -l Wurtzel2017.list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "23fcd81d",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "seqtk subseq /hydra/sexual_genome_annotation_files/sexual_genome_annotation/dd_Smed_v6.pcf.contigs.fasta Wurtzel2017.list > Wurtzel2017.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "60b836d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "Wurtzel2017 = pd.DataFrame({\"ID\": [\"dd_Smed_v6_8720_0_1\", \"dd_Smed_v6_10673_0_1\", \"dd_Smed_v6_5942_0_1\", \"dd_Smed_v6_10152_0_1\", \n",
    "                                  \"dd_Smed_v6_9061_0_1\", \"dd_Smed_v6_12010_0_1\", \"dd_Smed_v6_18719_0_1\", \"dd_Smed_v6_12170_0_1\", \"dd_Smed_v6_16021_0_1\",\n",
    "                                  \"dd_Smed_v6_21347_0_1\", \"dd_Smed_v6_13056_0_1\", \"dd_Smed_v6_10216_0_1\", \"dd_Smed_v6_15516_0_1\",\n",
    "                                  \"dd_Smed_v6_11693_0_1\", \"dd_Smed_v6_11840_0_1\"], \n",
    "                           \"Name\": [\"zfp-1\", \"ovo-2\", \"soxP-3\", \"foxJ1-4\", \"Post-2b\", \"ovo-3\", \"BARHL1\", \"foxJ2\", \"Post-2a\", \"tlx-1\", \"tcf-1\", \"snail\", \n",
    "                                    \"otxb\", \"tbx2\", \"PRDM1-1\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "e6a710b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Wurtzel2017.fa -out blast_Wurtzel2017 -outfmt 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "3ce00f3e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Wurtzel2017 = pd.read_csv(\"/hydra/TF_data/blast_Wurtzel2017\", sep=\"\\t\", header=None)\n",
    "blast_Wurtzel2017.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Wurtzel2017 = blast_Wurtzel2017.loc[blast_Wurtzel2017.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "2eedfaa6",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_tableWurtzel2017 = tf_table(blast_Wurtzel2017, \"WUrtzel et al. (2017)\", Wurtzel2017)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "2220ffa4",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_tableWurtzel2017.to_csv(\"/hydra/TF_data/TF_Wurtzel2017.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "b8eeb474",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SMEST010030001.1</td>\n",
       "      <td>SMESG000010030.1</td>\n",
       "      <td>MSTRG.3843</td>\n",
       "      <td>NaN</td>\n",
       "      <td>foxJ1-4</td>\n",
       "      <td>WUrtzel et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMEST064846001.1</td>\n",
       "      <td>SMESG000064846.1</td>\n",
       "      <td>MSTRG.17678</td>\n",
       "      <td>NaN</td>\n",
       "      <td>snail</td>\n",
       "      <td>WUrtzel et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SMEST016051001.1</td>\n",
       "      <td>SMESG000016051.1</td>\n",
       "      <td>MSTRG.5778</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ovo-2</td>\n",
       "      <td>WUrtzel et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SMEST061015001.1</td>\n",
       "      <td>SMESG000061015.1</td>\n",
       "      <td>MSTRG.18243</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tbx2</td>\n",
       "      <td>WUrtzel et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SMEST069311002.1</td>\n",
       "      <td>SMESG000069311.1</td>\n",
       "      <td>MSTRG.19125</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRDM1-1</td>\n",
       "      <td>WUrtzel et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SMEST006550001.1</td>\n",
       "      <td>SMESG000006550.1</td>\n",
       "      <td>MSTRG.1647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ovo-3</td>\n",
       "      <td>WUrtzel et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SMEST044628002.1</td>\n",
       "      <td>SMESG000044628.1</td>\n",
       "      <td>MSTRG.13758</td>\n",
       "      <td>NaN</td>\n",
       "      <td>foxJ2</td>\n",
       "      <td>WUrtzel et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SMEST009225001.1</td>\n",
       "      <td>SMESG000009225.1</td>\n",
       "      <td>MSTRG.2747</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tcf-1</td>\n",
       "      <td>WUrtzel et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SMEST023717004.1</td>\n",
       "      <td>SMESG000023717.1</td>\n",
       "      <td>MSTRG.8421</td>\n",
       "      <td>NaN</td>\n",
       "      <td>otxb</td>\n",
       "      <td>WUrtzel et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SMEST054107001.1</td>\n",
       "      <td>SMESG000054107.1</td>\n",
       "      <td>MSTRG.16300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Post-2a</td>\n",
       "      <td>WUrtzel et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SMEST074896001.1</td>\n",
       "      <td>SMESG000074896.1</td>\n",
       "      <td>MSTRG.21923</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BARHL1</td>\n",
       "      <td>WUrtzel et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SMEST027543002.1</td>\n",
       "      <td>SMESG000027543.1</td>\n",
       "      <td>MSTRG.9335</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tlx-1</td>\n",
       "      <td>WUrtzel et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MSTRG.13411.1</td>\n",
       "      <td>SMESG000051170.1</td>\n",
       "      <td>MSTRG.13411</td>\n",
       "      <td>NaN</td>\n",
       "      <td>soxP-3</td>\n",
       "      <td>WUrtzel et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MSTRG.19740.1</td>\n",
       "      <td>SMESG000066384.1</td>\n",
       "      <td>MSTRG.19740</td>\n",
       "      <td>NaN</td>\n",
       "      <td>zfp-1</td>\n",
       "      <td>WUrtzel et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SMEST033048001.1</td>\n",
       "      <td>SMESG000033048.1</td>\n",
       "      <td>MSTRG.10796</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Post-2b</td>\n",
       "      <td>WUrtzel et al. (2017)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Transcript.ID           Rink.ID     Neiro.ID  Original.ID  \\\n",
       "0   SMEST010030001.1  SMESG000010030.1   MSTRG.3843          NaN   \n",
       "1   SMEST064846001.1  SMESG000064846.1  MSTRG.17678          NaN   \n",
       "2   SMEST016051001.1  SMESG000016051.1   MSTRG.5778          NaN   \n",
       "3   SMEST061015001.1  SMESG000061015.1  MSTRG.18243          NaN   \n",
       "4   SMEST069311002.1  SMESG000069311.1  MSTRG.19125          NaN   \n",
       "5   SMEST006550001.1  SMESG000006550.1   MSTRG.1647          NaN   \n",
       "6   SMEST044628002.1  SMESG000044628.1  MSTRG.13758          NaN   \n",
       "7   SMEST009225001.1  SMESG000009225.1   MSTRG.2747          NaN   \n",
       "8   SMEST023717004.1  SMESG000023717.1   MSTRG.8421          NaN   \n",
       "9   SMEST054107001.1  SMESG000054107.1  MSTRG.16300          NaN   \n",
       "10  SMEST074896001.1  SMESG000074896.1  MSTRG.21923          NaN   \n",
       "11  SMEST027543002.1  SMESG000027543.1   MSTRG.9335          NaN   \n",
       "12     MSTRG.13411.1  SMESG000051170.1  MSTRG.13411          NaN   \n",
       "13     MSTRG.19740.1  SMESG000066384.1  MSTRG.19740          NaN   \n",
       "14  SMEST033048001.1  SMESG000033048.1  MSTRG.10796          NaN   \n",
       "\n",
       "   Original.name                 Source  \n",
       "0        foxJ1-4  WUrtzel et al. (2017)  \n",
       "1          snail  WUrtzel et al. (2017)  \n",
       "2          ovo-2  WUrtzel et al. (2017)  \n",
       "3           tbx2  WUrtzel et al. (2017)  \n",
       "4        PRDM1-1  WUrtzel et al. (2017)  \n",
       "5          ovo-3  WUrtzel et al. (2017)  \n",
       "6          foxJ2  WUrtzel et al. (2017)  \n",
       "7          tcf-1  WUrtzel et al. (2017)  \n",
       "8           otxb  WUrtzel et al. (2017)  \n",
       "9        Post-2a  WUrtzel et al. (2017)  \n",
       "10        BARHL1  WUrtzel et al. (2017)  \n",
       "11         tlx-1  WUrtzel et al. (2017)  \n",
       "12        soxP-3  WUrtzel et al. (2017)  \n",
       "13         zfp-1  WUrtzel et al. (2017)  \n",
       "14       Post-2b  WUrtzel et al. (2017)  "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Wurtzel2017.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb53d788",
   "metadata": {},
   "source": [
    "## 4.8 Antagonistic Self-Organizing Patterning Systems Control Maintenance and Regeneration of the Anteroposterior Axis in Planarians\n",
    "Stückemann, T., Cleland, J.P., Werner, S., Vu, H.T.K., Bayersdorf, R., Liu, S.Y., Friedrich, B., Jülicher, F. and Rink, J.C., 2017. Antagonistic self-organizing patterning systems control maintenance and regeneration of the anteroposterior axis in planarians. Developmental cell, 40(3), pp.248-263."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f44c57f2",
   "metadata": {},
   "source": [
    "* Smed-Hox1\tdd_Smed_v6_62255_0_1\n",
    "* Smed-Isl-1\tdd_Smed_v6_8820_0_1\n",
    "* Smed-Plox6\tdd_Smed_v6_22524_0_1\n",
    "* Smed-Post-2c\tdd_Smed_v6_12467_0_1\n",
    "* Smed-Ski\tdd_Smed_v6_10921_0_1\n",
    "* Smed-Xlox\tdd_Smed_v6_25769_0_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "27b4f140",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6 Stuckemann2017.list\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "echo \"dd_Smed_v6_62255_0_1\" > Stuckemann2017.list\n",
    "echo \"dd_Smed_v6_8820_0_1\" >> Stuckemann2017.list\n",
    "echo \"dd_Smed_v6_22524_0_1\" >> Stuckemann2017.list\n",
    "echo \"dd_Smed_v6_12467_0_1\" >> Stuckemann2017.list\n",
    "echo \"dd_Smed_v6_10921_0_1\" >> Stuckemann2017.list\n",
    "echo \"dd_Smed_v6_25769_0_1\" >> Stuckemann2017.list\n",
    "wc -l Stuckemann2017.list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c69ac3af",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "seqtk subseq /hydra/sexual_genome_annotation_files/sexual_genome_annotation/dd_Smed_v6.pcf.contigs.fasta Stuckemann2017.list > Stuckemann2017.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a4ebd541",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "Stuckemann2017 = pd.DataFrame({\"ID\": [\"dd_Smed_v6_62255_0_1\", \"dd_Smed_v6_8820_0_1\", \"dd_Smed_v6_22524_0_1\", \"dd_Smed_v6_12467_0_1\", \"dd_Smed_v6_10921_0_1\", \"dd_Smed_v6_25769_0_1\"], \n",
    "                           \"Name\": [\"Hox1\", \"Isl-1\", \"Plox6\", \"Post-2c\", \"Ski\", \"Xlox\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "376eaaa4",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Stuckemann2017.fa -out blast_Stuckemann2017 -outfmt 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "3fd74a5f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Stuckemann2017 = pd.read_csv(\"/hydra/TF_data/blast_Stuckemann2017\", sep=\"\\t\", header=None)\n",
    "blast_Stuckemann2017.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Stuckemann2017 = blast_Stuckemann2017.loc[blast_Stuckemann2017.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "043baaa8",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_tableStuckemann2017 = tf_table(blast_Stuckemann2017, \"Stuckemann et al. (2017)\", Stuckemann2017)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "d3400bb6",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_tableStuckemann2017.to_csv(\"/hydra/TF_data/TF_Stuckemann2017.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "eea8552a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SMEST067290001.1</td>\n",
       "      <td>SMESG000067290.1</td>\n",
       "      <td>MSTRG.19902</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Ski</td>\n",
       "      <td>Stuckemann et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMEST022668001.1</td>\n",
       "      <td>SMESG000022668.1</td>\n",
       "      <td>MSTRG.7999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Post-2c</td>\n",
       "      <td>Stuckemann et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SMEST079807002.1</td>\n",
       "      <td>SMESG000079807.1</td>\n",
       "      <td>MSTRG.23420</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Plox6</td>\n",
       "      <td>Stuckemann et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MSTRG.442.2</td>\n",
       "      <td>SMESG000025656.1</td>\n",
       "      <td>MSTRG.442</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Xlox</td>\n",
       "      <td>Stuckemann et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SMEST080661001.1</td>\n",
       "      <td>SMESG000080661.1</td>\n",
       "      <td>MSTRG.23578</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hox1</td>\n",
       "      <td>Stuckemann et al. (2017)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SMEST059633001.1</td>\n",
       "      <td>SMESG000059633.1</td>\n",
       "      <td>MSTRG.17848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Isl-1</td>\n",
       "      <td>Stuckemann et al. (2017)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Transcript.ID           Rink.ID     Neiro.ID  Original.ID Original.name  \\\n",
       "0  SMEST067290001.1  SMESG000067290.1  MSTRG.19902          NaN           Ski   \n",
       "1  SMEST022668001.1  SMESG000022668.1   MSTRG.7999          NaN       Post-2c   \n",
       "2  SMEST079807002.1  SMESG000079807.1  MSTRG.23420          NaN         Plox6   \n",
       "3       MSTRG.442.2  SMESG000025656.1    MSTRG.442          NaN          Xlox   \n",
       "4  SMEST080661001.1  SMESG000080661.1  MSTRG.23578          NaN          Hox1   \n",
       "5  SMEST059633001.1  SMESG000059633.1  MSTRG.17848          NaN         Isl-1   \n",
       "\n",
       "                     Source  \n",
       "0  Stuckemann et al. (2017)  \n",
       "1  Stuckemann et al. (2017)  \n",
       "2  Stuckemann et al. (2017)  \n",
       "3  Stuckemann et al. (2017)  \n",
       "4  Stuckemann et al. (2017)  \n",
       "5  Stuckemann et al. (2017)  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Stuckemann2017.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "035f59e1",
   "metadata": {},
   "source": [
    "# 5. 2018"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5d40d859",
   "metadata": {},
   "source": [
    "## 5.1 foxF-1 controls specification of non-body wall muscle and phagocytic cells in planarians\n",
    "Scimone, M.L., Wurtzel, O., Malecek, K., Fincher, C.T., Oderberg, I.M., Kravarik, K.M. and Reddien, P.W., 2018. foxF-1 controls specification of non-body wall muscle and phagocytic cells in planarians. Current Biology, 28(23), pp.3787-3801"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9507b635",
   "metadata": {},
   "source": [
    "* Scimone et al. describe the transcriptome of major muscle subsets, identify the transcription factors required for their specification, and analyze their regenerative function. \n",
    "* Besides a role in muscle, foxF-1 is also required for specification of previously unknown planarian phagocytic cells."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "05342951",
   "metadata": {},
   "outputs": [],
   "source": [
    "#%%bash\n",
    "#cd /hydra/TF_data\n",
    "#wget https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6411049/bin/NIHMS1511590-supplement-5.xlsx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "bc1f2622",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "Scimone2018xls = pd.read_excel(\"/hydra/TF_data/NIHMS1511590-supplement-5.xlsx\", skiprows=[0,1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "26d8c0ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "echo \"dd_Smed_v6_10216_0_1\" > Scimone2018.list\n",
    "echo \"dd_Smed_v6_12634_0_1\" >> Scimone2018.list\n",
    "echo \"dd_Smed_v6_31435_0_1\" >> Scimone2018.list\n",
    "echo \"dd_Smed_v6_11521_0_1\" >> Scimone2018.list\n",
    "echo \"dd_Smed_v6_12873_0_1\" >> Scimone2018.list\n",
    "echo \"dd_Smed_v6_10868_0_1\" >> Scimone2018.list\n",
    "echo \"dd_Smed_v6_15253_0_1\" >> Scimone2018.list\n",
    "echo \"dd_Smed_v6_19327_0_1\" >> Scimone2018.list\n",
    "echo \"dd_Smed_v6_13781_0_1\" >> Scimone2018.list\n",
    "echo \"dd_Smed_v6_9910_0_1\" >> Scimone2018.list\n",
    "echo \"dd_Smed_v6_8208_0_1\" >> Scimone2018.list\n",
    "echo \"dd_Smed_v6_15951_0_1\" >> Scimone2018.list\n",
    "echo \"dd_Smed_v6_7470_0_1\" >> Scimone2018.list\n",
    "echo \"dd_Smed_v6_17951_0_1\" >> Scimone2018.list\n",
    "echo \"dd_Smed_v6_11372_0_1\" >> Scimone2018.list\n",
    "echo \"dd_Smed_v6_5579_0_1\" >> Scimone2018.list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f540b42b",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "seqtk subseq /hydra/sexual_genome_annotation_files/sexual_genome_annotation/dd_Smed_v6.pcf.contigs.fasta Scimone2018.list > Scimone2018.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d9af65ff",
   "metadata": {},
   "outputs": [],
   "source": [
    "Scimone2018 = pd.DataFrame({\"ID\": [\"dd_Smed_v6_10216_0_1\", \"dd_Smed_v6_12634_0_1\", \"dd_Smed_v6_31435_0_1\", \"dd_Smed_v6_11521_0_1\", \n",
    "                                   \"dd_Smed_v6_12873_0_1\", \"dd_Smed_v6_10868_0_1\", \"dd_Smed_v6_15253_0_1\", \"dd_Smed_v6_19327_0_1\", \n",
    "                                   \"dd_Smed_v6_13781_0_1\", \"dd_Smed_v6_9910_0_1\", \"dd_Smed_v6_8208_0_1\", \"dd_Smed_v6_15951_0_1\",\n",
    "                                  \"dd_Smed_v6_7470_0_1\", \"dd_Smed_v6_17951_0_1\", \"dd_Smed_v6_11372_0_1\", \"dd_Smed_v6_5579_0_1\"], \n",
    "                           \"Name\": [\"snail\", \"myoD\", \"lbx1\", \"lhx1-5\", \"nkx6-2\", \"fer3l-2\", \"pitx\", \"nkx1-1\", \"tlx-1\", \"gata456-2\", \"gata456-3\", \"meis\", \n",
    "                                    \"gli-1\", \"twist\", \"eya\", \"da\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "849a9946",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Scimone2018.fa -out blast_Scimone2018 -outfmt 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "b052f717",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Scimone2018 = pd.read_csv(\"/hydra/TF_data/blast_Scimone2018\", sep=\"\\t\", header=None)\n",
    "blast_Scimone2018.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Scimone2018 = blast_Scimone2018.loc[blast_Scimone2018.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "1d42c5b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Scimone2018 = tf_table(blast_Scimone2018, \"Scimone et al. (2018)\", Scimone2018)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "d5bd8612",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Scimone2018.to_csv(\"/hydra/TF_data/TF_Scimone2018.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "8b92687d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SMEST064846001.1</td>\n",
       "      <td>SMESG000064846.1</td>\n",
       "      <td>MSTRG.17678</td>\n",
       "      <td>NaN</td>\n",
       "      <td>snail</td>\n",
       "      <td>Scimone et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMEST019274001.1</td>\n",
       "      <td>SMESG000019274.1</td>\n",
       "      <td>MSTRG.6136</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fer3l-2</td>\n",
       "      <td>Scimone et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SMEST001561001.1</td>\n",
       "      <td>SMESG000001561.1</td>\n",
       "      <td>MSTRG.1422</td>\n",
       "      <td>NaN</td>\n",
       "      <td>eya</td>\n",
       "      <td>Scimone et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SMEST043039005.1</td>\n",
       "      <td>SMESG000043039.1</td>\n",
       "      <td>MSTRG.13692</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lhx1-5</td>\n",
       "      <td>Scimone et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SMEST034317001.1</td>\n",
       "      <td>SMESG000034317.1</td>\n",
       "      <td>MSTRG.11214</td>\n",
       "      <td>NaN</td>\n",
       "      <td>myoD</td>\n",
       "      <td>Scimone et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SMEST023118001.1</td>\n",
       "      <td>SMESG000023118.1</td>\n",
       "      <td>MSTRG.7287</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nkx6-2</td>\n",
       "      <td>Scimone et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SMEST017206001.1</td>\n",
       "      <td>SMESG000017206.1</td>\n",
       "      <td>MSTRG.5414</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tlx-1</td>\n",
       "      <td>Scimone et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SMEST080359001.1</td>\n",
       "      <td>SMESG000080359.1</td>\n",
       "      <td>MSTRG.23506</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pitx</td>\n",
       "      <td>Scimone et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SMEST006530002.1</td>\n",
       "      <td>SMESG000006530.1</td>\n",
       "      <td>MSTRG.1897</td>\n",
       "      <td>NaN</td>\n",
       "      <td>meis</td>\n",
       "      <td>Scimone et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SMEST026385001.1</td>\n",
       "      <td>SMESG000026385.1</td>\n",
       "      <td>MSTRG.657</td>\n",
       "      <td>NaN</td>\n",
       "      <td>twist</td>\n",
       "      <td>Scimone et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SMEST051002001.1</td>\n",
       "      <td>SMESG000051002.1</td>\n",
       "      <td>MSTRG.13371</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nkx1-1</td>\n",
       "      <td>Scimone et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SMEST063219001.1</td>\n",
       "      <td>SMESG000063219.1</td>\n",
       "      <td>MSTRG.18888</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lbx1</td>\n",
       "      <td>Scimone et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MSTRG.10086.3</td>\n",
       "      <td>SMESG000032044.1</td>\n",
       "      <td>MSTRG.10086</td>\n",
       "      <td>NaN</td>\n",
       "      <td>da</td>\n",
       "      <td>Scimone et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MSTRG.20466.7</td>\n",
       "      <td>SMESG000073727.1</td>\n",
       "      <td>MSTRG.20466</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gli-1</td>\n",
       "      <td>Scimone et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MSTRG.9120.1</td>\n",
       "      <td>SMESG000028503.1</td>\n",
       "      <td>MSTRG.9120</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gata456-3</td>\n",
       "      <td>Scimone et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MSTRG.8018.2</td>\n",
       "      <td>SMESG000022702.1</td>\n",
       "      <td>MSTRG.8018</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gata456-2</td>\n",
       "      <td>Scimone et al. (2018)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Transcript.ID           Rink.ID     Neiro.ID  Original.ID  \\\n",
       "0   SMEST064846001.1  SMESG000064846.1  MSTRG.17678          NaN   \n",
       "1   SMEST019274001.1  SMESG000019274.1   MSTRG.6136          NaN   \n",
       "2   SMEST001561001.1  SMESG000001561.1   MSTRG.1422          NaN   \n",
       "3   SMEST043039005.1  SMESG000043039.1  MSTRG.13692          NaN   \n",
       "4   SMEST034317001.1  SMESG000034317.1  MSTRG.11214          NaN   \n",
       "5   SMEST023118001.1  SMESG000023118.1   MSTRG.7287          NaN   \n",
       "6   SMEST017206001.1  SMESG000017206.1   MSTRG.5414          NaN   \n",
       "7   SMEST080359001.1  SMESG000080359.1  MSTRG.23506          NaN   \n",
       "8   SMEST006530002.1  SMESG000006530.1   MSTRG.1897          NaN   \n",
       "9   SMEST026385001.1  SMESG000026385.1    MSTRG.657          NaN   \n",
       "10  SMEST051002001.1  SMESG000051002.1  MSTRG.13371          NaN   \n",
       "11  SMEST063219001.1  SMESG000063219.1  MSTRG.18888          NaN   \n",
       "12     MSTRG.10086.3  SMESG000032044.1  MSTRG.10086          NaN   \n",
       "13     MSTRG.20466.7  SMESG000073727.1  MSTRG.20466          NaN   \n",
       "14      MSTRG.9120.1  SMESG000028503.1   MSTRG.9120          NaN   \n",
       "15      MSTRG.8018.2  SMESG000022702.1   MSTRG.8018          NaN   \n",
       "\n",
       "   Original.name                 Source  \n",
       "0          snail  Scimone et al. (2018)  \n",
       "1        fer3l-2  Scimone et al. (2018)  \n",
       "2            eya  Scimone et al. (2018)  \n",
       "3         lhx1-5  Scimone et al. (2018)  \n",
       "4           myoD  Scimone et al. (2018)  \n",
       "5         nkx6-2  Scimone et al. (2018)  \n",
       "6          tlx-1  Scimone et al. (2018)  \n",
       "7           pitx  Scimone et al. (2018)  \n",
       "8           meis  Scimone et al. (2018)  \n",
       "9          twist  Scimone et al. (2018)  \n",
       "10        nkx1-1  Scimone et al. (2018)  \n",
       "11          lbx1  Scimone et al. (2018)  \n",
       "12            da  Scimone et al. (2018)  \n",
       "13         gli-1  Scimone et al. (2018)  \n",
       "14     gata456-3  Scimone et al. (2018)  \n",
       "15     gata456-2  Scimone et al. (2018)  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Scimone2018.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9383e11a",
   "metadata": {},
   "source": [
    "## 5.2 Prospectively isolated tetraspanin+ neoblasts are adult pluripotent stem cells underlying planaria regeneration\n",
    "Zeng, A., Li, H., Guo, L., Gao, X., McKinney, S., Wang, Y., Yu, Z., Park, J., Semerad, C., Ross, E. and Cheng, L.C., 2018. Prospectively isolated tetraspanin+ neoblasts are adult pluripotent stem cells underlying planaria regeneration. Cell, 173(7), pp.1593-1608."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0d41eb5a",
   "metadata": {},
   "source": [
    "## 5.3 The planarian TCF/LEF factor Smed-tcf1 is required for the regeneration of dorsal-lateral neuronal subtypes\n",
    "Brown, D.D., Molinaro, A.M. and Pearson, B.J., 2018. The planarian TCF/LEF factor Smed-tcf1 is required for the regeneration of dorsal-lateral neuronal subtypes. Developmental biology, 433(2), pp.374-383.\n",
    "\n",
    "Smed-tcf1 is required for the regeneration of dorsal-lateral brain neurons. Smed-tcf1 is required for the proper dorsalization of dorsal-lateral GABA neurons. Smed-tcf1 defines a dorsal-lateral brain progenitor cell type.\n",
    "\n",
    "Here we performed a 6 day RNAseq time course on planarians that were regenerating either 0,1, or 2 heads in order to identify novel regulators of brain regeneration. Focusing on transcription factors, we identified a TCF/LEF factor, Smed-tcf1, which was required to correctly pattern the dorsal-lateral cell types of the regenerating brain. \n",
    "\n",
    "A planarian TCF/LEF, Smed-tcf1, was identified as an enriched transcription factor in head regeneration and was detected via whole-mount in situ hybridization (WISH) to be primarily expressed in the brain. Previous studies have shown that Smed-tcf1 (formerly known as 03133HH, Tcf/Lef-1, Smed-tcf7) is expressed in the brain and stem cells, but its function has yet to be characterized.\n",
    "\n",
    "We identified 11 such transcription factors and an additional 5 factors that were enriched in tail regeneration.\n",
    "\n",
    "Smed-tcf1 has previously been shown to be expressed in the CNS and in stem cells, making it a prime candidate in having a putative role in brain regeneration, and therefore, became the focus of this study\n",
    "\n",
    "S. mediterranea possesses 5 TCF/LEF transcription factor homologs. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e67ddd45",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">A.1 Smed-tcf1\n",
      ">B.1 Smed-tcf5\n",
      ">C.1 Smed-tcf2\n",
      ">D.1 Smed-tcf4\n",
      ">E.1 Smed-tcf3\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "grep \">\" Brown2018.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "3dc4b52e",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "tblastn -db /hydra/FACS/stringtie_transcripts.fa -query Brown2018.fa -out blast_Brown2018 -outfmt 6 &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "f2cb33fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Brown2018 = pd.read_csv(\"/hydra/TF_data/blast_Brown2018\", sep=\"\\t\", header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "b1fb061d",
   "metadata": {},
   "outputs": [],
   "source": [
    "blast_Brown2018.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Brown2018 = blast_Brown2018.loc[blast_Brown2018.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "27d7dae7",
   "metadata": {},
   "outputs": [],
   "source": [
    "Brown2018 = pd.DataFrame({\"ID\": [\"A.1\", \"B.1\", \"C.1\", \"D.1\", \"E.1\"], \"Name\": [\"tcf1\", \"tcf5\", \"tcf2\", \"tcf4\", \"tcf3\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "5803f7e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Brown2018= tf_table(blast_Brown2018, \"Brown et al. (2018)\", Brown2018)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "14d33b4b",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Brown2018.to_csv(\"/hydra/TF_data/TF_Brown2018.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "7add9b4d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>SMEST009225001.1</td>\n",
       "      <td>SMESG000009225.1</td>\n",
       "      <td>MSTRG.2747</td>\n",
       "      <td>A</td>\n",
       "      <td>tcf1</td>\n",
       "      <td>Brown et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>MSTRG.5099.3</td>\n",
       "      <td>SMESG000017328.1</td>\n",
       "      <td>MSTRG.5099</td>\n",
       "      <td>B</td>\n",
       "      <td>tcf5</td>\n",
       "      <td>Brown et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>SMEST059443001.1</td>\n",
       "      <td>SMESG000059443.1</td>\n",
       "      <td>MSTRG.15807</td>\n",
       "      <td>C</td>\n",
       "      <td>tcf2</td>\n",
       "      <td>Brown et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>SMEST038786001.1</td>\n",
       "      <td>SMESG000038786.1</td>\n",
       "      <td>MSTRG.12140</td>\n",
       "      <td>D</td>\n",
       "      <td>tcf4</td>\n",
       "      <td>Brown et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>SMEST052392001.1</td>\n",
       "      <td>SMESG000052392.1</td>\n",
       "      <td>MSTRG.15964</td>\n",
       "      <td>E</td>\n",
       "      <td>tcf3</td>\n",
       "      <td>Brown et al. (2018)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0     Transcript.ID           Rink.ID     Neiro.ID Original.ID  \\\n",
       "0           0  SMEST009225001.1  SMESG000009225.1   MSTRG.2747           A   \n",
       "1           1      MSTRG.5099.3  SMESG000017328.1   MSTRG.5099           B   \n",
       "2           2  SMEST059443001.1  SMESG000059443.1  MSTRG.15807           C   \n",
       "3           3  SMEST038786001.1  SMESG000038786.1  MSTRG.12140           D   \n",
       "4           4  SMEST052392001.1  SMESG000052392.1  MSTRG.15964           E   \n",
       "\n",
       "  Original.name               Source  \n",
       "0          tcf1  Brown et al. (2018)  \n",
       "1          tcf5  Brown et al. (2018)  \n",
       "2          tcf2  Brown et al. (2018)  \n",
       "3          tcf4  Brown et al. (2018)  \n",
       "4          tcf3  Brown et al. (2018)  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Brown2018.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5f7f3acc",
   "metadata": {},
   "source": [
    "## 5.4 Smed-myb-1 Specifies Early Temporal Identity during Planarian Epidermal Differentiation\n",
    "\n",
    "Zhu, S.J. and Pearson, B.J., 2018. Smed-myb-1 specifies early temporal identity during planarian epidermal differentiation. Cell reports, 25(1), pp.38-46.\n",
    "\n",
    "Using flow cytometry-isolated cells enriched in epidermal progenitors, we performed transcriptional profiling and RNAi screening to uncover regulators of epidermal differentiation. We identified a MYB-type transcription factor (Smed-myb-1) required for the specification of the first temporal phase of post-mitotic maturation. Knockdown of myb-1 abolished the early progenitor phase of differentiation without ceasing production of subsequent epidermal progenitor states or homeostatic turnover and regeneration of the epidermis.\n",
    "\n",
    "At the neoblast level, the zinc finger protein zfp-1 (van Wolfswinkel et al., 2014), transcription factor p53 (Pearson and Sánchez Alvarado, 2010), and RNA-binding protein mex3-1 (Zhu et al., 2015) have been shown to be required for the production of epidermal progenitors, either through regulating zeta neoblast specification (zfp-1, p53) or general commitment to differentiation (mex3-1). Downstream of neoblasts, it remains relatively unknown why multiple distinct and transient gene expression changes are needed or how they are regulated for this lineage. pax-2/5/8 and soxP-3 transcription factors were found to regulate a subset of early progeny genes but not the progenitor state as a whole (Cheng et al., 2018), whereas the chromatin remodeler chd4 (Scimone et al., 2010) and transcription factor egr-5 (Tu et al., 2015) are required for proper late progeny specification and subsequent lineage maturation (Figure 1A).\n",
    "\n",
    "The myb-1 sequence can be found in the SmedASXL transcriptome assembly under NCBI BioProject PRJNA215411 (transcript number SmedASXL_005044), as well as the Planmine database (http://planmine.mpi-cbg.de/planmine) transcriptome dd_v6 (transcript number dd_Smed_v6_4381_0_1)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6402fece",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "echo \"dd_Smed_v6_4381_0_1\" > Zhu2018.list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "bc6c9cf2",
   "metadata": {},
   "outputs": [],
   "source": [
    "Zhu2018 = pd.DataFrame({\"ID\": [\"dd_Smed_v6_4381_0_1\"], \"Name\": [\"myb-1\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "ae62879c",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "seqtk subseq /hydra/sexual_genome_annotation_files/sexual_genome_annotation/dd_Smed_v6.pcf.contigs.fasta Zhu2018.list > Zhu2018.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "55e2db6c",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Zhu2018.fa -out blast_Zhu2018 -outfmt 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "44f72ed1",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Zhu2018 = pd.read_csv(\"/hydra/TF_data/blast_Zhu2018\", sep=\"\\t\", header=None)\n",
    "blast_Zhu2018.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Zhu2018 = blast_Zhu2018.loc[blast_Zhu2018.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "af89db6b",
   "metadata": {},
   "outputs": [],
   "source": [
    "transcript_id = []\n",
    "gene_id_rink = []\n",
    "gene_id_neiro = []\n",
    "original_id = []\n",
    "original_name = []\n",
    "for i in range(len(blast_Zhu2018)):\n",
    "    t_id = blast_Zhu2018.iloc[i,1]\n",
    "    o_id = blast_Zhu2018.iloc[i,0]\n",
    "    transcript_id.append(t_id)\n",
    "    gene_id_rink.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,0])\n",
    "    gene_id_neiro.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,1])\n",
    "    original_id.append(o_id)\n",
    "    original_name.append(Zhu2018[Zhu2018[\"ID\"] == o_id].iloc[0,1])\n",
    "source = [\"Zhu and Pearson (2018)\"] * len(original_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "be527440",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Zhu2018 = pd.DataFrame({\"Transcript.ID\": transcript_id, \"Rink.ID\": gene_id_rink, \"Neiro.ID\": gene_id_neiro, \"Original.ID\": original_id, \"Original.name\": original_name, \"Source\": source})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "11f52bdf",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Zhu2018.to_csv(\"/hydra/TF_data/TF_Zhu2018.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "58c6ad6b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>SMEST016880001.1</td>\n",
       "      <td>SMESG000016880.1</td>\n",
       "      <td>MSTRG.5224</td>\n",
       "      <td>dd_Smed_v6_4381_0_1</td>\n",
       "      <td>myb-1</td>\n",
       "      <td>Zhu and Pearson (2018)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0     Transcript.ID           Rink.ID    Neiro.ID  \\\n",
       "0           0  SMEST016880001.1  SMESG000016880.1  MSTRG.5224   \n",
       "\n",
       "           Original.ID Original.name                  Source  \n",
       "0  dd_Smed_v6_4381_0_1         myb-1  Zhu and Pearson (2018)  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Zhu2018.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9b697df0",
   "metadata": {},
   "source": [
    "## 5.5 SoxB1 Activity Regulates Sensory Neuron Regeneration, Maintenance, and Function in Planarians\n",
    "Ross, K.G., Molinaro, A.M., Romero, C., Dockter, B., Cable, K.L., Gonzalez, K., Zhang, S., Collins, E.M.S., Pearson, B.J. and Zayas, R.M., 2018. SoxB1 activity regulates sensory neuron regeneration, maintenance, and function in planarians. Developmental cell, 47(3), pp.331-347.\n",
    "\n",
    "The genome of the planarian Schmidtea mediterranea (Grohme et al., 2018) encodes two soxB1 genes named Smed-soxB1-1 and Smed-soxB1-2 (soxB1-1 and -2 hereon). soxB1-1 is expressed in a subset of stem cells and is required for regeneration of anterior photoreceptor neurons (Lapan and Reddien, 2012). soxB1 homologs have also been analyzed during embryonic development in the planarian Schmidtea polychroa, in which they are expressed in putative neural progenitors (Monjo and Romero, 2015). However, the function of SoxB1 genes in planarian neurogenesis, neural regeneration, and neuronal activity is unknown.\n",
    "\n",
    "* dd_Smed_v6_8104_0_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "4ec65c8f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">A.1\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "grep \">\" Ross2018.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "feb1e88b",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Ross2018.fa -out blast_Ross2018 -outfmt 6 &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "30167de4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Ross2018 = pd.read_csv(\"/hydra/TF_data/blast_Ross2018\", sep=\"\\t\", header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "b853eecc",
   "metadata": {},
   "outputs": [],
   "source": [
    "blast_Ross2018.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Ross2018 = blast_Ross2018.loc[blast_Ross2018.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "ad41f547",
   "metadata": {},
   "outputs": [],
   "source": [
    "Ross2018 = pd.DataFrame({\"ID\": [\"A.1\"], \"Name\": [\"soxB1-2\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "be657135",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Ross2018= tf_table(blast_Ross2018, \"Ross et al. (2018)\", Ross2018)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "63b0dd96",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Ross2018.to_csv(\"/hydra/TF_data/TF_Ross2018.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "5594a07a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>SMEST067682001.1</td>\n",
       "      <td>SMESG000067682.1</td>\n",
       "      <td>MSTRG.20007</td>\n",
       "      <td>A</td>\n",
       "      <td>soxB1-2</td>\n",
       "      <td>Ross et al. (2018)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0     Transcript.ID           Rink.ID     Neiro.ID Original.ID  \\\n",
       "0           0  SMEST067682001.1  SMESG000067682.1  MSTRG.20007           A   \n",
       "\n",
       "  Original.name              Source  \n",
       "0       soxB1-2  Ross et al. (2018)  "
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Ross2018.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "df018796",
   "metadata": {},
   "source": [
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d132155e",
   "metadata": {},
   "source": [
    "## 5.6 Comparative transcriptomic analyses and single-cell RNA sequencing of the freshwater planarian Schmidtea mediterranea identify major cell types and pathway conservation\n",
    "Swapna, L.S., Molinaro, A.M., Lindsay-Mosher, N., Pearson, B.J. and Parkinson, J., 2018. Comparative transcriptomic analyses and single-cell RNA sequencing of the freshwater planarian Schmidtea mediterranea identify major cell types and pathway conservation. Genome biology, 19(1), pp.1-22."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9b5d3d28",
   "metadata": {},
   "source": [
    "Several studies have analyzed the role of transcription factors in S. mediterranea - involving pigmentation (Wang 2016), gametogenesis (Iyer 2016), epidermal lineage differentiation (Abnave 2017), regeneration (Flores 2016)\n",
    "\n",
    "Smed-egr-5, AF-4 and Myc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4d6a6e20",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">KU852680.1 Schmidtea mediterranea protein associated with Myc (PAM) mRNA, partial cds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "grep \">\" Swapna2018.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1028bcac",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "Swapna2018 = pd.DataFrame({\"ID\": [\"KU852680.1\"], \"Name\": [\"Myc\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba73bfc1",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Swapna2018.fa -out blast_Swapna2018 -outfmt 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "56c18dce",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Swapna2018 = pd.read_csv(\"/hydra/TF_data/blast_Swapna2018\", sep=\"\\t\", header=None)\n",
    "blast_Swapna2018.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Swapna2018 = blast_Swapna2018.loc[blast_Swapna2018.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e96ae21c",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Swapna2018 = tf_table(blast_Swapna2018, \"Swapna et al. (2018)\", Swapna2018)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "276d5b25",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Swapna2018.to_csv(\"/hydra/TF_data/TF_Swapna2018.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "386e7809",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MSTRG.19667.1</td>\n",
       "      <td>SMESG000066158.1</td>\n",
       "      <td>MSTRG.19667</td>\n",
       "      <td>KU852680</td>\n",
       "      <td>Myc</td>\n",
       "      <td>Swapna et al. (2018)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Transcript.ID           Rink.ID     Neiro.ID Original.ID Original.name  \\\n",
       "0  MSTRG.19667.1  SMESG000066158.1  MSTRG.19667    KU852680           Myc   \n",
       "\n",
       "                 Source  \n",
       "0  Swapna et al. (2018)  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Swapna2018.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "06e9801e",
   "metadata": {},
   "source": [
    "## 5.7 Conservation of epigenetic regulation by the MLL3/4 tumour suppressor in planarian pluripotent stem cells\n",
    "Mihaylova, Y., Abnave, P., Kao, D., Hughes, S., Lai, A., Jaber-Hijazi, F., Kosaka, N. and Aboobaker, A.A., 2018. Conservation of epigenetic regulation by the MLL3/4 tumour suppressor in planarian pluripotent stem cells. Nature communications, 9(1), pp.1-17."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "327b247f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">sp6-9\n",
      ">SoxP1\n",
      ">SoxP2\n",
      ">zfp1\n",
      ">SoxP3\n",
      ">Gata4/5/6\n",
      ">Elf5\n",
      ">Cut-like1\n",
      ">RREBP1\n",
      ">ETS4\n",
      ">FoxA1\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "grep \">\" Mihaylova2018.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bcc22678",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db ~/FACS/stringtie_transcripts.fa -query Mihaylova2018.fa -out blast_Mihaylova2018 -outfmt 6 -evalue 1.0 -word_size 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "d867aa83",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Mihaylova2018 = pd.read_csv(\"/hydra/TF_data/blast_Mihaylova2018\", sep=\"\\t\", header=None)\n",
    "blast_Mihaylova2018.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Mihaylova2018 = blast_Mihaylova2018.loc[blast_Mihaylova2018.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "93df582c",
   "metadata": {},
   "outputs": [],
   "source": [
    "transcript_id = []\n",
    "gene_id_rink = []\n",
    "gene_id_neiro = []\n",
    "original_id = []\n",
    "original_name = []\n",
    "for i in range(len(blast_Mihaylova2018)):\n",
    "    t_id = blast_Mihaylova2018.iloc[i,1]\n",
    "    o_id = \"NA\"\n",
    "    transcript_id.append(t_id)\n",
    "    gene_id_rink.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,0])\n",
    "    gene_id_neiro.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,1])\n",
    "    original_id.append(o_id)\n",
    "    original_name.append(blast_Mihaylova2018.iloc[i,0])\n",
    "source = [\"Mihaylova et al. (2018)\"] * len(original_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "04abcee5",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Mihaylova2018.to_csv(\"/hydra/TF_data/TF_Mihaylova2018.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "88eada6f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SMEST065026002.1</td>\n",
       "      <td>SMESG000065026.1</td>\n",
       "      <td>MSTRG.17728</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cut-like1</td>\n",
       "      <td>Mihaylova et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MSTRG.6709.1</td>\n",
       "      <td>-</td>\n",
       "      <td>MSTRG.6709</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ETS4</td>\n",
       "      <td>Mihaylova et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MSTRG.5700.1</td>\n",
       "      <td>-</td>\n",
       "      <td>MSTRG.5700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Elf5</td>\n",
       "      <td>Mihaylova et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SMEST065670001.1</td>\n",
       "      <td>SMESG000065670.1</td>\n",
       "      <td>MSTRG.19539</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FoxA1</td>\n",
       "      <td>Mihaylova et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MSTRG.12700.8</td>\n",
       "      <td>SMESG000039559.1</td>\n",
       "      <td>MSTRG.12700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gata4/5/6</td>\n",
       "      <td>Mihaylova et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SMEST027936002.1</td>\n",
       "      <td>SMESG000027936.1</td>\n",
       "      <td>MSTRG.9440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RREBP1</td>\n",
       "      <td>Mihaylova et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MSTRG.3856.7</td>\n",
       "      <td>SMESG000010063.1</td>\n",
       "      <td>MSTRG.3856</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SoxP1</td>\n",
       "      <td>Mihaylova et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MSTRG.21964.4</td>\n",
       "      <td>SMESG000074761.1</td>\n",
       "      <td>MSTRG.21964</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SoxP2</td>\n",
       "      <td>Mihaylova et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SMEST051170001.1</td>\n",
       "      <td>SMESG000051170.1</td>\n",
       "      <td>MSTRG.13411</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SoxP3</td>\n",
       "      <td>Mihaylova et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MSTRG.15425.4</td>\n",
       "      <td>SMESG000050160.1</td>\n",
       "      <td>MSTRG.15425</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sp6-9</td>\n",
       "      <td>Mihaylova et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SMEST066384001.1</td>\n",
       "      <td>SMESG000066384.1</td>\n",
       "      <td>MSTRG.19740</td>\n",
       "      <td>NaN</td>\n",
       "      <td>zfp1</td>\n",
       "      <td>Mihaylova et al. (2018)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Transcript.ID           Rink.ID     Neiro.ID  Original.ID  \\\n",
       "0   SMEST065026002.1  SMESG000065026.1  MSTRG.17728          NaN   \n",
       "1       MSTRG.6709.1                 -   MSTRG.6709          NaN   \n",
       "2       MSTRG.5700.1                 -   MSTRG.5700          NaN   \n",
       "3   SMEST065670001.1  SMESG000065670.1  MSTRG.19539          NaN   \n",
       "4      MSTRG.12700.8  SMESG000039559.1  MSTRG.12700          NaN   \n",
       "5   SMEST027936002.1  SMESG000027936.1   MSTRG.9440          NaN   \n",
       "6       MSTRG.3856.7  SMESG000010063.1   MSTRG.3856          NaN   \n",
       "7      MSTRG.21964.4  SMESG000074761.1  MSTRG.21964          NaN   \n",
       "8   SMEST051170001.1  SMESG000051170.1  MSTRG.13411          NaN   \n",
       "9      MSTRG.15425.4  SMESG000050160.1  MSTRG.15425          NaN   \n",
       "10  SMEST066384001.1  SMESG000066384.1  MSTRG.19740          NaN   \n",
       "\n",
       "   Original.name                   Source  \n",
       "0      Cut-like1  Mihaylova et al. (2018)  \n",
       "1           ETS4  Mihaylova et al. (2018)  \n",
       "2           Elf5  Mihaylova et al. (2018)  \n",
       "3          FoxA1  Mihaylova et al. (2018)  \n",
       "4      Gata4/5/6  Mihaylova et al. (2018)  \n",
       "5         RREBP1  Mihaylova et al. (2018)  \n",
       "6          SoxP1  Mihaylova et al. (2018)  \n",
       "7          SoxP2  Mihaylova et al. (2018)  \n",
       "8          SoxP3  Mihaylova et al. (2018)  \n",
       "9          sp6-9  Mihaylova et al. (2018)  \n",
       "10          zfp1  Mihaylova et al. (2018)  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Mihaylova2018.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1cf0b49d",
   "metadata": {},
   "source": [
    "## 5.8 Cellular, ultrastructural and molecular analyses of epidermal cell development in the planarian S. mediterranea\n",
    "Cheng, L.C., Tu, K.C., Seidel, C.W., Robb, S.M., Guo, F. and Alvarado, A.S., 2018. Cellular, ultrastructural and molecular analyses of epidermal cell development in the planarian Schmidtea mediterranea. Developmental biology, 433(2), pp.357-373."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "cc0e7096",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cheng2018.tab\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "ls Cheng2018.tab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "9487f656",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "Cheng2018 = pd.read_csv(\"/hydra/TF_data/Cheng2018.tab\", sep=\"\\t\", header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c7757b79",
   "metadata": {},
   "outputs": [],
   "source": [
    "f = open(\"/hydra/TF_data/Cheng2018.fa\", \"a\")\n",
    "for i in range(len(Cheng2018)):\n",
    "    f.write(\">\"+Cheng2018.iloc[i,1]+\"\\n\")\n",
    "    f.write(Cheng2018.iloc[i,2]+\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71770509",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Cheng2018.fa -out blast_Cheng2018 -outfmt 6 -evalue 1.0 -word_size 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "4d2b4a87",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Cheng2018 = pd.read_csv(\"/hydra/TF_data/blast_Cheng2018\", sep=\"\\t\", header=None)\n",
    "blast_Cheng2018.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Cheng2018 = blast_Cheng2018.loc[blast_Cheng2018.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "d95a8794",
   "metadata": {},
   "outputs": [],
   "source": [
    "transcript_id = []\n",
    "gene_id_rink = []\n",
    "gene_id_neiro = []\n",
    "original_id = []\n",
    "original_name = []\n",
    "for i in range(len(blast_Cheng2018)):\n",
    "    t_id = blast_Cheng2018.iloc[i,1]\n",
    "    o_id = Cheng2018[Cheng2018.iloc[:,1] == blast_Cheng2018.iloc[i,0]].iloc[0,0]\n",
    "    transcript_id.append(t_id)\n",
    "    gene_id_rink.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,0])\n",
    "    gene_id_neiro.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,1])\n",
    "    original_id.append(o_id)\n",
    "    original_name.append(blast_Cheng2018.iloc[i,0])\n",
    "source = [\"Cheng et al. (2018)\"] * len(original_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "77f0b4ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Cheng2018 = pd.DataFrame({\"Transcript.ID\": transcript_id, \"Rink.ID\": gene_id_rink, \"Neiro.ID\": gene_id_neiro, \"Original.ID\": original_id, \"Original.name\": original_name, \"Source\": source})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e7edd088",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Cheng2018.to_csv(\"/hydra/TF_data/TF_Cheng2018.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "e8b16389",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SMEST030375001.1</td>\n",
       "      <td>SMESG000030375.1</td>\n",
       "      <td>MSTRG.9638</td>\n",
       "      <td>SMED30010086</td>\n",
       "      <td>barh2-1</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMEST069311002.1</td>\n",
       "      <td>SMESG000069311.1</td>\n",
       "      <td>MSTRG.19125</td>\n",
       "      <td>SMED30000485</td>\n",
       "      <td>blimp-1</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MSTRG.11197.3</td>\n",
       "      <td>SMESG000034285.1</td>\n",
       "      <td>MSTRG.11197</td>\n",
       "      <td>SMED30012332</td>\n",
       "      <td>egr-1</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MSTRG.23309.4</td>\n",
       "      <td>SMESG000079514.1</td>\n",
       "      <td>MSTRG.23309</td>\n",
       "      <td>SMED30011950</td>\n",
       "      <td>gata123</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MSTRG.20466.14</td>\n",
       "      <td>SMESG000073727.1</td>\n",
       "      <td>MSTRG.20466</td>\n",
       "      <td>SMED30011047</td>\n",
       "      <td>gli-1</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MSTRG.11748.1</td>\n",
       "      <td>SMESG000037556.1</td>\n",
       "      <td>MSTRG.11748</td>\n",
       "      <td>SMED30023969</td>\n",
       "      <td>lmo-1</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MSTRG.10142.2</td>\n",
       "      <td>-</td>\n",
       "      <td>MSTRG.10142</td>\n",
       "      <td>SMED30022043</td>\n",
       "      <td>osr</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SMEST078256002.1</td>\n",
       "      <td>SMESG000078256.1</td>\n",
       "      <td>MSTRG.21524</td>\n",
       "      <td>SMED30026364</td>\n",
       "      <td>p53</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SMEST004009001.1</td>\n",
       "      <td>SMESG000004009.1</td>\n",
       "      <td>MSTRG.2111</td>\n",
       "      <td>SMED30006353</td>\n",
       "      <td>pax-2/5/8</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MSTRG.17933.2</td>\n",
       "      <td>SMESG000060403.1</td>\n",
       "      <td>MSTRG.17933</td>\n",
       "      <td>SMED30022628</td>\n",
       "      <td>plac8</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MSTRG.16300.2</td>\n",
       "      <td>SMESG000054107.1</td>\n",
       "      <td>MSTRG.16300</td>\n",
       "      <td>SMED30021036</td>\n",
       "      <td>post-2a</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MSTRG.10796.3</td>\n",
       "      <td>SMESG000033048.1</td>\n",
       "      <td>MSTRG.10796</td>\n",
       "      <td>SMED30013674</td>\n",
       "      <td>post-2b</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MSTRG.7999.3</td>\n",
       "      <td>SMESG000022668.1</td>\n",
       "      <td>MSTRG.7999</td>\n",
       "      <td>SMED30000303</td>\n",
       "      <td>post-2c</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MSTRG.9883.2</td>\n",
       "      <td>-</td>\n",
       "      <td>MSTRG.9883</td>\n",
       "      <td>SMED30015704</td>\n",
       "      <td>post-2d</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MSTRG.16446.4</td>\n",
       "      <td>SMESG000054304.1</td>\n",
       "      <td>MSTRG.16446</td>\n",
       "      <td>SMED30015041</td>\n",
       "      <td>rar-b</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SMEST075452001.1</td>\n",
       "      <td>SMESG000075452.1</td>\n",
       "      <td>MSTRG.22163</td>\n",
       "      <td>SMED30032499</td>\n",
       "      <td>rest</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MSTRG.6831.5</td>\n",
       "      <td>SMESG000021468.1</td>\n",
       "      <td>MSTRG.6831</td>\n",
       "      <td>SMED30019046</td>\n",
       "      <td>sd2</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SMEST009894001.1</td>\n",
       "      <td>SMESG000009894.1</td>\n",
       "      <td>MSTRG.3785</td>\n",
       "      <td>SMED30010878</td>\n",
       "      <td>six3-1</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MSTRG.18230.2</td>\n",
       "      <td>SMESG000060967.1</td>\n",
       "      <td>MSTRG.18230</td>\n",
       "      <td>SMED30000938</td>\n",
       "      <td>six6</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SMEST051170001.1</td>\n",
       "      <td>SMESG000051170.1</td>\n",
       "      <td>MSTRG.13411</td>\n",
       "      <td>SMED30008505</td>\n",
       "      <td>soxP-3</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>SMEST023227001.1</td>\n",
       "      <td>SMESG000023227.1</td>\n",
       "      <td>MSTRG.7606</td>\n",
       "      <td>SMED30033839</td>\n",
       "      <td>sp5</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>MSTRG.13829.2</td>\n",
       "      <td>SMESG000044395.1</td>\n",
       "      <td>MSTRG.13829</td>\n",
       "      <td>SMED30006464</td>\n",
       "      <td>tcf-15</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>SMEST066384001.1</td>\n",
       "      <td>SMESG000066384.1</td>\n",
       "      <td>MSTRG.19740</td>\n",
       "      <td>SMED30023708</td>\n",
       "      <td>zfp-1</td>\n",
       "      <td>Cheng et al. (2018)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Transcript.ID           Rink.ID     Neiro.ID   Original.ID  \\\n",
       "0   SMEST030375001.1  SMESG000030375.1   MSTRG.9638  SMED30010086   \n",
       "1   SMEST069311002.1  SMESG000069311.1  MSTRG.19125  SMED30000485   \n",
       "2      MSTRG.11197.3  SMESG000034285.1  MSTRG.11197  SMED30012332   \n",
       "3      MSTRG.23309.4  SMESG000079514.1  MSTRG.23309  SMED30011950   \n",
       "4     MSTRG.20466.14  SMESG000073727.1  MSTRG.20466  SMED30011047   \n",
       "5      MSTRG.11748.1  SMESG000037556.1  MSTRG.11748  SMED30023969   \n",
       "6      MSTRG.10142.2                 -  MSTRG.10142  SMED30022043   \n",
       "7   SMEST078256002.1  SMESG000078256.1  MSTRG.21524  SMED30026364   \n",
       "8   SMEST004009001.1  SMESG000004009.1   MSTRG.2111  SMED30006353   \n",
       "9      MSTRG.17933.2  SMESG000060403.1  MSTRG.17933  SMED30022628   \n",
       "10     MSTRG.16300.2  SMESG000054107.1  MSTRG.16300  SMED30021036   \n",
       "11     MSTRG.10796.3  SMESG000033048.1  MSTRG.10796  SMED30013674   \n",
       "12      MSTRG.7999.3  SMESG000022668.1   MSTRG.7999  SMED30000303   \n",
       "13      MSTRG.9883.2                 -   MSTRG.9883  SMED30015704   \n",
       "14     MSTRG.16446.4  SMESG000054304.1  MSTRG.16446  SMED30015041   \n",
       "15  SMEST075452001.1  SMESG000075452.1  MSTRG.22163  SMED30032499   \n",
       "16      MSTRG.6831.5  SMESG000021468.1   MSTRG.6831  SMED30019046   \n",
       "17  SMEST009894001.1  SMESG000009894.1   MSTRG.3785  SMED30010878   \n",
       "18     MSTRG.18230.2  SMESG000060967.1  MSTRG.18230  SMED30000938   \n",
       "19  SMEST051170001.1  SMESG000051170.1  MSTRG.13411  SMED30008505   \n",
       "20  SMEST023227001.1  SMESG000023227.1   MSTRG.7606  SMED30033839   \n",
       "21     MSTRG.13829.2  SMESG000044395.1  MSTRG.13829  SMED30006464   \n",
       "22  SMEST066384001.1  SMESG000066384.1  MSTRG.19740  SMED30023708   \n",
       "\n",
       "   Original.name               Source  \n",
       "0        barh2-1  Cheng et al. (2018)  \n",
       "1        blimp-1  Cheng et al. (2018)  \n",
       "2          egr-1  Cheng et al. (2018)  \n",
       "3        gata123  Cheng et al. (2018)  \n",
       "4          gli-1  Cheng et al. (2018)  \n",
       "5          lmo-1  Cheng et al. (2018)  \n",
       "6            osr  Cheng et al. (2018)  \n",
       "7            p53  Cheng et al. (2018)  \n",
       "8      pax-2/5/8  Cheng et al. (2018)  \n",
       "9          plac8  Cheng et al. (2018)  \n",
       "10       post-2a  Cheng et al. (2018)  \n",
       "11       post-2b  Cheng et al. (2018)  \n",
       "12       post-2c  Cheng et al. (2018)  \n",
       "13       post-2d  Cheng et al. (2018)  \n",
       "14         rar-b  Cheng et al. (2018)  \n",
       "15          rest  Cheng et al. (2018)  \n",
       "16           sd2  Cheng et al. (2018)  \n",
       "17        six3-1  Cheng et al. (2018)  \n",
       "18          six6  Cheng et al. (2018)  \n",
       "19        soxP-3  Cheng et al. (2018)  \n",
       "20           sp5  Cheng et al. (2018)  \n",
       "21        tcf-15  Cheng et al. (2018)  \n",
       "22         zfp-1  Cheng et al. (2018)  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Cheng2018.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d05a1a3c",
   "metadata": {},
   "source": [
    "## 5.9 Cell type transcriptome atlas for the planarian Schmidtea mediterranea\n",
    "Fincher, C.T., Wurtzel, O., de Hoog, T., Kravarik, K.M. and Reddien, P.W., 2018. Cell type transcriptome atlas for the planarian Schmidtea mediterranea. Science, 360(6391).\n",
    "\n",
    "* prox-1\tdd_Smed_v4_13772_0_1\n",
    "* zfp-1\tdd_Smed_v4_8720_0_1\n",
    "* POU2/3\tdd_Smed_v4_8234_0_1\n",
    "* hnf-4\tdd_Smed_v4_1694_0_1\n",
    "* gata4/5/6\tdd_Smed_v4_4075_0_1\n",
    "* ETS1 (dd2092)\tdd_Smed_v4_2092_0_1\n",
    "* FOXF1 (dd6910)\tdd_Smed_v4_6910_0_1\n",
    "* foxA-1\tdd_Smed_v4_10718_0_1\n",
    "* GCM2 (dd7752)\tdd_Smed_v4_7752_0_1\n",
    "* nkx2-like\tdd_Smed_v4_13898_0_1\n",
    "* pax6A\tdd_Smed_v4_17726_0_1\n",
    "* nkx6-like\tdd_Smed_v4_11198_0_1\n",
    "* lox5A\tdd_Smed_v4_16227_0_1\n",
    "* Post-2c\tdd_Smed_v4_12467_0_1\n",
    "* hox4b\tdd_Smed_v4_2442_0_1\n",
    "* sp-5\tdd_Smed_v4_7824_0_1\n",
    "* PTF1A (dd6869)\tdd_Smed_v4_6869_0_1\n",
    "* ASCL4 (dd1854)\tdd_Smed_v4_1854_0_1\n",
    "* ascl-2\tdd_Smed_v4_14753_0_1\n",
    "* zic-1\tdd_Smed_v4_22585_0_1\n",
    "* foxD\tdd_Smed_v4_23249_0_1\n",
    "* zic-2\tdd_Smed_v4_15531_0_1\n",
    "* runt-1\tdd_Smed_v4_16222_0_1\n",
    "* ZNF596 (dd10911)\tdd_Smed_v4_10911_0_1\n",
    "* nkx2.2\tdd_Smed_v4_2716_0_1\n",
    "* ovo\tdd_Smed_v4_48430_0_1\n",
    "* eya\tdd_Smed_v4_11372_0_1\n",
    "* six-1/2\tdd_Smed_v4_15436_0_1\n",
    "* prep\tdd_Smed_v4_8606_0_1\n",
    "* egf-6\tdd_Smed_v4_6222_0_1 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "b22887a6",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "echo \"dd_Smed_v6_13772_0_1\" > Fincher2018.list\n",
    "echo \"dd_Smed_v6_8720_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_8234_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_1694_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_4075_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_2092_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_6910_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_10718_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_7752_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_13898_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_17726_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_11198_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_16227_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_12467_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_2442_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_7824_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_6869_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_1854_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_14753_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_22585_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_23249_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_15531_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_16222_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_10911_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_2716_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_48430_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_11372_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_15436_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_8606_0_1\" >> Fincher2018.list\n",
    "echo \"dd_Smed_v6_6222_0_1\" >> Fincher2018.list "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "8ebb6a7a",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "seqtk subseq /hydra/sexual_genome_annotation_files/sexual_genome_annotation/dd_Smed_v6.pcf.contigs.fasta Fincher2018.list > Fincher2018.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "3577c59f",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Fincher2018.fa -out blast_Fincher2018 -outfmt 6 &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "a64f587f",
   "metadata": {},
   "outputs": [],
   "source": [
    "Fincher2018 = pd.DataFrame({\"ID\": [\"dd_Smed_v6_13772_0_1\", \"dd_Smed_v6_8720_0_1\", \"dd_Smed_v6_8234_0_1\", \"dd_Smed_v6_1694_0_1\", \"dd_Smed_v6_4075_0_1\", \n",
    "                                 \"dd_Smed_v6_2092_0_1\", \"dd_Smed_v6_6910_0_1\", \"dd_Smed_v6_10718_0_1\", \"dd_Smed_v6_7752_0_1\",\n",
    "                              \"dd_Smed_v6_13898_0_1\", \"dd_Smed_v6_17726_0_1\", \"dd_Smed_v6_11198_0_1\", \"dd_Smed_v6_16227_0_1\",\n",
    "                              \"dd_Smed_v6_12467_0_1\", \"dd_Smed_v6_2442_0_1\", \"dd_Smed_v6_7824_0_1\", \"dd_Smed_v6_6869_0_1\",\n",
    "                              \"dd_Smed_v6_1854_0_1\", \"dd_Smed_v6_14753_0_1\", \"dd_Smed_v6_22585_0_1\", \"dd_Smed_v6_23249_0_1\",\n",
    "                              \"dd_Smed_v6_15531_0_1\", \"dd_Smed_v6_16222_0_1\", \"dd_Smed_v6_10911_0_1\", \"dd_Smed_v6_2716_0_1\", \n",
    "                              \"dd_Smed_v6_48430_0_1\", \"dd_Smed_v6_11372_0_1\", \"dd_Smed_v6_15436_0_1\", \"dd_Smed_v6_8606_0_1\", \"dd_Smed_v6_6222_0_1\"], \n",
    "                           \"Name\": [\"prox-1\", \"zfp-1\", \"POU2/3\", \"hnf-4\", \"gata4/5/6\", \"ETS1\", \"FOXF1\", \n",
    "                                    \"foxA-1\", \"GCM2\", \"nkx2-like\", \"pax6A\", \"nkx6-like\", \"lox5A\", \"Post-2c\",\n",
    "                                   \"hox4b\", \"sp-5\", \"PTF1A\", \"ASCL4\", \"ascl-2\", \"zic-1\", \"foxD\", \"zic-2\", \"runt-1\",\n",
    "                                   \"ZNF596\", \"nkx2.2\", \"ovo\", \"eya\", \"six-1/2\", \"prep\", \"egf-6\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "e1945737",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Fincher2018 = pd.read_csv(\"/hydra/TF_data/blast_Fincher2018\", sep=\"\\t\", header=None)\n",
    "blast_Fincher2018.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Fincher2018 = blast_Fincher2018.loc[blast_Fincher2018.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "cb18b615",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Fincher2018= tf_table(blast_Fincher2018, \"Fincher et al. (2018)\", Fincher2018)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "800a8317",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Fincher2018.to_csv(\"/hydra/TF_data/TF_Fincher2018.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "84901a4f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MSTRG.19539.1</td>\n",
       "      <td>SMESG000065669.1</td>\n",
       "      <td>MSTRG.19539</td>\n",
       "      <td>NaN</td>\n",
       "      <td>foxA-1</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMEST022366002.1</td>\n",
       "      <td>SMESG000022366.1</td>\n",
       "      <td>MSTRG.7896</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ZNF596</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SMEST002774001.1</td>\n",
       "      <td>SMESG000002774.1</td>\n",
       "      <td>MSTRG.764</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nkx6-like</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SMEST001561001.1</td>\n",
       "      <td>SMESG000001561.1</td>\n",
       "      <td>MSTRG.1422</td>\n",
       "      <td>NaN</td>\n",
       "      <td>eya</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SMEST022668001.1</td>\n",
       "      <td>SMESG000022668.1</td>\n",
       "      <td>MSTRG.7999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Post-2c</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SMEST068287001.1</td>\n",
       "      <td>SMESG000068287.1</td>\n",
       "      <td>MSTRG.20222</td>\n",
       "      <td>NaN</td>\n",
       "      <td>prox-1</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SMEST081885001.1</td>\n",
       "      <td>SMESG000081885.1</td>\n",
       "      <td>MSTRG.22888</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nkx2-like</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MSTRG.10663.1</td>\n",
       "      <td>SMESG000032619.1</td>\n",
       "      <td>MSTRG.10663</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ascl-2</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SMEST001687001.1</td>\n",
       "      <td>SMESG000001687.1</td>\n",
       "      <td>MSTRG.1402</td>\n",
       "      <td>NaN</td>\n",
       "      <td>six-1/2</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SMEST052644001.1</td>\n",
       "      <td>SMESG000052644.1</td>\n",
       "      <td>MSTRG.16036</td>\n",
       "      <td>NaN</td>\n",
       "      <td>zic-2</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SMEST014779001.1</td>\n",
       "      <td>SMESG000014779.1</td>\n",
       "      <td>MSTRG.4469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>runt-1</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SMEST022667001.1</td>\n",
       "      <td>SMESG000022667.1</td>\n",
       "      <td>MSTRG.7998</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lox5A</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SMEST014588001.1</td>\n",
       "      <td>SMESG000014588.1</td>\n",
       "      <td>MSTRG.4403</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hnf-4</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SMEST039056001.1</td>\n",
       "      <td>SMESG000039056.1</td>\n",
       "      <td>MSTRG.12215</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pax6A</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MSTRG.27.1</td>\n",
       "      <td>SMESG000026861.1</td>\n",
       "      <td>MSTRG.27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASCL4</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MSTRG.19528.3</td>\n",
       "      <td>SMESG000065612.1</td>\n",
       "      <td>MSTRG.19528</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ETS1</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SMEST022958001.1</td>\n",
       "      <td>SMESG000022958.1</td>\n",
       "      <td>MSTRG.7487</td>\n",
       "      <td>NaN</td>\n",
       "      <td>zic-1</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SMEST077075001.1</td>\n",
       "      <td>SMESG000077075.1</td>\n",
       "      <td>MSTRG.22528</td>\n",
       "      <td>NaN</td>\n",
       "      <td>foxD</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MSTRG.13843.2</td>\n",
       "      <td>SMESG000044469.1</td>\n",
       "      <td>MSTRG.13843</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hox4b</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>MSTRG.23238.2</td>\n",
       "      <td>SMESG000079512.1</td>\n",
       "      <td>MSTRG.23238</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nkx2.2</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>MSTRG.12700.6</td>\n",
       "      <td>SMESG000039559.1</td>\n",
       "      <td>MSTRG.12700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gata4/5/6</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SMEST081129002.1</td>\n",
       "      <td>SMESG000081129.1</td>\n",
       "      <td>MSTRG.22732</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ovo</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>SMEST029319001.1</td>\n",
       "      <td>SMESG000029319.1</td>\n",
       "      <td>MSTRG.9845</td>\n",
       "      <td>NaN</td>\n",
       "      <td>egf-6</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SMEST040081001.1</td>\n",
       "      <td>SMESG000040081.1</td>\n",
       "      <td>MSTRG.12572</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTF1A</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>SMEST066497001.1</td>\n",
       "      <td>SMESG000066497.1</td>\n",
       "      <td>MSTRG.19765</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FOXF1</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>MSTRG.16974.3</td>\n",
       "      <td>SMESG000056241.1</td>\n",
       "      <td>MSTRG.16974</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCM2</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>SMEST023227001.1</td>\n",
       "      <td>SMESG000023227.1</td>\n",
       "      <td>MSTRG.7606</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sp-5</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>SMEST076173001.1</td>\n",
       "      <td>SMESG000076173.1</td>\n",
       "      <td>MSTRG.22352</td>\n",
       "      <td>NaN</td>\n",
       "      <td>POU2/3</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>MSTRG.4755.1</td>\n",
       "      <td>SMESG000013307.1</td>\n",
       "      <td>MSTRG.4755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>prep</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>MSTRG.19740.1</td>\n",
       "      <td>SMESG000066384.1</td>\n",
       "      <td>MSTRG.19740</td>\n",
       "      <td>NaN</td>\n",
       "      <td>zfp-1</td>\n",
       "      <td>Fincher et al. (2018)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Transcript.ID           Rink.ID     Neiro.ID  Original.ID  \\\n",
       "0      MSTRG.19539.1  SMESG000065669.1  MSTRG.19539          NaN   \n",
       "1   SMEST022366002.1  SMESG000022366.1   MSTRG.7896          NaN   \n",
       "2   SMEST002774001.1  SMESG000002774.1    MSTRG.764          NaN   \n",
       "3   SMEST001561001.1  SMESG000001561.1   MSTRG.1422          NaN   \n",
       "4   SMEST022668001.1  SMESG000022668.1   MSTRG.7999          NaN   \n",
       "5   SMEST068287001.1  SMESG000068287.1  MSTRG.20222          NaN   \n",
       "6   SMEST081885001.1  SMESG000081885.1  MSTRG.22888          NaN   \n",
       "7      MSTRG.10663.1  SMESG000032619.1  MSTRG.10663          NaN   \n",
       "8   SMEST001687001.1  SMESG000001687.1   MSTRG.1402          NaN   \n",
       "9   SMEST052644001.1  SMESG000052644.1  MSTRG.16036          NaN   \n",
       "10  SMEST014779001.1  SMESG000014779.1   MSTRG.4469          NaN   \n",
       "11  SMEST022667001.1  SMESG000022667.1   MSTRG.7998          NaN   \n",
       "12  SMEST014588001.1  SMESG000014588.1   MSTRG.4403          NaN   \n",
       "13  SMEST039056001.1  SMESG000039056.1  MSTRG.12215          NaN   \n",
       "14        MSTRG.27.1  SMESG000026861.1     MSTRG.27          NaN   \n",
       "15     MSTRG.19528.3  SMESG000065612.1  MSTRG.19528          NaN   \n",
       "16  SMEST022958001.1  SMESG000022958.1   MSTRG.7487          NaN   \n",
       "17  SMEST077075001.1  SMESG000077075.1  MSTRG.22528          NaN   \n",
       "18     MSTRG.13843.2  SMESG000044469.1  MSTRG.13843          NaN   \n",
       "19     MSTRG.23238.2  SMESG000079512.1  MSTRG.23238          NaN   \n",
       "20     MSTRG.12700.6  SMESG000039559.1  MSTRG.12700          NaN   \n",
       "21  SMEST081129002.1  SMESG000081129.1  MSTRG.22732          NaN   \n",
       "22  SMEST029319001.1  SMESG000029319.1   MSTRG.9845          NaN   \n",
       "23  SMEST040081001.1  SMESG000040081.1  MSTRG.12572          NaN   \n",
       "24  SMEST066497001.1  SMESG000066497.1  MSTRG.19765          NaN   \n",
       "25     MSTRG.16974.3  SMESG000056241.1  MSTRG.16974          NaN   \n",
       "26  SMEST023227001.1  SMESG000023227.1   MSTRG.7606          NaN   \n",
       "27  SMEST076173001.1  SMESG000076173.1  MSTRG.22352          NaN   \n",
       "28      MSTRG.4755.1  SMESG000013307.1   MSTRG.4755          NaN   \n",
       "29     MSTRG.19740.1  SMESG000066384.1  MSTRG.19740          NaN   \n",
       "\n",
       "   Original.name                 Source  \n",
       "0         foxA-1  Fincher et al. (2018)  \n",
       "1         ZNF596  Fincher et al. (2018)  \n",
       "2      nkx6-like  Fincher et al. (2018)  \n",
       "3            eya  Fincher et al. (2018)  \n",
       "4        Post-2c  Fincher et al. (2018)  \n",
       "5         prox-1  Fincher et al. (2018)  \n",
       "6      nkx2-like  Fincher et al. (2018)  \n",
       "7         ascl-2  Fincher et al. (2018)  \n",
       "8        six-1/2  Fincher et al. (2018)  \n",
       "9          zic-2  Fincher et al. (2018)  \n",
       "10        runt-1  Fincher et al. (2018)  \n",
       "11         lox5A  Fincher et al. (2018)  \n",
       "12         hnf-4  Fincher et al. (2018)  \n",
       "13         pax6A  Fincher et al. (2018)  \n",
       "14         ASCL4  Fincher et al. (2018)  \n",
       "15          ETS1  Fincher et al. (2018)  \n",
       "16         zic-1  Fincher et al. (2018)  \n",
       "17          foxD  Fincher et al. (2018)  \n",
       "18         hox4b  Fincher et al. (2018)  \n",
       "19        nkx2.2  Fincher et al. (2018)  \n",
       "20     gata4/5/6  Fincher et al. (2018)  \n",
       "21           ovo  Fincher et al. (2018)  \n",
       "22         egf-6  Fincher et al. (2018)  \n",
       "23         PTF1A  Fincher et al. (2018)  \n",
       "24         FOXF1  Fincher et al. (2018)  \n",
       "25          GCM2  Fincher et al. (2018)  \n",
       "26          sp-5  Fincher et al. (2018)  \n",
       "27        POU2/3  Fincher et al. (2018)  \n",
       "28          prep  Fincher et al. (2018)  \n",
       "29         zfp-1  Fincher et al. (2018)  "
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Fincher2018.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fbf312fa",
   "metadata": {},
   "source": [
    "# 6. 2019"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9bc916ca",
   "metadata": {},
   "source": [
    "## 6.1 Neoblast-enriched zinc finger protein FIR1 triggers local proliferation during planarian regeneration\n",
    "Han, X.S., Wang, C., Guo, F.H., Huang, S., Qin, Y.W., Zhao, X.X. and Jing, Q., 2019. Neoblast-enriched zinc finger protein FIR1 triggers local proliferation during planarian regeneration. Protein & cell, 10(1), pp.43-59.\n",
    "\n",
    "The mechanism by which stem cells initiate proliferation during regeneration in vivo is unclear. We screened 46 transcripts showing potential function in the regulation of local stem cell proliferation following 48 h regeneration. By analyzing the regeneration defects and the mitotic activity of animals under administration of RNAi, we identified factor for initiiating regeneration 1 (Fir 1) required for local proliferation. Our findings reveal that Fir1, enriched in neoblasts, promotes planarian regeneration in any tissue-missing context. We demonstrate that DIS3 like 3'-5' exoribonuclease 2 (Dis3l2) is required for the Fir1 phenotype. \n",
    "\n",
    "A recent study has revealed that allkinds of injury activate a common wound-response transcriptional progress, and neoblasts express most wound.induced genes (Wurtzel et al., 2015). Neoblasts play critical roles in response to early injury. \n",
    "\n",
    "First, we searched published papers for reports on neoblast regulators and phenotypic transcription factors screened by RNA interference (RNAi) and aimed to identify them in our lab. Considering the strength of regenerative phenotype upon RNAi and the expression pattern, we finally chose 46 genes, which were reported to strongly promote planarian regenration and be enriched in neoblasts. \n",
    "\n",
    "In the next phase of our study, we examined regeneration phenotypes of these genes after two rounds of dsRNA feeding and found that trunk pieces showed sever regeneraqtion defects following each of 23 genes. Interestingly, we identified a new gene, Fir1, knockdown of whoch caused mitoses to diminish significantly at the wound sites, without affecting remaining tail pieces. \n",
    "\n",
    "Fir1 (RNAi) animals could not regnerate blastemas after amputation. Fir1 mRNA is predominantly expressed in neoblasts. We performed cell FISH on \"X1\" cells sorted using FACS and found that 97.45% of X1 cells expressed Fir1 mRNA which was consitent with FIr1 and smedwi-1 co-localization data. \n",
    "\n",
    "We hypothesized that Fir1 promotes regeneration by controlling regeneratove proliferation. It is reported that many neoblast regulators could regulate planarian regeneration, amd we attempted to establish whether some known genes were related to the Fir1 function. Genes that downregulated significantly in both part 1 and part 2 belonged to Cluster two, which contained h2a, egr-1, smedwi-3, zfp-1, and inx-11. Although these genes were Fir1 downstream genes, they might not participate in regulating regenerative proliferation. Genes that downregulated significantly only in part 1 constituted Cluster three, which included soxP-2, bruli, cip29, rbap48, khd-1, cyclinB, rrm2, smedwi-2, soxP-1, rbbp4-1, setd8-1, sz12-1, and h2b. \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3ff1180f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">MF411143.1 Schmidtea mediterranea factor for initiating regeneration 1 (Fir1) mRNA, complete cds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "grep \">\" Han2019.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0aaa8d6b",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Han2019.fa -out blast_Han2019 -outfmt 6 &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "db00e651",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Han2019 = pd.read_csv(\"/hydra/TF_data/blast_Han2019\", sep=\"\\t\", header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "030aac5c",
   "metadata": {},
   "outputs": [],
   "source": [
    "blast_Han2019.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Han2019 = blast_Han2019.loc[blast_Han2019.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d2b64fd0",
   "metadata": {},
   "outputs": [],
   "source": [
    "Han2019 = pd.DataFrame({\"ID\": [\"MF411143.1\"], \"Name\": [\"Fir1\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "383b26fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Han2019= tf_table(blast_Han2019, \"Han et al. (2019)\", Han2019)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "517a78e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Han2019.to_csv(\"/hydra/TF_data/TF_Han2019.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "68fd3e41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>SMEST014324002.1</td>\n",
       "      <td>SMESG000014324.1</td>\n",
       "      <td>MSTRG.4330</td>\n",
       "      <td>MF411143</td>\n",
       "      <td>Fir1</td>\n",
       "      <td>Han et al. (2019)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0     Transcript.ID           Rink.ID    Neiro.ID Original.ID  \\\n",
       "0           0  SMEST014324002.1  SMESG000014324.1  MSTRG.4330    MF411143   \n",
       "\n",
       "  Original.name             Source  \n",
       "0          Fir1  Han et al. (2019)  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Han2019.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09ef725a",
   "metadata": {},
   "source": [
    "## 6.2 A small set of conserved genes, including sp5 and Hox, are activated by Wnt signaling in the posterior of planarians and acoels\n",
    "Tewari, A.G., Owen, J.H., Petersen, C.P., Wagner, D.E. and Reddien, P.W., 2019. A small set of conserved genes, including sp5 and Hox, are activated by Wnt signaling in the posterior of planarians and acoels. PLoS genetics, 15(10), p.e1008401."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "499eedd1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Tewari2019.xlsx\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "ls Tewari2019.xlsx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "6d7345ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "names = [\"lox5a\", \"Post-2d\", \"Post-2c\", \"hox4b\", \"sp5\", \"Tbx2/3c\", \"Tbx2/3b\", \"Tbx20\", \"Tbx2/3a\", \"Tbx1/10\"]\n",
    "Tewari2019xlsx = pd.read_excel(\"/hydra/TF_data/Tewari2019.xlsx\")\n",
    "Tewari2019xlsx = Tewari2019xlsx[(Tewari2019xlsx[\"Species\"] == \"S.mediterranea\") & (Tewari2019xlsx[\"Gene Name\"].isin(names))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "002f7ffc",
   "metadata": {},
   "outputs": [],
   "source": [
    "Tewari2019 = pd.DataFrame({\"ID\": [\"dd_Smed_v6_16227_0_1\", \"dd_Smed_v6_12467_0_1\", \"EU082823.1\", \"EU082824.1\", \"KP003818.1\", \"MN275808.1\", \"MN275809.1\", \"MN275810.1\", \"MN275811.1\", \"MN275812.1\"], \"Name\": [\"lox5a\", \"Post-2c\", \"Post-2d\", \"hox4b\", \"sp5\", \"T-box 2/3c\", \"T-box 2/3b\", \"T-box 20\", \"T-box 2/3a\", \"T-box 1/10\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "b09c2077",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">dd_Smed_v6_16227_0_1\n",
      ">dd_Smed_v6_12467_0_1\n",
      ">EU082823.1 Schmidtea mediterranea AbdBa Hox protein (abdba) mRNA, partial cds\n",
      ">EU082824.1 Schmidtea mediterranea HoxD-like protein (hoxD) mRNA, complete cds\n",
      ">KP003818.1 Schmidtea mediterranea Sp5 mRNA, partial cds\n",
      ">MN275808.1 Schmidtea mediterranea T-box 2/3c protein mRNA, complete cds\n",
      ">MN275809.1 Schmidtea mediterranea T-box 2/3b protein mRNA, complete cds\n",
      ">MN275810.1 Schmidtea mediterranea T-box 20 protein mRNA, complete cds\n",
      ">MN275811.1 Schmidtea mediterranea T-box 2/3a protein mRNA, complete cds\n",
      ">MN275812.1 Schmidtea mediterranea T-box 1/10 protein mRNA, complete cds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "grep \">\" Tewari2019.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "32e4c055",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Tewari2019.fa -out blast_Tewari2019 -outfmt 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "ca6f7f55",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Tewari2019 = pd.read_csv(\"/hydra/TF_data/blast_Tewari2019\", sep=\"\\t\", header=None)\n",
    "blast_Tewari2019.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Tewari2019 = blast_Tewari2019.loc[blast_Tewari2019.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "4ee4c9ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "transcript_id = []\n",
    "gene_id_rink = []\n",
    "gene_id_neiro = []\n",
    "original_id = []\n",
    "original_name = []\n",
    "for i in range(len(blast_Tewari2019)):\n",
    "    t_id = blast_Tewari2019.iloc[i,1]\n",
    "    o_id = blast_Tewari2019.iloc[i,0]\n",
    "    transcript_id.append(t_id)\n",
    "    gene_id_rink.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,0])\n",
    "    gene_id_neiro.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,1])\n",
    "    original_id.append(o_id.rpartition(\".\")[0])\n",
    "    original_name.append(Tewari2019[Tewari2019[\"ID\"] == o_id].iloc[0,1])\n",
    "source = [\"Tewari et al. (2019)\"] * len(original_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "82c9408c",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Tewari2019 = pd.DataFrame({\"Transcript.ID\": transcript_id, \"Rink.ID\": gene_id_rink, \"Neiro.ID\": gene_id_neiro, \"Original.ID\": original_id, \"Original.name\": original_name, \"Source\": source})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e808cd60",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Tewari2019.to_csv(\"/hydra/TF_data/TF_Tewari2019.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "7ea2e4d7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MSTRG.9883.2</td>\n",
       "      <td>-</td>\n",
       "      <td>MSTRG.9883</td>\n",
       "      <td>EU082823</td>\n",
       "      <td>Post-2d</td>\n",
       "      <td>Tewari et al. (2019)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MSTRG.13843.2</td>\n",
       "      <td>SMESG000044469.1</td>\n",
       "      <td>MSTRG.13843</td>\n",
       "      <td>EU082824</td>\n",
       "      <td>hox4b</td>\n",
       "      <td>Tewari et al. (2019)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SMEST023227001.1</td>\n",
       "      <td>SMESG000023227.1</td>\n",
       "      <td>MSTRG.7606</td>\n",
       "      <td>KP003818</td>\n",
       "      <td>sp5</td>\n",
       "      <td>Tewari et al. (2019)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SMEST061015001.1</td>\n",
       "      <td>SMESG000061015.1</td>\n",
       "      <td>MSTRG.18243</td>\n",
       "      <td>MN275808</td>\n",
       "      <td>T-box 2/3c</td>\n",
       "      <td>Tewari et al. (2019)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SMEST023378001.1</td>\n",
       "      <td>SMESG000023378.1</td>\n",
       "      <td>MSTRG.8305</td>\n",
       "      <td>MN275809</td>\n",
       "      <td>T-box 2/3b</td>\n",
       "      <td>Tewari et al. (2019)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SMEST071435001.1</td>\n",
       "      <td>SMESG000071435.1</td>\n",
       "      <td>MSTRG.21005</td>\n",
       "      <td>MN275810</td>\n",
       "      <td>T-box 20</td>\n",
       "      <td>Tewari et al. (2019)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SMEST080056002.1</td>\n",
       "      <td>SMESG000080056.1</td>\n",
       "      <td>MSTRG.23477</td>\n",
       "      <td>MN275811</td>\n",
       "      <td>T-box 2/3a</td>\n",
       "      <td>Tewari et al. (2019)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SMEST064684001.1</td>\n",
       "      <td>SMESG000064684.1</td>\n",
       "      <td>MSTRG.17627</td>\n",
       "      <td>MN275812</td>\n",
       "      <td>T-box 1/10</td>\n",
       "      <td>Tewari et al. (2019)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SMEST022668001.1</td>\n",
       "      <td>SMESG000022668.1</td>\n",
       "      <td>MSTRG.7999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Post-2c</td>\n",
       "      <td>Tewari et al. (2019)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SMEST022667001.1</td>\n",
       "      <td>SMESG000022667.1</td>\n",
       "      <td>MSTRG.7998</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lox5a</td>\n",
       "      <td>Tewari et al. (2019)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Transcript.ID           Rink.ID     Neiro.ID Original.ID Original.name  \\\n",
       "0      MSTRG.9883.2                 -   MSTRG.9883    EU082823       Post-2d   \n",
       "1     MSTRG.13843.2  SMESG000044469.1  MSTRG.13843    EU082824         hox4b   \n",
       "2  SMEST023227001.1  SMESG000023227.1   MSTRG.7606    KP003818           sp5   \n",
       "3  SMEST061015001.1  SMESG000061015.1  MSTRG.18243    MN275808    T-box 2/3c   \n",
       "4  SMEST023378001.1  SMESG000023378.1   MSTRG.8305    MN275809    T-box 2/3b   \n",
       "5  SMEST071435001.1  SMESG000071435.1  MSTRG.21005    MN275810      T-box 20   \n",
       "6  SMEST080056002.1  SMESG000080056.1  MSTRG.23477    MN275811    T-box 2/3a   \n",
       "7  SMEST064684001.1  SMESG000064684.1  MSTRG.17627    MN275812    T-box 1/10   \n",
       "8  SMEST022668001.1  SMESG000022668.1   MSTRG.7999         NaN       Post-2c   \n",
       "9  SMEST022667001.1  SMESG000022667.1   MSTRG.7998         NaN         lox5a   \n",
       "\n",
       "                 Source  \n",
       "0  Tewari et al. (2019)  \n",
       "1  Tewari et al. (2019)  \n",
       "2  Tewari et al. (2019)  \n",
       "3  Tewari et al. (2019)  \n",
       "4  Tewari et al. (2019)  \n",
       "5  Tewari et al. (2019)  \n",
       "6  Tewari et al. (2019)  \n",
       "7  Tewari et al. (2019)  \n",
       "8  Tewari et al. (2019)  \n",
       "9  Tewari et al. (2019)  "
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Tewari2019.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9419a048",
   "metadata": {},
   "source": [
    "## 6.3 Nuclear receptor NR4A is required for patterning at the ends of the planarian anterior-posterior axis\n",
    "Li, D.J., McMann, C.L. and Reddien, P.W., 2019. Nuclear receptor NR4A is required for patterning at the ends of the planarian anterior-posterior axis. Elife, 8, p.e42015."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "675f95ce",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">A.1\n",
      ">A.2\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "grep \">\" Li2019.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "44a08646",
   "metadata": {},
   "outputs": [],
   "source": [
    "Li2019 = pd.DataFrame({\"ID\": [\"A.1\", \"A.2\"], \"Name\": [\"nr4a\", \"nr4a\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "6411665e",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Li2019.fa -out blast_Li2019 -outfmt 6 -evalue 1.0 -word_size 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "0d701eed",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Li2019 = pd.read_csv(\"/hydra/TF_data/blast_Li2019\", sep=\"\\t\", header=None)\n",
    "blast_Li2019.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Li2019 = blast_Li2019.loc[blast_Li2019.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "58b164d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Li2019 = tf_table(blast_Li2019, \"Li et al. (2019)\", Li2019)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "cf3da7d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Li2019.to_csv(\"/hydra/TF_data/TF_Li2019.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "a02f5771",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>MSTRG.18823.4</td>\n",
       "      <td>SMESG000063104.1</td>\n",
       "      <td>MSTRG.18823</td>\n",
       "      <td>A</td>\n",
       "      <td>nr4a</td>\n",
       "      <td>Li et al. (2019)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0  Transcript.ID           Rink.ID     Neiro.ID Original.ID  \\\n",
       "0           0  MSTRG.18823.4  SMESG000063104.1  MSTRG.18823           A   \n",
       "\n",
       "  Original.name            Source  \n",
       "0          nr4a  Li et al. (2019)  "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Li2019.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0f8244b4",
   "metadata": {},
   "source": [
    "# 7. 2020"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4820a8bd",
   "metadata": {},
   "source": [
    "## 7.1 Muscle and neuronal guidepost-like cells facilitate planarian visual system regeneration\n",
    "Scimone, M.L., Atabay, K.D., Fincher, C.T., Bonneau, A.R., Li, D.J. and Reddien, P.W., 2020. Muscle and neuronal guidepost-like cells facilitate planarian visual system regeneration. Science, 368(6498)."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b2b8d4fc",
   "metadata": {},
   "source": [
    "Notum+; frizzled 5/8-4+ muscle cells near the eye were defined as NMEs: notum+ muscle cells near the eye and those at axon choice points were defined as NMCs: notum+ muscle cells at the c point. \n",
    "\n",
    "We refer to a third population of cells associated with photoreceptor axons during regeneration as notum+ brain cells or NBCs. arrowhead is required for NBC specification and optic chiasm formation. soxP-5 is required for NME/NMC specification.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "6bb823d5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">KY024326.1 Schmidtea mediterranea arrowhead mRNA, complete cds\n",
      ">JX010525.1 Schmidtea mediterranea clone Smed_01958_V2 soxP-5 mRNA, partial cds\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "grep \">\" Scimone2020.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "16d2fc63",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db /hydra/FACS/stringtie_transcripts.fa -query Scimone2020.fa -out blast_Scimone2020 -outfmt 6 &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7a5105be",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Scimone2020 = pd.read_csv(\"/hydra/TF_data/blast_Scimone2020\", sep=\"\\t\", header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "81faa3b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "blast_Scimone2020.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Scimone2020 = blast_Scimone2020.loc[blast_Scimone2020.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "928b6304",
   "metadata": {},
   "outputs": [],
   "source": [
    "Scimone2020 = pd.DataFrame({\"ID\": [\"KY024326.1\", \"JX010525.1\"], \"Name\": [\"Awh\", \"soxP-5\"]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "f999b893",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Scimone2020 = tf_table(blast_Scimone2020, \"Scimone et al. (2020)\", Scimone2020)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "635d428b",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Scimone2020.to_csv(\"/hydra/TF_data/TF_Scimone2020.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b25f183d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>MSTRG.19831.4</td>\n",
       "      <td>SMESG000066725.1</td>\n",
       "      <td>MSTRG.19831</td>\n",
       "      <td>JX010525</td>\n",
       "      <td>soxP-5</td>\n",
       "      <td>Scimone et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>SMEST081952001.1</td>\n",
       "      <td>SMESG000081952.1</td>\n",
       "      <td>MSTRG.22894</td>\n",
       "      <td>KY024326</td>\n",
       "      <td>Awh</td>\n",
       "      <td>Scimone et al. (2020)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0     Transcript.ID           Rink.ID     Neiro.ID Original.ID  \\\n",
       "0           0     MSTRG.19831.4  SMESG000066725.1  MSTRG.19831    JX010525   \n",
       "1           1  SMEST081952001.1  SMESG000081952.1  MSTRG.22894    KY024326   \n",
       "\n",
       "  Original.name                 Source  \n",
       "0        soxP-5  Scimone et al. (2020)  \n",
       "1           Awh  Scimone et al. (2020)  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "pd.read_csv(\"/hydra/TF_data/TF_Scimone2020.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "259d4df8",
   "metadata": {},
   "source": [
    "## 7.2 Cell-type diversity and regionalized gene expression in the planarian intestine\n",
    "Forsthoefel, D.J., Cejda, N.I., Khan, U.W. and Newmark, P.A., 2020. Cell-type diversity and regionalized gene expression in the planarian intestine. Elife, 9, p.e52613."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ca4bf576",
   "metadata": {},
   "source": [
    "* Intriguingly, we found that knockdown of two other TFs, 48 related 1 (fer1) (also called pancreas transcription factor one subunit alpha or PTF1A [Fincher et al., 2018]), and LIM homeobox 2 (lhx2b), also modestly reduced goblet cells in pre-existing branches (Figure 8—figure supplement 5A–C and Supplementary file 5). mRNAs encoding these TFs are enriched in basal regions of the intestine (Figure 6—figure supplement 1). In addition, single-cell transcriptome data suggest PTF1A expression is elevated in differentiating neoblast progeny in the basal cell lineage (Fincher et al., 2018), and PTF1A also reduces basal intestinal cell numbers (Fincher et al., 2018)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4bf7ecf5",
   "metadata": {},
   "outputs": [],
   "source": [
    "#%%bash\n",
    "#cd /hydra/TF_data\n",
    "#wget https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7117911/bin/elife-52613-supp5.xlsx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "3e624851",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "Forsthoefel2020xlsx = pd.read_excel(\"/hydra/TF_data/Forsthoefel.xlsx\", sheet_name=2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aa8d1726",
   "metadata": {},
   "outputs": [],
   "source": [
    "Forsthoefel2020xlsx[\"geneID\"].to_csv(\"/hydra/TF_data/Forsthoefel2020.txt\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5f1ceafb",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data\n",
    "seqtk subseq /hydra/sexual_genome_annotation_files/sexual_genome_annotation/dd_Smed_v6.pcf.contigs.fasta Forsthoefel2020.txt > Forsthoefel2020.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4934aede",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /hydra/TF_data/\n",
    "blastn -db ~/FACS/stringtie_transcripts.fa -query Forsthoefel2020.fa -out blast_Forsthoefel2020 -outfmt 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "75170893",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "blast_Forsthoefel2020 = pd.read_csv(\"/hydra/TF_data/blast_Forsthoefel2020\" , sep=\"\\t\", header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "36753a5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "blast_Forsthoefel2020.columns = [\"Seqid\", \"qid\", \"Per\", \"Len\", \"Mis\", \"Gap\", \"Startq\", \"Endq\", \"Starts\", \"Ends\", \"E\", \"Bit\"]\n",
    "blast_Forsthoefel2020 = blast_Forsthoefel2020.loc[blast_Forsthoefel2020.groupby(\"Seqid\")[\"Bit\"].idxmax()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "92bb3546",
   "metadata": {},
   "outputs": [],
   "source": [
    "transcript_id = []\n",
    "gene_id_rink = []\n",
    "gene_id_neiro = []\n",
    "original_id = []\n",
    "original_name = []\n",
    "for i in range(len(blast_Forsthoefel2020)):\n",
    "    t_id = blast_Forsthoefel2020.iloc[i,1]\n",
    "    o_id = blast_Forsthoefel2020.iloc[i,0]\n",
    "    transcript_id.append(t_id)\n",
    "    gene_id_rink.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,0])\n",
    "    gene_id_neiro.append(gene2transcript[gene2transcript[\"qry_id\"] == t_id].iloc[0,1])\n",
    "    original_id.append(o_id)\n",
    "    original_name.append(Forsthoefel2020xlsx[Forsthoefel2020xlsx[\"geneID\"] == o_id].iloc[0,1])\n",
    "source = [\"Forsthoefel et al. (2020)\"] * len(original_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0410cf54",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Forsthoefel2020 = pd.DataFrame({\"Transcript.ID\": transcript_id, \"Rink.ID\": gene_id_rink, \"Neiro.ID\": gene_id_neiro, \"Original.ID\": original_id, \"Original.name\": original_name, \"Source\": source})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f6fb92c6",
   "metadata": {},
   "outputs": [],
   "source": [
    "TF_table_Forsthoefel2020.to_csv(\"/hydra/TF_data/TF_Forsthoefel2020.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "f6867f15",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcript.ID</th>\n",
       "      <th>Rink.ID</th>\n",
       "      <th>Neiro.ID</th>\n",
       "      <th>Original.ID</th>\n",
       "      <th>Original.name</th>\n",
       "      <th>Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MSTRG.17842.2</td>\n",
       "      <td>SMESG000059614.1</td>\n",
       "      <td>MSTRG.17842</td>\n",
       "      <td>dd_Smed_v6_10103_0_1</td>\n",
       "      <td>RREB1</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMEST040789001.1</td>\n",
       "      <td>SMESG000040789.1</td>\n",
       "      <td>MSTRG.12759</td>\n",
       "      <td>dd_Smed_v6_11118_0_1</td>\n",
       "      <td>ZNF281</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SMEST006550001.1</td>\n",
       "      <td>SMESG000006550.1</td>\n",
       "      <td>MSTRG.1647</td>\n",
       "      <td>dd_Smed_v6_12010_0_1</td>\n",
       "      <td>OVOL1</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MSTRG.6245.1</td>\n",
       "      <td>SMESG000017486.1</td>\n",
       "      <td>MSTRG.6245</td>\n",
       "      <td>dd_Smed_v6_12219_0_1</td>\n",
       "      <td>HELZ2</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SMEST009894001.1</td>\n",
       "      <td>SMESG000009894.1</td>\n",
       "      <td>MSTRG.3785</td>\n",
       "      <td>dd_Smed_v6_15178_0_1</td>\n",
       "      <td>SIX6</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SMEST014588001.1</td>\n",
       "      <td>SMESG000014588.1</td>\n",
       "      <td>MSTRG.4403</td>\n",
       "      <td>dd_Smed_v6_1694_0_1</td>\n",
       "      <td>HNF4G</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SMEST036235001.1</td>\n",
       "      <td>SMESG000036235.1</td>\n",
       "      <td>MSTRG.11351</td>\n",
       "      <td>dd_Smed_v6_2670_0_1</td>\n",
       "      <td>MED21</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MSTRG.23238.2</td>\n",
       "      <td>SMESG000079512.1</td>\n",
       "      <td>MSTRG.23238</td>\n",
       "      <td>dd_Smed_v6_2716_0_1</td>\n",
       "      <td>NKX2-2</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MSTRG.22069.1</td>\n",
       "      <td>SMESG000075102.1</td>\n",
       "      <td>MSTRG.22069</td>\n",
       "      <td>dd_Smed_v6_3443_0_1</td>\n",
       "      <td>ZFP37</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MSTRG.396.5</td>\n",
       "      <td>SMESG000025482.1</td>\n",
       "      <td>MSTRG.396</td>\n",
       "      <td>dd_Smed_v6_3638_0_1</td>\n",
       "      <td>KLF13</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MSTRG.12700.6</td>\n",
       "      <td>SMESG000039559.1</td>\n",
       "      <td>MSTRG.12700</td>\n",
       "      <td>dd_Smed_v6_4075_0_1</td>\n",
       "      <td>GATA4</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SMEST033655002.1</td>\n",
       "      <td>SMESG000033655.1</td>\n",
       "      <td>MSTRG.10558</td>\n",
       "      <td>dd_Smed_v6_4685_0_1</td>\n",
       "      <td>CREB1</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SMEST073676002.1</td>\n",
       "      <td>SMESG000073676.1</td>\n",
       "      <td>MSTRG.20458</td>\n",
       "      <td>dd_Smed_v6_5177_0_1</td>\n",
       "      <td>ZFP64</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SMEST001287001.1</td>\n",
       "      <td>SMESG000001287.1</td>\n",
       "      <td>MSTRG.1359</td>\n",
       "      <td>dd_Smed_v6_5978_0_1</td>\n",
       "      <td>NFAT5</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SMEST040081001.1</td>\n",
       "      <td>SMESG000040081.1</td>\n",
       "      <td>MSTRG.12572</td>\n",
       "      <td>dd_Smed_v6_6869_0_1</td>\n",
       "      <td>PTF1A</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MSTRG.20466.7</td>\n",
       "      <td>SMESG000073727.1</td>\n",
       "      <td>MSTRG.20466</td>\n",
       "      <td>dd_Smed_v6_7470_0_1</td>\n",
       "      <td>GLI3</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MSTRG.20883.6</td>\n",
       "      <td>SMESG000071247.1</td>\n",
       "      <td>MSTRG.20883</td>\n",
       "      <td>dd_Smed_v6_7863_0_1</td>\n",
       "      <td>MEIS2</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SMEST051880001.1</td>\n",
       "      <td>SMESG000051880.1</td>\n",
       "      <td>MSTRG.13225</td>\n",
       "      <td>dd_Smed_v6_7938_0_1</td>\n",
       "      <td>MITF</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MSTRG.564.6</td>\n",
       "      <td>SMESG000026073.1</td>\n",
       "      <td>MSTRG.564</td>\n",
       "      <td>dd_Smed_v6_8176_0_1</td>\n",
       "      <td>HELZ2</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SMEST001774001.1</td>\n",
       "      <td>SMESG000001774.1</td>\n",
       "      <td>MSTRG.1469</td>\n",
       "      <td>dd_Smed_v6_8190_0_1</td>\n",
       "      <td>LHX9</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>SMEST022856001.1</td>\n",
       "      <td>SMESG000022856.1</td>\n",
       "      <td>MSTRG.7427</td>\n",
       "      <td>dd_Smed_v6_8252_0_1</td>\n",
       "      <td>LHX1</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SMEST069596001.1</td>\n",
       "      <td>SMESG000069596.1</td>\n",
       "      <td>MSTRG.19189</td>\n",
       "      <td>dd_Smed_v6_8886_0_1</td>\n",
       "      <td>ELF3</td>\n",
       "      <td>Forsthoefel et al. (2020)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Transcript.ID           Rink.ID     Neiro.ID           Original.ID  \\\n",
       "0      MSTRG.17842.2  SMESG000059614.1  MSTRG.17842  dd_Smed_v6_10103_0_1   \n",
       "1   SMEST040789001.1  SMESG000040789.1  MSTRG.12759  dd_Smed_v6_11118_0_1   \n",
       "2   SMEST006550001.1  SMESG000006550.1   MSTRG.1647  dd_Smed_v6_12010_0_1   \n",
       "3       MSTRG.6245.1  SMESG000017486.1   MSTRG.6245  dd_Smed_v6_12219_0_1   \n",
       "4   SMEST009894001.1  SMESG000009894.1   MSTRG.3785  dd_Smed_v6_15178_0_1   \n",
       "5   SMEST014588001.1  SMESG000014588.1   MSTRG.4403   dd_Smed_v6_1694_0_1   \n",
       "6   SMEST036235001.1  SMESG000036235.1  MSTRG.11351   dd_Smed_v6_2670_0_1   \n",
       "7      MSTRG.23238.2  SMESG000079512.1  MSTRG.23238   dd_Smed_v6_2716_0_1   \n",
       "8      MSTRG.22069.1  SMESG000075102.1  MSTRG.22069   dd_Smed_v6_3443_0_1   \n",
       "9        MSTRG.396.5  SMESG000025482.1    MSTRG.396   dd_Smed_v6_3638_0_1   \n",
       "10     MSTRG.12700.6  SMESG000039559.1  MSTRG.12700   dd_Smed_v6_4075_0_1   \n",
       "11  SMEST033655002.1  SMESG000033655.1  MSTRG.10558   dd_Smed_v6_4685_0_1   \n",
       "12  SMEST073676002.1  SMESG000073676.1  MSTRG.20458   dd_Smed_v6_5177_0_1   \n",
       "13  SMEST001287001.1  SMESG000001287.1   MSTRG.1359   dd_Smed_v6_5978_0_1   \n",
       "14  SMEST040081001.1  SMESG000040081.1  MSTRG.12572   dd_Smed_v6_6869_0_1   \n",
       "15     MSTRG.20466.7  SMESG000073727.1  MSTRG.20466   dd_Smed_v6_7470_0_1   \n",
       "16     MSTRG.20883.6  SMESG000071247.1  MSTRG.20883   dd_Smed_v6_7863_0_1   \n",
       "17  SMEST051880001.1  SMESG000051880.1  MSTRG.13225   dd_Smed_v6_7938_0_1   \n",
       "18       MSTRG.564.6  SMESG000026073.1    MSTRG.564   dd_Smed_v6_8176_0_1   \n",
       "19  SMEST001774001.1  SMESG000001774.1   MSTRG.1469   dd_Smed_v6_8190_0_1   \n",
       "20  SMEST022856001.1  SMESG000022856.1   MSTRG.7427   dd_Smed_v6_8252_0_1   \n",
       "21  SMEST069596001.1  SMESG000069596.1  MSTRG.19189   dd_Smed_v6_8886_0_1   \n",
       "\n",
       "   Original.name                     Source  \n",
       "0          RREB1  Forsthoefel et al. (2020)  \n",
       "1         ZNF281  Forsthoefel et al. (2020)  \n",
       "2          OVOL1  Forsthoefel et al. (2020)  \n",
       "3          HELZ2  Forsthoefel et al. (2020)  \n",
       "4           SIX6  Forsthoefel et al. (2020)  \n",
       "5          HNF4G  Forsthoefel et al. (2020)  \n",
       "6          MED21  Forsthoefel et al. (2020)  \n",
       "7         NKX2-2  Forsthoefel et al. (2020)  \n",
       "8          ZFP37  Forsthoefel et al. (2020)  \n",
       "9          KLF13  Forsthoefel et al. (2020)  \n",
       "10         GATA4  Forsthoefel et al. (2020)  \n",
       "11         CREB1  Forsthoefel et al. (2020)  \n",
       "12         ZFP64  Forsthoefel et al. (2020)  \n",
       "13         NFAT5  Forsthoefel et al. (2020)  \n",
       "14         PTF1A  Forsthoefel et al. (2020)  \n",
       "15          GLI3  Forsthoefel et al. (2020)  \n",
       "16         MEIS2  Forsthoefel et al. (2020)  \n",
       "17          MITF  Forsthoefel et al. (2020)  \n",
       "18         HELZ2  Forsthoefel et al. (2020)  \n",
       "19          LHX9  Forsthoefel et al. (2020)  \n",
       "20          LHX1  Forsthoefel et al. (2020)  \n",
       "21          ELF3  Forsthoefel et al. (2020)  "
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"/hydra/TF_data/TF_Forsthoefel2020.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "791d2625",
   "metadata": {},
   "source": [
    "# 8. 2021"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e69d5572",
   "metadata": {},
   "source": [
    "## 8.1 Analysis of Fox genes in Schmidtea mediterranea reveals new families and a conserved role of Smed-foxO in controlling cell death\n",
    "Pascual-Carreras, E., Herrera-Úbeda, C., Rosselló, M., Coronel-Córdoba, P., Garcia-Fernàndez, J., Saló, E. and Adell, T., 2021. Analysis of Fox genes in Schmidtea mediterranea reveals new families and a conserved role of Smed-foxO in controlling cell death. Scientific reports, 11(1), pp.1-18."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1dec7ef5",
   "metadata": {},
   "source": [
    "According to the new phylogeny, the 27 Fox genes found in Schmidtea mediterranea were classified into 12 families."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "33e5e91f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#%%bash\n",
    "#cd /hydra/TF_data/\n",
    "#wget https://static-content.springer.com/esm/art%3A10.1038%2Fs41598-020-80627-0/MediaObjects/41598_2020_80627_MOESM3_ESM.xlsx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "234e2829",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "PascualCarreras2021 = pd.read_excel(\"/hydra/TF_data/41598_2020_80627_MOESM3_ESM.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "a5352c45",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Genome ID                         SMESG000065670.1/SMESG000065671.1/SMESG0000656...\n",
       "Scaffold                                                                         61\n",
       "Transciptome ID                                                dd_Smed_v6_10718_0_1\n",
       "Family                                                                            A\n",
       "New Gene Name                                                            Smed-foxA1\n",
       "Previous Name                                                             Smed-FoxA\n",
       "Other homolgs                                                 Djap-FoxA / Spol-FoxA\n",
       "Gene Bank ID                                                    JX010556 / BK013013\n",
       "Mentioned references                                              [21] / This study\n",
       "RNAi phenotype                                                   Defects in pharynx\n",
       "RNAi References                                                                [21]\n",
       "ISH pattern                                           pharynx / around the pharynx \n",
       "SCS cell type (Plass et al.)                           phyarnx / neuronal (npp 18+)\n",
       "SCS cell type (Fincher et al.)           neural / phyarnx / parenchymal / neoblasts\n",
       "Pattern References                                                             [21]\n",
       "Clade                                                                             I\n",
       "Name: 0, dtype: object"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PascualCarreras2021.iloc[0,:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "ef4bf424",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ref_gene_id</th>\n",
       "      <th>qry_gene_id</th>\n",
       "      <th>qry_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>74424</th>\n",
       "      <td>SMESG000065670.1</td>\n",
       "      <td>MSTRG.19539</td>\n",
       "      <td>MSTRG.19539.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74425</th>\n",
       "      <td>SMESG000065670.1</td>\n",
       "      <td>MSTRG.19539</td>\n",
       "      <td>MSTRG.19539.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74426</th>\n",
       "      <td>SMESG000065670.1</td>\n",
       "      <td>MSTRG.19539</td>\n",
       "      <td>SMEST065670001.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74428</th>\n",
       "      <td>SMESG000065670.1</td>\n",
       "      <td>MSTRG.19540</td>\n",
       "      <td>MSTRG.19540.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74429</th>\n",
       "      <td>SMESG000065670.1</td>\n",
       "      <td>MSTRG.19540</td>\n",
       "      <td>MSTRG.19540.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74430</th>\n",
       "      <td>SMESG000065670.1</td>\n",
       "      <td>MSTRG.19540</td>\n",
       "      <td>MSTRG.19540.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            ref_gene_id  qry_gene_id            qry_id\n",
       "74424  SMESG000065670.1  MSTRG.19539     MSTRG.19539.8\n",
       "74425  SMESG000065670.1  MSTRG.19539     MSTRG.19539.9\n",
       "74426  SMESG000065670.1  MSTRG.19539  SMEST065670001.1\n",
       "74428  SMESG000065670.1  MSTRG.19540     MSTRG.19540.1\n",
       "74429  SMESG000065670.1  MSTRG.19540     MSTRG.19540.2\n",
       "74430  SMESG000065670.1  MSTRG.19540     MSTRG.19540.3"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene2transcript[gene2transcript[\"ref_gene_id\"] == \"SMESG000065670.1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a5395c1e",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene2transcript[gene2transcript[\"ref_gene_id\"] == \"SMESG000065670.1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "3eeb4e87",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ref_gene_id</th>\n",
       "      <th>qry_gene_id</th>\n",
       "      <th>qry_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [ref_gene_id, qry_gene_id, qry_id]\n",
       "Index: []"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene2transcript[gene2transcript[\"ref_gene_id\"] == \"SMESG000065671.1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "88c9d69b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ref_gene_id</th>\n",
       "      <th>qry_gene_id</th>\n",
       "      <th>qry_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [ref_gene_id, qry_gene_id, qry_id]\n",
       "Index: []"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene2transcript[gene2transcript[\"ref_gene_id\"] == \"SMESG000065672.1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "9e605866",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "            ref_gene_id qry_gene_id            qry_id\n",
      "29837  SMESG000021762.1  MSTRG.7741      MSTRG.7741.1\n",
      "29839  SMESG000021762.1  MSTRG.7741      MSTRG.7741.3\n",
      "29840  SMESG000021762.1  MSTRG.7741  SMEST021762001.1\n",
      "29841  SMESG000021762.1  MSTRG.7741      MSTRG.7741.5\n",
      "            ref_gene_id qry_gene_id            qry_id\n",
      "29838  SMESG000021761.1  MSTRG.7741  SMEST021761001.1\n"
     ]
    }
   ],
   "source": [
    "print(gene2transcript[gene2transcript[\"ref_gene_id\"] == \"SMESG000021762.1\"])\n",
    "print(gene2transcript[gene2transcript[\"ref_gene_id\"] == \"SMESG000021761.1\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "8da6e37d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "            ref_gene_id  qry_gene_id            qry_id\n",
      "44888  SMESG000037781.1  MSTRG.11814  SMEST037781002.1\n",
      "44889  SMESG000037781.1  MSTRG.11814  SMEST037781001.1\n"
     ]
    }
   ],
   "source": [
    "print(gene2transcript[gene2transcript[\"ref_gene_id\"] == \"SMESG000037781.1\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "5f4491f3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome ID</th>\n",
       "      <th>Scaffold</th>\n",
       "      <th>Transciptome ID</th>\n",
       "      <th>Family</th>\n",
       "      <th>New Gene Name</th>\n",
       "      <th>Previous Name</th>\n",
       "      <th>Other homolgs</th>\n",
       "      <th>Gene Bank ID</th>\n",
       "      <th>Mentioned references</th>\n",
       "      <th>RNAi phenotype</th>\n",
       "      <th>RNAi References</th>\n",
       "      <th>ISH pattern</th>\n",
       "      <th>SCS cell type (Plass et al.)</th>\n",
       "      <th>SCS cell type (Fincher et al.)</th>\n",
       "      <th>Pattern References</th>\n",
       "      <th>Clade</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SMESG000065670.1/SMESG000065671.1/SMESG0000656...</td>\n",
       "      <td>61</td>\n",
       "      <td>dd_Smed_v6_10718_0_1</td>\n",
       "      <td>A</td>\n",
       "      <td>Smed-foxA1</td>\n",
       "      <td>Smed-FoxA</td>\n",
       "      <td>Djap-FoxA / Spol-FoxA</td>\n",
       "      <td>JX010556 / BK013013</td>\n",
       "      <td>[21] / This study</td>\n",
       "      <td>Defects in pharynx</td>\n",
       "      <td>[21]</td>\n",
       "      <td>pharynx / around the pharynx</td>\n",
       "      <td>phyarnx / neuronal (npp 18+)</td>\n",
       "      <td>neural / phyarnx / parenchymal / neoblasts</td>\n",
       "      <td>[21]</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMESG000067799.1</td>\n",
       "      <td>67</td>\n",
       "      <td>dd_Smed_v6_39758_0_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smed-foxA2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BK013028</td>\n",
       "      <td>This study</td>\n",
       "      <td>Mild Regeneration defects</td>\n",
       "      <td>This study</td>\n",
       "      <td>ventral discrete cells in D/V margin</td>\n",
       "      <td>otf+ cells 2 /  neuronal (npp 18+)</td>\n",
       "      <td>N.S.C.</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SMESG000028135.1</td>\n",
       "      <td>207</td>\n",
       "      <td>dd_Smed_v6_30453_0_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smed-foxA(P)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BK013026</td>\n",
       "      <td>This study</td>\n",
       "      <td>Anterior Regeneration defects</td>\n",
       "      <td>This study</td>\n",
       "      <td>N.D.</td>\n",
       "      <td>epidermal (activated early epidermal progenitors)</td>\n",
       "      <td>neural (non-ciliated neurons)</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SMESG000015673.1</td>\n",
       "      <td>153</td>\n",
       "      <td>dd_Smed_v6_16297_0_1</td>\n",
       "      <td>C</td>\n",
       "      <td>Smed-foxC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BK013019</td>\n",
       "      <td>This study</td>\n",
       "      <td>Mild Regeneration defects</td>\n",
       "      <td>This study</td>\n",
       "      <td>pharynx / discrete cells around the pharynx</td>\n",
       "      <td>muscle (muscle pharynx) / epidermal (activated...</td>\n",
       "      <td>intestine / muscle / neural</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SMESG000004239.1</td>\n",
       "      <td>113</td>\n",
       "      <td>dd_Smed_v6_18389_0_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smed-foxC2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BK013022</td>\n",
       "      <td>This study</td>\n",
       "      <td>Anterior Regeneration defects</td>\n",
       "      <td>This study</td>\n",
       "      <td>discrete cells around pharynx</td>\n",
       "      <td>muscle (muscle pharynx) / secretory 1</td>\n",
       "      <td>muscle</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SMESG000077075.1</td>\n",
       "      <td>88</td>\n",
       "      <td>dd_Smed_v6_23249_0_1</td>\n",
       "      <td>D</td>\n",
       "      <td>Smed-foxD1</td>\n",
       "      <td>Smed-FoxD</td>\n",
       "      <td>Djap-FoxD</td>\n",
       "      <td>KC577557 / BK013024</td>\n",
       "      <td>[22] / This study</td>\n",
       "      <td>Lack of anterior identity</td>\n",
       "      <td>[22-24]</td>\n",
       "      <td>anterior pole</td>\n",
       "      <td>muscle (muscle body) / epidermal (activated ea...</td>\n",
       "      <td>muscle / neural</td>\n",
       "      <td>[22-24]</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SMESG000065690.1</td>\n",
       "      <td>181</td>\n",
       "      <td>dd_Smed_v6_30720_0_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smed-foxD2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BK013027</td>\n",
       "      <td>This study</td>\n",
       "      <td>Anterior Regeneration defects</td>\n",
       "      <td>This study</td>\n",
       "      <td>N.D.</td>\n",
       "      <td>neuronal (GABA)</td>\n",
       "      <td>neural (cilialted neurons)</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SMESG000021761.1</td>\n",
       "      <td>61</td>\n",
       "      <td>dd_Smed_v6_17749_0_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smed-foxD3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BK013021</td>\n",
       "      <td>This study</td>\n",
       "      <td>Anterior Regeneration defects</td>\n",
       "      <td>This study</td>\n",
       "      <td>N.D.</td>\n",
       "      <td>muscle (muscle body) / epidermal (activated ea...</td>\n",
       "      <td>muscle</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SMESG000075929.1</td>\n",
       "      <td>85</td>\n",
       "      <td>dd_Smed_v6_15035_0_1</td>\n",
       "      <td>F</td>\n",
       "      <td>Smed-foxF1</td>\n",
       "      <td>Smed-FoxF-like</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KJ934801 / BK013018</td>\n",
       "      <td>[25] / This study</td>\n",
       "      <td>Mild Regeneration defects</td>\n",
       "      <td>This study</td>\n",
       "      <td>dicrete cells in anterior D/V margin / diecret...</td>\n",
       "      <td>muscle (muscle pharynx / muscle body)</td>\n",
       "      <td>muscle</td>\n",
       "      <td>[25] / This study</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SMESG000066497.1</td>\n",
       "      <td>63</td>\n",
       "      <td>dd_Smed_v6_6910_0_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smed-foxF2</td>\n",
       "      <td>Smed-FoxF-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MH392337 / BK013034</td>\n",
       "      <td>[26] / This study</td>\n",
       "      <td>Defects in muscular system and pigment cells</td>\n",
       "      <td>[26-27]</td>\n",
       "      <td>ubiquitous</td>\n",
       "      <td>parenchymal (aqp / psap/ pigment)</td>\n",
       "      <td>cathepsin+ / muscle</td>\n",
       "      <td>[26-27]</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SMESG000010270.1</td>\n",
       "      <td>133</td>\n",
       "      <td>dd_Smed_v6_16466_0_1</td>\n",
       "      <td>G</td>\n",
       "      <td>Smed-foxG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Djap-FoxG</td>\n",
       "      <td>BK013020</td>\n",
       "      <td>This study</td>\n",
       "      <td>Mild posterior regeneration defects</td>\n",
       "      <td>This study</td>\n",
       "      <td>dicrete cells at D/V margin / dorsal midline /...</td>\n",
       "      <td>muscle (muscle body / muscle pharynx) / otf+ c...</td>\n",
       "      <td>muscle</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SMESG000021434.1</td>\n",
       "      <td>17</td>\n",
       "      <td>dd_Smed_v6_19255_0_1</td>\n",
       "      <td>L1</td>\n",
       "      <td>Smed-foxL1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BK013023</td>\n",
       "      <td>This study</td>\n",
       "      <td>Mild Regeneration defects</td>\n",
       "      <td>This study</td>\n",
       "      <td>N.D.</td>\n",
       "      <td>muscle (muscle pharynx)</td>\n",
       "      <td>N.S.C.</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SMESG000062929.1</td>\n",
       "      <td>57</td>\n",
       "      <td>dd_Smed_v6_50245_0_1</td>\n",
       "      <td>Q/D</td>\n",
       "      <td>Smed-foxQ/D</td>\n",
       "      <td>Smed-FoxQ2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JX091076 / BK013031</td>\n",
       "      <td>[25] / This study</td>\n",
       "      <td>Defects in eyes</td>\n",
       "      <td>[25]</td>\n",
       "      <td>eyes / scatered discrete cells following the eyes</td>\n",
       "      <td>pharynx / epidermal (epidermis DVb)</td>\n",
       "      <td>neural (non-ciliated neurons)</td>\n",
       "      <td>[25] [28-29]</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SMESG000072183.1</td>\n",
       "      <td>76</td>\n",
       "      <td>dd_Smed_v6_14635_0_1</td>\n",
       "      <td>J1</td>\n",
       "      <td>Smed-foxJ1-1</td>\n",
       "      <td>Smed-FoxJ1-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JX842843</td>\n",
       "      <td>[30]</td>\n",
       "      <td>N.P.</td>\n",
       "      <td>[30]</td>\n",
       "      <td>Mid dorsal stripe</td>\n",
       "      <td>pharynx</td>\n",
       "      <td>pharynx</td>\n",
       "      <td>[30]</td>\n",
       "      <td>II</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SMESG000072873.1</td>\n",
       "      <td>78</td>\n",
       "      <td>dd_Smed_v6_99867_0_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smed-foxJ1-2</td>\n",
       "      <td>Smed-FoxJ1-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JX842844</td>\n",
       "      <td>[30]</td>\n",
       "      <td>N.P.</td>\n",
       "      <td>[30]</td>\n",
       "      <td>Mid dorsal stripe</td>\n",
       "      <td>parenchymal (aqp+) / epidermal (epidermis)</td>\n",
       "      <td>N.S.C.</td>\n",
       "      <td>[30]</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SMESG000067319.1</td>\n",
       "      <td>65</td>\n",
       "      <td>dd_Smed_v6_103874_0_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smed-foxJ1-3</td>\n",
       "      <td>Smed-FoxJ1-3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JX842845</td>\n",
       "      <td>[30]</td>\n",
       "      <td>N.P.</td>\n",
       "      <td>[30]</td>\n",
       "      <td>N.D.</td>\n",
       "      <td>muscle (muscle body)</td>\n",
       "      <td>N.S.C.</td>\n",
       "      <td>[30]</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SMESG000010030.1</td>\n",
       "      <td>132</td>\n",
       "      <td>dd_Smed_v6_10152_0_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smed-foxJ1-4</td>\n",
       "      <td>Smed-FoxJ1-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JX842846</td>\n",
       "      <td>[30]</td>\n",
       "      <td>Edemas and Cilia defects</td>\n",
       "      <td>[30]</td>\n",
       "      <td>Ventral widely</td>\n",
       "      <td>neuronal (GABA / ChAT / spp11+) / protonephrid...</td>\n",
       "      <td>neural / pharynx / epidermal</td>\n",
       "      <td>[30]</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SMESG000017088.1</td>\n",
       "      <td>15</td>\n",
       "      <td>dd_Smed_v6_13009_0_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smed-foxJ1-5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BK013017</td>\n",
       "      <td>This study</td>\n",
       "      <td>Mild Regeneration defects</td>\n",
       "      <td>This study</td>\n",
       "      <td>Dorsal widely</td>\n",
       "      <td>neuronal (GABA / ChAT) / protonephridia</td>\n",
       "      <td>N.S.C.</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>SMESG000061695.1</td>\n",
       "      <td>55</td>\n",
       "      <td>dd_Smed_v6_4500_0_1</td>\n",
       "      <td>K</td>\n",
       "      <td>Smed-foxK1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BK013030</td>\n",
       "      <td>This study</td>\n",
       "      <td>Anterior Regeneration defects</td>\n",
       "      <td>This study</td>\n",
       "      <td>CNS</td>\n",
       "      <td>parenchymal (pigment) / secretory 4 / neuronal...</td>\n",
       "      <td>parenchymal</td>\n",
       "      <td>[31]</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SMESG000040645.1</td>\n",
       "      <td>298</td>\n",
       "      <td>dd_Smed_v6_5767_0_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smed-foxK2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BK013032</td>\n",
       "      <td>This study</td>\n",
       "      <td>Anterior Regeneration defects</td>\n",
       "      <td>This study</td>\n",
       "      <td>CNS</td>\n",
       "      <td>parenchymal (aqp+ / psap+ / pigment / glia) / ...</td>\n",
       "      <td>N.S.C.</td>\n",
       "      <td>[31]</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>SMESG000064173.1</td>\n",
       "      <td>5</td>\n",
       "      <td>dd_Smed_v6_7583_0_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smed-foxK3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BK013035</td>\n",
       "      <td>This study</td>\n",
       "      <td>Anterior Regeneration defects</td>\n",
       "      <td>This study</td>\n",
       "      <td>CNS</td>\n",
       "      <td>neuronal (spp11+)</td>\n",
       "      <td>N.S.C.</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SMESG000048193.1</td>\n",
       "      <td>36</td>\n",
       "      <td>dd_Smed_v6_11337_0_1</td>\n",
       "      <td>N2/3</td>\n",
       "      <td>Smed-foxN2/3-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BK013014</td>\n",
       "      <td>This study</td>\n",
       "      <td>Anterior Regeneration defects</td>\n",
       "      <td>This study</td>\n",
       "      <td>CNS</td>\n",
       "      <td>neuronal (ChAT) / secretory 4 / protonephridia...</td>\n",
       "      <td>neural / neoblasts</td>\n",
       "      <td>[31] / This study</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>SMESG000015734.1</td>\n",
       "      <td>154</td>\n",
       "      <td>dd_Smed_v6_13005_0_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smed-foxN2/3-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BK013016</td>\n",
       "      <td>This study</td>\n",
       "      <td>Anterior Regeneration defects</td>\n",
       "      <td>This study</td>\n",
       "      <td>CNS</td>\n",
       "      <td>parenchymal (aqp+ / glia) / neuronal (GABA / n...</td>\n",
       "      <td>parenchymal</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SMESG000033244.1</td>\n",
       "      <td>238</td>\n",
       "      <td>dd_Smed_v6_4078_0_2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smed-foxN2/3-3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BK013029</td>\n",
       "      <td>This study</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>N.D.</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>SMESG000044628.1</td>\n",
       "      <td>31</td>\n",
       "      <td>dd_Smed_v6_12170_0_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smed-foxN(P)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BK013015</td>\n",
       "      <td>This study</td>\n",
       "      <td>Anterior Regeneration defects</td>\n",
       "      <td>This study</td>\n",
       "      <td>CNS (branches)</td>\n",
       "      <td>neuronal (npp 18+ / spp11+ ) / secretory 2 / e...</td>\n",
       "      <td>neural</td>\n",
       "      <td>This study</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>SMESG000037781.1</td>\n",
       "      <td>26</td>\n",
       "      <td>dd_Smed_v6_3040_0_1</td>\n",
       "      <td>O</td>\n",
       "      <td>Smed-foxO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BK013025</td>\n",
       "      <td>This study</td>\n",
       "      <td>Defects in cell death</td>\n",
       "      <td>This study</td>\n",
       "      <td>ubiquitous</td>\n",
       "      <td>neuronal (npp 18+ / spp11+ ) / parenchymal (aq...</td>\n",
       "      <td>cathepsin+ / neoblasts</td>\n",
       "      <td>[31] / This study</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>SMESG000068148.1</td>\n",
       "      <td>68</td>\n",
       "      <td>dd_Smed_v6_6316_0_1</td>\n",
       "      <td>P</td>\n",
       "      <td>Smed-foxP</td>\n",
       "      <td>Smed-Albino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KX531078 / BK013033</td>\n",
       "      <td>[32] / This study</td>\n",
       "      <td>Defects in pigment cells</td>\n",
       "      <td>[27] [32]</td>\n",
       "      <td>ubiquitous</td>\n",
       "      <td>parenchymal (pigment)</td>\n",
       "      <td>cathepsin+ / neoblasts</td>\n",
       "      <td>[27] [31-32]</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Genome ID  Scaffold  \\\n",
       "0   SMESG000065670.1/SMESG000065671.1/SMESG0000656...        61   \n",
       "1                                    SMESG000067799.1        67   \n",
       "2                                    SMESG000028135.1       207   \n",
       "3                                    SMESG000015673.1       153   \n",
       "4                                    SMESG000004239.1       113   \n",
       "5                                    SMESG000077075.1        88   \n",
       "6                                    SMESG000065690.1       181   \n",
       "7                                    SMESG000021761.1        61   \n",
       "8                                    SMESG000075929.1        85   \n",
       "9                                    SMESG000066497.1        63   \n",
       "10                                   SMESG000010270.1       133   \n",
       "11                                   SMESG000021434.1        17   \n",
       "12                                   SMESG000062929.1        57   \n",
       "13                                   SMESG000072183.1        76   \n",
       "14                                   SMESG000072873.1        78   \n",
       "15                                   SMESG000067319.1        65   \n",
       "16                                   SMESG000010030.1       132   \n",
       "17                                   SMESG000017088.1        15   \n",
       "18                                   SMESG000061695.1        55   \n",
       "19                                   SMESG000040645.1       298   \n",
       "20                                   SMESG000064173.1         5   \n",
       "21                                   SMESG000048193.1        36   \n",
       "22                                   SMESG000015734.1       154   \n",
       "23                                  SMESG000033244.1        238   \n",
       "24                                   SMESG000044628.1        31   \n",
       "25                                   SMESG000037781.1        26   \n",
       "26                                   SMESG000068148.1        68   \n",
       "\n",
       "          Transciptome ID Family   New Gene Name   Previous Name  \\\n",
       "0    dd_Smed_v6_10718_0_1      A      Smed-foxA1       Smed-FoxA   \n",
       "1    dd_Smed_v6_39758_0_1    NaN      Smed-foxA2             NaN   \n",
       "2    dd_Smed_v6_30453_0_1    NaN    Smed-foxA(P)             NaN   \n",
       "3    dd_Smed_v6_16297_0_1      C      Smed-foxC1             NaN   \n",
       "4    dd_Smed_v6_18389_0_1    NaN      Smed-foxC2             NaN   \n",
       "5    dd_Smed_v6_23249_0_1      D      Smed-foxD1       Smed-FoxD   \n",
       "6    dd_Smed_v6_30720_0_1    NaN      Smed-foxD2             NaN   \n",
       "7    dd_Smed_v6_17749_0_1    NaN      Smed-foxD3             NaN   \n",
       "8    dd_Smed_v6_15035_0_1      F      Smed-foxF1  Smed-FoxF-like   \n",
       "9   dd_Smed_v6_6910_0_1      NaN      Smed-foxF2     Smed-FoxF-1   \n",
       "10   dd_Smed_v6_16466_0_1      G       Smed-foxG             NaN   \n",
       "11   dd_Smed_v6_19255_0_1     L1      Smed-foxL1             NaN   \n",
       "12   dd_Smed_v6_50245_0_1    Q/D     Smed-foxQ/D      Smed-FoxQ2   \n",
       "13   dd_Smed_v6_14635_0_1     J1    Smed-foxJ1-1    Smed-FoxJ1-1   \n",
       "14   dd_Smed_v6_99867_0_1    NaN    Smed-foxJ1-2    Smed-FoxJ1-2   \n",
       "15  dd_Smed_v6_103874_0_1    NaN    Smed-foxJ1-3    Smed-FoxJ1-3   \n",
       "16   dd_Smed_v6_10152_0_1    NaN    Smed-foxJ1-4    Smed-FoxJ1-4   \n",
       "17   dd_Smed_v6_13009_0_1    NaN    Smed-foxJ1-5             NaN   \n",
       "18    dd_Smed_v6_4500_0_1      K      Smed-foxK1             NaN   \n",
       "19    dd_Smed_v6_5767_0_1    NaN      Smed-foxK2             NaN   \n",
       "20    dd_Smed_v6_7583_0_1    NaN      Smed-foxK3             NaN   \n",
       "21   dd_Smed_v6_11337_0_1   N2/3  Smed-foxN2/3-1             NaN   \n",
       "22   dd_Smed_v6_13005_0_1    NaN  Smed-foxN2/3-2             NaN   \n",
       "23    dd_Smed_v6_4078_0_2    NaN  Smed-foxN2/3-3             NaN   \n",
       "24   dd_Smed_v6_12170_0_1    NaN    Smed-foxN(P)             NaN   \n",
       "25    dd_Smed_v6_3040_0_1      O       Smed-foxO             NaN   \n",
       "26   dd_Smed_v6_6316_0_1       P       Smed-foxP     Smed-Albino   \n",
       "\n",
       "            Other homolgs         Gene Bank ID Mentioned references  \\\n",
       "0   Djap-FoxA / Spol-FoxA  JX010556 / BK013013    [21] / This study   \n",
       "1                     NaN             BK013028           This study   \n",
       "2                     NaN             BK013026           This study   \n",
       "3                     NaN             BK013019           This study   \n",
       "4                     NaN             BK013022           This study   \n",
       "5               Djap-FoxD  KC577557 / BK013024    [22] / This study   \n",
       "6                     NaN             BK013027           This study   \n",
       "7                     NaN             BK013021           This study   \n",
       "8                     NaN  KJ934801 / BK013018    [25] / This study   \n",
       "9                     NaN  MH392337 / BK013034    [26] / This study   \n",
       "10              Djap-FoxG             BK013020           This study   \n",
       "11                    NaN             BK013023           This study   \n",
       "12                    NaN  JX091076 / BK013031    [25] / This study   \n",
       "13                    NaN             JX842843                 [30]   \n",
       "14                    NaN             JX842844                 [30]   \n",
       "15                    NaN             JX842845                 [30]   \n",
       "16                    NaN             JX842846                 [30]   \n",
       "17                    NaN             BK013017           This study   \n",
       "18                    NaN             BK013030           This study   \n",
       "19                    NaN             BK013032           This study   \n",
       "20                    NaN             BK013035           This study   \n",
       "21                    NaN             BK013014           This study   \n",
       "22                    NaN             BK013016           This study   \n",
       "23                    NaN             BK013029           This study   \n",
       "24                    NaN             BK013015           This study   \n",
       "25                    NaN             BK013025           This study   \n",
       "26                    NaN  KX531078 / BK013033    [32] / This study   \n",
       "\n",
       "                                  RNAi phenotype RNAi References  \\\n",
       "0                             Defects in pharynx            [21]   \n",
       "1                      Mild Regeneration defects      This study   \n",
       "2                  Anterior Regeneration defects      This study   \n",
       "3                      Mild Regeneration defects      This study   \n",
       "4                  Anterior Regeneration defects      This study   \n",
       "5                      Lack of anterior identity         [22-24]   \n",
       "6                  Anterior Regeneration defects      This study   \n",
       "7                  Anterior Regeneration defects      This study   \n",
       "8                      Mild Regeneration defects      This study   \n",
       "9   Defects in muscular system and pigment cells         [26-27]   \n",
       "10           Mild posterior regeneration defects      This study   \n",
       "11                     Mild Regeneration defects      This study   \n",
       "12                               Defects in eyes            [25]   \n",
       "13                                          N.P.            [30]   \n",
       "14                                          N.P.            [30]   \n",
       "15                                          N.P.            [30]   \n",
       "16                      Edemas and Cilia defects            [30]   \n",
       "17                     Mild Regeneration defects      This study   \n",
       "18                 Anterior Regeneration defects      This study   \n",
       "19                 Anterior Regeneration defects      This study   \n",
       "20                 Anterior Regeneration defects      This study   \n",
       "21                 Anterior Regeneration defects      This study   \n",
       "22                 Anterior Regeneration defects      This study   \n",
       "23                                             -               -   \n",
       "24                 Anterior Regeneration defects      This study   \n",
       "25                         Defects in cell death      This study   \n",
       "26                      Defects in pigment cells       [27] [32]   \n",
       "\n",
       "                                          ISH pattern  \\\n",
       "0                       pharynx / around the pharynx    \n",
       "1                ventral discrete cells in D/V margin   \n",
       "2                                                N.D.   \n",
       "3         pharynx / discrete cells around the pharynx   \n",
       "4                       discrete cells around pharynx   \n",
       "5                                       anterior pole   \n",
       "6                                                N.D.   \n",
       "7                                                N.D.   \n",
       "8   dicrete cells in anterior D/V margin / diecret...   \n",
       "9                                          ubiquitous   \n",
       "10  dicrete cells at D/V margin / dorsal midline /...   \n",
       "11                                               N.D.   \n",
       "12  eyes / scatered discrete cells following the eyes   \n",
       "13                                  Mid dorsal stripe   \n",
       "14                                  Mid dorsal stripe   \n",
       "15                                               N.D.   \n",
       "16                                     Ventral widely   \n",
       "17                                      Dorsal widely   \n",
       "18                                                CNS   \n",
       "19                                                CNS   \n",
       "20                                                CNS   \n",
       "21                                                CNS   \n",
       "22                                                CNS   \n",
       "23                                               N.D.   \n",
       "24                                     CNS (branches)   \n",
       "25                                         ubiquitous   \n",
       "26                                         ubiquitous   \n",
       "\n",
       "                         SCS cell type (Plass et al.)  \\\n",
       "0                        phyarnx / neuronal (npp 18+)   \n",
       "1                  otf+ cells 2 /  neuronal (npp 18+)   \n",
       "2   epidermal (activated early epidermal progenitors)   \n",
       "3   muscle (muscle pharynx) / epidermal (activated...   \n",
       "4               muscle (muscle pharynx) / secretory 1   \n",
       "5   muscle (muscle body) / epidermal (activated ea...   \n",
       "6                                     neuronal (GABA)   \n",
       "7   muscle (muscle body) / epidermal (activated ea...   \n",
       "8               muscle (muscle pharynx / muscle body)   \n",
       "9                   parenchymal (aqp / psap/ pigment)   \n",
       "10  muscle (muscle body / muscle pharynx) / otf+ c...   \n",
       "11                            muscle (muscle pharynx)   \n",
       "12                pharynx / epidermal (epidermis DVb)   \n",
       "13                                            pharynx   \n",
       "14         parenchymal (aqp+) / epidermal (epidermis)   \n",
       "15                               muscle (muscle body)   \n",
       "16  neuronal (GABA / ChAT / spp11+) / protonephrid...   \n",
       "17           neuronal (GABA / ChAT) / protonephridia    \n",
       "18  parenchymal (pigment) / secretory 4 / neuronal...   \n",
       "19  parenchymal (aqp+ / psap+ / pigment / glia) / ...   \n",
       "20                                  neuronal (spp11+)   \n",
       "21  neuronal (ChAT) / secretory 4 / protonephridia...   \n",
       "22  parenchymal (aqp+ / glia) / neuronal (GABA / n...   \n",
       "23                                                  -   \n",
       "24  neuronal (npp 18+ / spp11+ ) / secretory 2 / e...   \n",
       "25  neuronal (npp 18+ / spp11+ ) / parenchymal (aq...   \n",
       "26                              parenchymal (pigment)   \n",
       "\n",
       "                SCS cell type (Fincher et al.) Pattern References Clade  \n",
       "0   neural / phyarnx / parenchymal / neoblasts               [21]     I  \n",
       "1                                       N.S.C.         This study   NaN  \n",
       "2                neural (non-ciliated neurons)         This study   NaN  \n",
       "3                  intestine / muscle / neural         This study   NaN  \n",
       "4                                       muscle         This study   NaN  \n",
       "5                              muscle / neural            [22-24]   NaN  \n",
       "6                   neural (cilialted neurons)         This study   NaN  \n",
       "7                                       muscle         This study   NaN  \n",
       "8                                       muscle  [25] / This study   NaN  \n",
       "9                          cathepsin+ / muscle            [26-27]   NaN  \n",
       "10                                      muscle         This study   NaN  \n",
       "11                                      N.S.C.         This study   NaN  \n",
       "12               neural (non-ciliated neurons)       [25] [28-29]   NaN  \n",
       "13                                     pharynx               [30]    II  \n",
       "14                                      N.S.C.               [30]   NaN  \n",
       "15                                      N.S.C.               [30]   NaN  \n",
       "16                neural / pharynx / epidermal               [30]   NaN  \n",
       "17                                      N.S.C.         This study   NaN  \n",
       "18                                 parenchymal               [31]   NaN  \n",
       "19                                      N.S.C.               [31]   NaN  \n",
       "20                                      N.S.C.         This study   NaN  \n",
       "21                          neural / neoblasts  [31] / This study   NaN  \n",
       "22                                 parenchymal         This study   NaN  \n",
       "23                                           -                  -   NaN  \n",
       "24                                      neural         This study   NaN  \n",
       "25                      cathepsin+ / neoblasts  [31] / This study   NaN  \n",
       "26                      cathepsin+ / neoblasts       [27] [31-32]   NaN  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PascualCarreras2021.iloc[:,:]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2dde6ff9",
   "metadata": {},
   "source": [
    "## 8.2 Planarian stem cells specify fate yet retain potency during the cell cycle\n",
    "Raz, A.A., Wurtzel, O. and Reddien, P.W., 2021. Planarian stem cells specify fate yet retain potency during the cell cycle. Cell Stem Cell.\n",
    "\n",
    "\n",
    "* Expression of fate-specifying factors is correlated with cell-cycle progression\n",
    "* "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d5bda628",
   "metadata": {},
   "source": [
    "* soxP-2\tdd_8483\tWagner 2012\n",
    "* znf333\tdd_13890\tWurtzel 2015\n",
    "* zfp-1\tdd_8720\tWagner 2012\n",
    "* soxP-3\tdd_5942\tWagner 2012\n",
    "* p53\tdd_5563\tPearson 2010\n",
    "* POU2/3\tdd_8234\tScimone 2011\n",
    "* six1/2-2\tdd_9774\tScimone 2011\n",
    "* FoxD\tdd_23249\tVogg 2014, Scimone 2014\n",
    "* prep\tdd_8606\tFelix 2010, Scimone 2014\n",
    "* myoD\tdd_12634 \tCowles 2013, Scimone 2017\n",
    "* soxP-5\tdd_9050\tWenemoser 2012, Fincher 2018, Scimone 2020\n",
    "* gata4/5/6-2\tdd_9910\tGonzalez-Sastre 2017, Scimone 2018\n",
    "* foxF-1\tdd_6910\tScimone 2018\n",
    "* nkx1-1\tdd_19327\tScimone 2017\n",
    "* foxA\tdd_10718\tWenemoser 2012, Adler 2014\n",
    "* meis\tdd_16375\tLapan 2012, Scimone 2014\n",
    "* hnf4\tdd_1694\tWagner 2011\n",
    "* gata4/5/6-1\tdd_4075\tWagner 2011, Gonzalez-Sastre 2017\n",
    "* nkx2.2\tdd_2716\tForsthoefel 2012\n",
    "* prox-1\tdd_13772\tScimone 2014\n",
    "* pax6A\tdd_17726\tWenemoser 2012, Scimone 2014\n",
    "* pax6B\tdd_35892\tScimone 2014\n",
    "* otxA\tdd_14633 \tLapan 2011, Scimone 2014\n",
    "* nkx2\tdd_13898\tScimone 2014\n",
    "* nkx6\tdd_11198\tScimone 2014\n",
    "* elf-1\tdd_14611\tScimone 2014\n",
    "* six3\tdd_15178\tWenemoser 2012, Scimone 2014, Pineda 1999\n",
    "* lhx3/4\tdd_16893\tScimone 2014\n",
    "* Tcf/Lef-1\tdd_13056\tScimone 2014, Stuckemann 2017\n",
    "* neuroD -1\tdd_21717\tScimone 2014, Monjo 2015, Fincher 2018\n",
    "* soxB-2\tdd_25321\tScimone 2014\n",
    "* lhx2/9\tdd_15144\tScimone 2014\n",
    "* soxB\tdd_13215 \tLapan 2012, Scimone 2014\n",
    "* ski-3\tdd_10394\tMolinaro 2016\n",
    "* ovo\tdd_48430 \tLapan 2012\n",
    "* sp6-9\tdd_17385\tLapan 2011\n",
    "* pitx\tdd_15253\tCurrie 2013, ﻿März 2013\n",
    "* coe\tdd_9893\tCowles 2013, Scimone 2014\n",
    "* ap-2\tdd_15104\tWenemoser 2012, Scimone 2014\n",
    "* klf\tdd_95726 \tLapan 2012, Scimone 2014\n",
    "* pax3/7\tdd_21801\tScimone 2014\n",
    "* pou4l-1\tdd_15555\tCowles 2014\n",
    "* runt-1\tdd_16222\tWenemoser 2012"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "33683de4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
